Load packages and functions. Some packages need to be loaded in this order, or they will clash with other packages. So, keep this order.  
```{r message=FALSE, warning=FALSE, include=FALSE, results='hide'}
setwd("C:/Users/Loay Jabre/Desktop/PhD/Data/FragTranscriptome/Transcriptome")
library(stringr)
library(ggplot2)
library(plyr)
library(dplyr)
library(gplots)
library(heatmap.plus)
library(RColorBrewer)
library(reshape2)
library(scales)
library(tidyr)
library(plotly)
library(gsubfn)
library(ggrepel)
library(gridExtra)
library(htmlwidgets)
library(VennDiagram)
library(forcats)
library(caroline)
library(edgeR)
library(data.table)
library(stringi)
library(seqinr)

source("fragtranscript_functions.R")

#library(scatterplot3d)
library(rgl)
library(calibrate)
#library(manta)
```
 
Upload ORF  data
```{r message=FALSE, warning=FALSE, include=FALSE}
all_ORF <- read.csv("tfg_de_annotation_frag_pseudoTFG.grpnorm_mmetsp_fc_pn_reclassified.csv", header = TRUE, sep = "," )
#all_ORF <- read.csv("tfg_de_annotation_allTFG.grpnorm_mmetsp_fc_pn_reclassified.csv", header = TRUE, sep = "," )
colnames(all_ORF)[91] <- "Taxa.group"
#create a 'name' column that includes all the annotations from the different libraries
all_ORF$name <- paste(all_ORF$orf_id, all_ORF$cluster, all_ORF$kegg_desc, all_ORF$KO_desc, all_ORF$KO_pathway, all_ORF$KOG_desc, all_ORF$KOG_class, all_ORF$best_hit_annotation, all_ORF$PFams_desc, all_ORF$TIGRFams_desc)

#subset the all_ORF dataset into a D.F containing only the columns I need 
all_ORF_sub <- all_ORF [, c(47,48, 91, 178, 111:128, 2:46)]
all_ORF_sub <- all_ORF_sub [, -c(23, 26, 29, 32, 35, 38, 41, 44, 47, 50, 53, 56, 59, 62, 65)]
all_ORF_sub <- all_ORF_sub[,c(1, 3:52, 2)]

#rename the different treatment columns so headings make more sense
#names(all_ORF_sub)[4:21] <- c("noFe_05 _A","noFe_05_B", "noFe_05_C","Fe_05_A","Fe_05_B", "Fe_05_C", "noFe_3_A","noFe_3_B", "noFe_3_C","Fe_3_A","Fe_3_B", "Fe_3_C", "noFe_6_A","noFe_6_B", "noFe_6_C","Fe_6_A","Fe_6_B", "Fe_6_C")

#subset to have the data only for Frag and pseudo-nitzs. 
frag_ORF <- dplyr::filter (all_ORF_sub, grepl('Fragilariopsis', Taxa.group))
pseudo_ORF <- dplyr::filter (all_ORF_sub, grepl('Pseudo-nitzschi', Taxa.group))
pseudo_frag_ORF <- dplyr::filter (all_ORF_sub, grepl('Pseudo-nit|Fragilariopsis',Taxa.group))
#all_diatoms_ORF <- dplyr::filter (all_ORF_sub, grepl('Pseudo-nit|Fragilariopsis|Diatom',Taxa.group))
```

upload MCL cluster data
```{r message=FALSE, warning=FALSE, include=FALSE}
#do the same for the MCL dataset
all_MCL <- read.csv("MCL_tfg_de_annotation_frag-pseudo_allTFG_grpnorm_mmetsp_fc_pn_reclassified.csv", header = TRUE, sep = "," )
colnames(all_MCL)[47] <- "Taxa.group"
all_MCL$name <- paste (all_MCL$cluster, all_MCL$ann_desc, all_MCL$Taxa.group)
all_MCL_sub <- all_MCL [,c(48,47,148, 74:91, 2:46 )]
all_MCL_sub <- all_MCL_sub [, -c(22, 25, 28, 31, 34, 37, 40, 43, 46,49, 52, 55, 58, 61, 64)]
frag_MCL <- dplyr::filter (all_MCL_sub, grepl('Fragilariopsis', Taxa.group))
pseudo_MCL <- dplyr::filter (all_MCL_sub, grepl('Pseudo-nit', Taxa.group))
pseudo_frag_MCL <- dplyr::filter (all_MCL_sub, grepl('Pseudo-nit|Fragilariopsis', Taxa.group))
```

Piechart for number of DE transcrips  
```{r}
#make pie chart showing percentage of reads assigned to each diatom taxon
#This however does not mean there is more Frag in the water, it just means more reads were assigned to frag. 

#unique(all_ORF$Taxa.group)
#we want percent contribution for T0, so we'll pull data only for T0, and remove any ORFs that have 0 abundance in the triplicates. We can do this for all other Ts

#all_ORF_T0 <- all_ORF [c(91, 93:95)] #T0
#all_ORF_T0 <- all_ORF [c(91, 92)] #IEE
#all_ORF_T0$sum <- rowSums(all_ORF_T0[c(2)])

#all_ORF_T0 <- all_ORF [c(91, 111:113)] #0C_noFe
#all_ORF_T0 <- all_ORF [c(91, 114:116)] #0C_Fe
#all_ORF_T0 <- all_ORF [c(91, 117:119)] #3C_noFe
#all_ORF_T0 <- all_ORF [c(91, 120:122)] #3C_Fe
#all_ORF_T0 <- all_ORF [c(91, 123:125)] #6C_noFe
all_ORF_T0 <- all_ORF [c(91, 126:128)] #6C_Fe
all_ORF_T0$sum <- rowSums(all_ORF_T0[c(2:4)])

all_ORF_T0_initial <- subset(all_ORF_T0,(all_ORF_T0$sum != 0))

x <- dplyr::filter (all_ORF_T0_initial, !grepl('Fragilariopsis|Pseudo-nitzschia|Dinophyta|Ciliophora|Diatom|Rhodobacter|Alteromonadales', Taxa.group))

a<- sum(all_ORF_T0_initial$Taxa.group == "Fragilariopsis")
b <- sum(all_ORF_T0_initial$Taxa.group == "Pseudo-nitzschia")
c <- sum(all_ORF_T0_initial$Taxa.group == "Diatom")
d <- sum(all_ORF_T0_initial$Taxa.group == "Dinophyta")
e <- sum(all_ORF_T0_initial$Taxa.group == "Ciliophora")
g <- sum(all_ORF$Taxa.group == "Rhodobacterales")
#h <- sum(all_ORF$Taxa.group == "SAR11")
i<- sum(all_ORF$Taxa.group == "Alteromonadales")
f<- NROW(x)


slices <- c(a,b,c,d,e,g,i,f)
lbls <- c("Fragilariopsis", "Pseudo-nitzschia","Other Diatoms", "Dinophyta", "Ciliophora", "Rhodobacterales", "Alteromonodales", "Other Taxa")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 

#png(filename="piechart_T5_6C_Fe.png",width=70, height=45, units="cm", res = 900)
pie(slices,
    labels = lbls,
    col=gray.colors(length(lbls)),
    main="6C_+Fe", #title
    cex = 1) #font size
 
#dev.off()
```

LHCs_Figure3
```{r}
#load the LHC annotations from BEV
lhc_annotations <- read.csv("LHC_annotation_bev.csv", header=T)

#subset the ice-edge, T0 and T1 measurements. 
all_ORF_t1 <- all_ORF [,c(47,91, 178, 92:96,98,99,101,102,104 )]
names(all_ORF_t1) <- c("orf_id","taxa","name","ice-edge","T0_A", "T0_B", "T0_C", "T1_noFe_0","T1_Fe_0","T1_noFe_3","T1_Fe_3","T1_noFe_6","T1_Fe_6")

all_MCL_t1 <- all_MCL[,c(48, 47, 148,55:59,61,62,64,65,67)] #no T0
names(all_MCL_t1) <- c("cluster","taxa","name", "ice-edge","T0_A", "T0_B", "T0_C", "T1_noFe_0","T1_Fe_0","T1_noFe_3","T1_Fe_3","T1_noFe_6","T1_Fe_6")
##.............................###

##############LHCs##############
#LHCr-cluster_3363
#assign Bev's annotations to TFG LHC clusters
#first make a map of all the LHC clusters to  manually asign them an LHC annotation
all_MCL_t1$cluster <- as.character(all_MCL_t1$cluster)
lhc_annotation_map <- c("clust_1044" = "LHCf_III",
                        "clust_1084" = "LHCr", 
                        "clust_1194" = "LHCf_I", 
                        "clust_1236" = "LHCf_I", 
                        "clust_1787" = "LHCx_4/6", 
                        "clust_1938" = "LHCr", 
                        "clust_2256" = "LHCx_1/6", 
                        "clust_2549" = "LHCr",
                        "clust_3219" = "LHCr", 
                        "clust_3282" = "LHCf_II", 
                        "clust_332"  = "LHCf_I", 
                        "clust_3363" = "LHCr", 
                        "clust_3551" = "LHCr", 
                        "clust_402"  = "LHCf_I", 
                        "clust_646"  = "LHCz",
                        "clust_712"  = "LHCx_1", 
                        "clust_80"   = "LHCf_I"
                        )

all_MCL_t1$lhc_annotation <- lhc_annotation_map[all_MCL_t1$cluster]

#cluster heatmap
lhc_clusters_t1 <- dplyr::filter (all_MCL_t1, grepl('light harvesting|LHC|Lhc|lhc|harvesting|clust_1044 |clust_1084 |clust_1194 |clust_1236 |clust_1787 |clust_1938 |clust_2256 |clust_2549 |clust_3219 |clust_3282 |clust_332 |clust_3363 |clust_3551 |clust_402 |clust_646 |clust_712 |clust_80 ', name))

lhc_clusters_t1$annotation <- paste(lhc_clusters_t1$lhc_annotation,'-',lhc_clusters_t1$cluster)
lhc_clusters_t1 <- lhc_clusters_t1[order(lhc_clusters_t1$annotation),]
lhc_clusters_t1 <- lhc_clusters_t1 [c(1,2,15,4:13)]

heatmap_unclustered_t1(lhc_clusters_t1, "Fragilariopsis",  "Fragilariopsis")
heatmap_unclustered_t1(lhc_clusters_t1, "Pseudo-nitzschia", "Pseudo-nitzschia")

####LHC T5####
lhc_clusters_t5 <- dplyr::filter (frag_MCL, grepl('light harvesting|LHC|Lhc|lhc|harvesting|clust_1044 |clust_1084 |clust_1194 |clust_1236 |clust_1787 |clust_1938 |clust_2256 |clust_2549 |clust_3219 |clust_3282 |clust_332 |clust_3363 |clust_3551 |clust_402 |clust_646 |clust_712 |clust_80 ', name))

lhc_clusters_t5$cluster <- as.character(lhc_clusters_t5$cluster)
lhc_clusters_t5$lhc_annotation <- lhc_annotation_map[lhc_clusters_t5$cluster]
lhc_clusters_t5$name <- paste(lhc_clusters_t5$lhc_annotation,'-',lhc_clusters_t5$cluster)
lhc_clusters_t5 <- lhc_clusters_t5[order(lhc_clusters_t5$name),]

heatmap_unclustered(lhc_clusters_t5, "Fragilariopsis", "C")

#### LHC cluster abundance #### 
lhc_clusters_abundance <- dplyr::filter (all_MCL, grepl('light harvesting|LHC|Lhc|lhc|harvesting|clust_1044 |clust_1084 |clust_1194 |clust_1236 |clust_1787 |clust_1938 |clust_2256 |clust_2549 |clust_3219 |clust_3282 |clust_332 |clust_3363 |clust_3551 |clust_402 |clust_646 |clust_712 |clust_80 ', name)) 

lhc_clusters_abundance <- lhc_clusters_abundance [c(48, 47,92 )]
lhc_clusters_abundance$cluster <- as.character(lhc_clusters_abundance$cluster)
lhc_clusters_abundance$lhc_annotation <- lhc_annotation_map[lhc_clusters_abundance$cluster]
lhc_clusters_abundance$name <- paste(lhc_clusters_abundance$lhc_annotation,'-',lhc_clusters_abundance$cluster)

#lhc_clusters_abundance <- lhc_clusters_abundance[order(lhc_clusters_abundance$name),]

#lhc_clusters_abundance <- lhc_clusters_abundance[order(lhc_clusters_abundance$total_reads),]

lhc_clusters_abundance <- lhc_clusters_abundance[with(lhc_clusters_abundance, order(name, total_reads)),]

ggplot (data = subset (lhc_clusters_abundance, Taxa.group == c( "Fragilariopsis" )), aes(x = total_reads, y = reorder(name, desc(name)), group = Taxa.group ))+
  geom_point(size = 5)+
  scale_x_log10()+
  theme_bw()+
  ylab("")+
  xlab ("Relative Abundance")+
  theme(axis.text.x=element_text(face = "bold", size = 20, color = "black"),
         axis.title.x=element_text(size=28,face="bold", color = "black"), 
         axis.text.y=element_text(face = "bold", size = 20, color = "black"),
         strip.background =element_rect(fill="white"))+
  facet_grid( Taxa.group ~.)



###T1 and T5 LHCs together####
LHC_t1_t5 <- dplyr::filter (all_MCL, grepl("light harvesting|LHC|Lhc|lhc|harvesting|clust_1044 |clust_1084 |clust_1194 |clust_1236 |clust_1787 |clust_1938 |clust_2256 |clust_2549 |clust_3219 |clust_3282 |clust_332 |clust_3363 |clust_3551 |clust_402 |clust_646 |clust_712 |clust_80 ", name))
LHC_t1_t5_v2 <- LHC_t1_t5[c(47, 48, 55:59,61,62,64,65,67, 74:91)]

names(LHC_t1_t5_v2) <- c("Taxa", "cluster", "ice-edge","T0_A", "T0_B", "T0_C", "T1_noFe_0","T1_Fe_0","T1_noFe_3","T1_Fe_3","T1_noFe_6","T1_Fe_6","A_NoFe_0.5", "B_NoFe_0.5", "C_NoFe_0.5", "A_Fe_0.5", "B_Fe_0.5", "C_Fe_0.5", "A_NoFe_3", "B_NoFe_3", "C_NoFe_3", "A_Fe_3", "B_Fe_3", "C_Fe_3","A_NoFe_6", "B_NoFe_6", "C_NoFe_6", "A_Fe_6", "B_Fe_6", "C_Fe_6" )

Frag_LHCs <- dplyr::filter(LHC_t1_t5_v2, grepl("Frag", Taxa))

Frag_LHCs$cluster <- as.character(Frag_LHCs$cluster)
Frag_LHCs$lhc_annotation <- lhc_annotation_map[Frag_LHCs$cluster]
Frag_LHCs$name <- paste(Frag_LHCs$lhc_annotation,'-',Frag_LHCs$cluster)
Frag_LHCs <- Frag_LHCs[order(Frag_LHCs$name),]
Frag_LHCs_2 <- Frag_LHCs[c(32,3:30)]

Frag_LHCs_2 <- data.frame(Frag_LHCs_2[,-1], row.names = Frag_LHCs_2 [,1])

heatmap.2 (as.matrix(Frag_LHCs_2), 
             trace = "none",
             density = "none", 
             col = bluered(100),
             breaks = seq(-2.5,2.5, length.out = 101),
          # symbreaks = FALSE,#changes the scale of the color key
             cexRow = 0.5, 
             cexCol =1,
             margins = c(8,20),
             scale = "row" ,
             dendrogram = 'none',  
             Colv = FALSE, 
             Rowv = FALSE,
             keysize = 1,
             colsep=c(1,4,6,8,10,13,16,19,22,25),
             sepcolor = "black",
             sepwidth = c(0.02, 0.02))





###T1 and T5 LHCs together_V2####
LHC_t1_t5 <- dplyr::filter (all_MCL, grepl("light harvesting|LHC|Lhc|lhc|harvesting|clust_1044 |clust_1084 |clust_1194 |clust_1236 |clust_1787 |clust_1938 |clust_2256 |clust_2549 |clust_1820 |clust_3219 |clust_3282 |clust_332 |clust_3363 |clust_3551 |clust_402 |clust_646 |clust_712 |clust_80 ", name))
LHC_t1_t5_v2 <- LHC_t1_t5[c(148,55:59,61,62,64,65,67, 74:91)]

names(LHC_t1_t5_v2) <- c("cluster", "ice-edge","T0_A", "T0_B", "T0_C", "T1_noFe_0","T1_Fe_0","T1_noFe_3","T1_Fe_3","T1_noFe_6","T1_Fe_6","A_NoFe_0.5", "B_NoFe_0.5", "C_NoFe_0.5", "A_Fe_0.5", "B_Fe_0.5", "C_Fe_0.5", "A_NoFe_3", "B_NoFe_3", "C_NoFe_3", "A_Fe_3", "B_Fe_3", "C_Fe_3","A_NoFe_6", "B_NoFe_6", "C_NoFe_6", "A_Fe_6", "B_Fe_6", "C_Fe_6" )

LHC_t1_t5_v3 <- dplyr::filter (LHC_t1_t5_v2, grepl("Fragilario", cluster))

LHC_t1_t5_v3 <- data.frame(LHC_t1_t5_v3[,-1], row.names = LHC_t1_t5_v3 [,1])

heatmap.2 (as.matrix(LHC_t1_t5_v3), 
             trace = "none",
             density = "none", 
             col = bluered(100),
             cexRow = 0.5, 
             cexCol =1,
             margins = c(8,20),
             scale = "row" ,
             dendrogram = 'none',  
             Colv = FALSE, 
             Rowv = FALSE,
             keysize = 1,
             colsep=c(1,4,6,8,10,13,16,19,22,25),
             sepcolor = "black",
             sepwidth = c(0.02, 0.02))





###############
#lhcx clusters that are mixed up. 
lhc_orfs_t5 <- merge(lhc_annotations, pseudo_frag_ORF, by = c("orf_id"))
lhcx1_bev_t5 <- dplyr::filter (lhc_orfs_t5, grepl('Lhcx6', Type))
lhcx1_bev_t5 <- lhcx1_bev_t5[c(1, 11:61)]
heatmap(lhcx1_bev_t5, "clust")


volcanoplot_interactive(lhcx1_bev_t5, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "lhcx1_Fe")

volcanoplot_interactive(lhcx1_bev_t5, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "lhcx1_temp")

volcanoplot_interactive(lhcx1_bev_t5, "int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", "pcyn_temp")

lhc_orfs_t1 <- merge(lhc_annotations, all_ORF_t1, by = c ("orf_id", "taxa" ))
lhc_orfs_t1 <- lhc_orfs_t1[c(1,2,6,12:21)]
lhcx1_bev <- dplyr::filter (lhc_orfs_t1, grepl('Lhcx6', Type))
lhcx1_bev <- lhcx1_bev [c(1,4:13)]
lhcx1_bev<- data.frame(lhcx1_bev[,-1], row.names = lhcx1_bev [,1])

heatmap.2 (as.matrix (lhcx1_bev), 
             trace = "none",
             density = "none", 
             col = bluered(100),
             cexRow = 0.5, 
             cexCol =1,
             margins = c(8,20),
             scale = "row" ,
             dendrogram = 'none',  
             Colv = FALSE, 
             Rowv = FALSE,
             keysize = 1,
             main = "LHC_X_6",
             colsep=c(1,4,6,8,10),
             sepcolor = "black",
             sepwidth = c(0.02, 0.02))



lhcx1_6 <- dplyr::filter (pseudo_frag_ORF, grepl('clust_2256 ', name))
lhcx1_6<- lhcx1_6[order(lhcx1_6$Taxa.group),]
heatmap_unclustered(lhcx1_6, "LHCx_1/6", "clust")

lhcx1_6_t1 <- dplyr::filter (all_ORF_t1, grepl('clust_2256 ', name))
lhcx1_6_t1<- lhcx1_6_t1[order(lhcx1_6_t1$taxa),]

heatmap_unclustered_t1(lhcx1_6_t1, "LHCx_1/6", "Fragilariopsis")



##LHC volcano T5
lhc_volcano_t5 <- dplyr::filter(pseudo_frag_MCL, grepl('light harvesting|LHC|Lhc|lhc|harvesting|clust_1044 |clust_1084 |clust_1194 |clust_1236 |clust_1787 |clust_1938 |clust_2256 |clust_2549 |clust_3219 |clust_3282 |clust_332 |clust_3363 |clust_3551 |clust_402 |clust_646 |clust_712 |clust_80 ', name))


volcanoplot_interactive(lhc_volcano_t5, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "pcyn_Fe")
volcanoplot_interactive(lhc_volcano_t5, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "pcyn_temp")

volcanoplot_interactive(lhc_volcano_t5, "int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", "pcyn_temp")
```

nitrogen stuff
```{r}
#nitrogen <- dplyr::filter(all_MCL, grepl('clust_411 |clust_417 |clust_991 ', name))
nitrogen <- dplyr::filter(all_MCL, grepl('clust_411 ', name))
nitrogen_1 <- nitrogen[,c(48, 47, 148,56:58, 74:91)] 
nitrogen_1 <- dplyr::filter(nitrogen_1, grepl('Frag', Taxa.group))

names(nitrogen_1) <- c("cluster", "taxa", "name", "A_initial_initial", "B_initial_initial", "C_initial_initial", "A_noFe_0", "B_noFe_0", "C_noFe_0", "A_Fe_0", "B_Fe_0", "C_Fe_0", "A_noFe_3", "B_noFe_3", "C_noFe_3", "A_Fe_3", "B_Fe_3", "C_Fe_3","A_noFe_6", "B_noFe_6", "C_noFe_6", "A_Fe_6", "B_Fe_6", "C_Fe_6")

nitrogen_2 <- nitrogen_1 %>%melt (id.vars  = c("taxa", "cluster", "name"))%>% separate(col = variable, into = c("rep", "Fe", "temperature"))

png("Alegend.png", width=30*0.75, height=23*0.75, units="cm", res=900)
ggplot(nitrogen_2, aes(x = temperature, y = value, shape = Fe))+
  geom_point(size = 8)+
  scale_x_discrete(limits=c("initial","0","3","6"), 
                   labels = c ("initial", "-0.5", "3", "6"))+
  scale_y_continuous(limits = c(0,0.075))+
  ylab (expression (paste("NR")))+
  xlab (expression ("Temperature"~(degree*C)))+
  scale_shape_manual(name="", 
                   breaks=c("initial","Fe","noFe"),
                   values=c(19,24,1))+
  theme_bw()+
  theme(axis.text.x=element_text(face = "bold", size = 40, color = "black"),
        axis.title.x=element_text(size=45,face="bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 40, color = "black"),
        axis.title.y=element_text(size=50,face="bold", color = "black"))+
  theme(panel.grid = element_blank(), legend.text = element_text(size=15))+
  guides(shape=guide_legend(override.aes = list(size=9)))+
   theme(legend.position = "top")
dev.off()

nitrogen <- dplyr::filter(all_ORF_sub, grepl('nitrate|Nitrate', name))
volcanoplot_interactive(nitrogen, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "pcyn_temp")

```



deleteafter
```{r}
isip1 <- dplyr::filter(all_MCL, grepl('clust_1814 |clust_343 |clust_1154 |clust_974 ', name))

isip1 <- dplyr::filter(all_ORF_sub, grepl('clust_1814 |clust_343 |clust_1154 |ISIP', name))


isip1 <- dplyr::filter(all_ORF_sub, grepl('clust_1814 ', name))
volcanoplot_interactive(isip1, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "pcyn_temp")

volcanoplot_interactive(isip1, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "pcyn_Fe")


# isip1$scaledtotal <- rowMeans(isip1 [,c(4:21)])
# 
# isip1$total_read <- rowSums(isip1[,c(4:21)])
# 
# ggplot (data = isip1 , aes(x = total_read, y = orf_id, size = total_read*4))+
#   geom_point(aes(size =total_read*5))+
#   #scale_x_log10()+
#   theme_bw()+
#   ylab("")+
#   xlab ("Relative Abundance")+
#   theme(axis.text.x=element_text(face = "bold", size = 20, color = "black"),
#          axis.title.x=element_text(size=28,face="bold", color = "black"), 
#          axis.text.y=element_text(face = "bold", size = 8, color = "black"),
#          strip.background =element_rect(fill="white"))

volcanoplot_interactive(isip1, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "pcyn_Fe")

petF <- dplyr::filter(all_ORF_sub, grepl('clust_1452 |contig_741377_290_673_-|contig_1334655_1094_1393_+|contig_851292_25_359_-|contig_607708_506_943_+|contig_1157730_229_528_-', name))



petF <- dplyr::filter(all_ORF_sub, grepl('contig_1330112_2528_2827_+|contig_851292_25_359_-|contig_1192188_676_975_-|contig_1205437_103_402_-|contig_1146257_3759_4057_-|contig_1007278_1488_1737_+|contig_695146_62_361_-|contig_1007745_8181_8480_|contig_1236417_28_301_-|contig_1157730_229_528_-|contig_486285_117_397_+|contig_741377_290_673_-|contig_713786_1_274_-|contig_1328545_2333_2632_+', name))

volcanoplot_interactive(isip1, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "pcyn_Fe")

```


Plastocyanin_Figure4
```{r}
##Plastocyanin###
#Volcanoplot
pcyn_volcano <- dplyr::filter(all_ORF_sub, grepl('clust_1820 ', name))

volcanoplot_interactive(pcyn_volcano, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "pcyn_Fe")
volcanoplot_interactive(pcyn_volcano, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "pcyn_temp")

#ORF heatmap
pcyn <- subset (all_ORF_t1, cluster == "clust_1820")
pcyn_frag <- subset (pcyn, taxa == "Fragilariopsis")
pcyn_frag <- pcyn_frag [c(1, 4:13)]
pcyn_frag<- data.frame(pcyn_frag[,-1], row.names = pcyn_frag [,1])
pcyn_frag <- as.matrix ((pcyn_frag))

heatmap.2 (pcyn_frag, 
             trace = "none",
             density = "none", 
             col = bluered(100),
             cexRow = 0.5, 
             cexCol =1,
             margins = c(8,20),
             scale = "row" ,
             dendrogram = 'none',  
             Colv = FALSE, 
             Rowv = FALSE,
             keysize = 1,
             colsep=c(1,4,6,8,10),
             sepcolor = "black",
             sepwidth = c(0.02, 0.02))


#Cluster heatmap
all_MCL_t1 <- all_MCL[,c(48, 47, 148,55:59,61,62,64,65,67)] #no T0
names(all_MCL_t1) <- c("cluster","taxa","name", "ice-edge","T0_A", "T0_B", "T0_C", "T1_noFe_0","T1_Fe_0","T1_noFe_3","T1_Fe_3","T1_noFe_6","T1_Fe_6")

pcyn<- dplyr::filter (all_MCL_t1, grepl('clust_1820 ', name))
pcyn_1 <- pcyn[c(3:13)]
pcyn_1 <- data.frame(pcyn_1[,-1], row.names = pcyn_1 [,1])
pcyn_1 <- as.matrix ((pcyn_1))
heatmap.2 (pcyn_1, 
             trace = "none",
             density = "none", 
             col = bluered(100),
             cexRow = 0.5, 
             cexCol =1,
             margins = c(8,20),
             scale = "row" ,
             dendrogram = 'none',  
             Colv = FALSE, 
             Rowv = FALSE,
             keysize = 1,
             colsep=c(1,4,6,8,10),
             sepcolor = "black",
             sepwidth = c(0.02, 0.02))


###PCYN_ORF abundance##
PCYN_ORF_abundance <- dplyr::filter (all_ORF, grepl('clust_1820 ', name)) 

PCYN_ORF_abundance$scaledtotal <- rowMeans(PCYN_ORF_abundance [,c(153:170)])

PCYN_ORF_abundance$total_read <- rowSums(PCYN_ORF_abundance[,c(92:128)])


PCYN_ORF_abundance <- PCYN_ORF_abundance [c(47, 91, 179, 180)]

ggplot (data = PCYN_ORF_abundance , aes(x = scaledtotal, y = orf_id), group = Taxa.group )+
  geom_point(aes(size =scaledtotal))+
  scale_x_log10()+
  theme_bw()+
  ylab("")+
  xlab ("Relative Abundance")+
  theme(axis.text.x=element_text(face = "bold", size = 20, color = "black"),
         axis.title.x=element_text(size=28,face="bold", color = "black"), 
         axis.text.y=element_text(face = "bold", size = 8, color = "black"),
         strip.background =element_rect(fill="white"))+
  facet_grid( Taxa.group ~.)


PCYN_clusters_abundance$cluster <- as.character(PCYN_clusters_abundance$cluster)
lhc_clusters_abundance$lhc_annotation <- lhc_annotation_map[lhc_clusters_abundance$cluster]
lhc_clusters_abundance$name <- paste(lhc_clusters_abundance$lhc_annotation,'-',lhc_clusters_abundance$cluster)

#lhc_clusters_abundance <- lhc_clusters_abundance[order(lhc_clusters_abundance$name),]

#lhc_clusters_abundance <- lhc_clusters_abundance[order(lhc_clusters_abundance$total_reads),]

lhc_clusters_abundance <- lhc_clusters_abundance[with(lhc_clusters_abundance, order(name, total_reads)),]





```

B12-FigureS6
```{r}
# "clust_3167" = "MetE", methionine independent synthase
# "clust_120" = "MetH", methionine dependent synthase
# "clust_3290" = "CBA1"

#clusters from erin to look at
#4490, 2147, 3511,8360,10329
#not using 3511.. one ORF in frag, nothing in Pseudo

B12_heatmap<- dplyr::filter (all_MCL, grepl("clust_3290 |clust_120 |clust_3167 |clust_4490 |clust_2147 |clust_8360 |clust_10329 |clust_19467 ", name))

B12_heatmap$cluster <- as.character(B12_heatmap$cluster)
B12_annotation_map <- c("clust_3290" = "1_CBA1_clust_3290",
                        "clust_120" = "2_MetH_clust_120",
                        "clust_3167" = "3_MetE_clust_3167",
                        "clust_19467" = "4_CobT_clust_19467",
                        "clust_3184" =  "4.2_CobC_clust_3184",
                        "clust_16330" =  "4.3_CobC_clust_16330",
                        "clust_4490" = "5_BluB_clust_4490", 
                        "clust_2147" = "6_CobN_clust_2147", 
                        "clust_8360" = "7_CobD/Cbib_clust_8360", 
                        "clust_10329" = "8_CobB/CobQ_clust_10329" )
                                                                      

B12_heatmap$B12_annotation <- B12_annotation_map[B12_heatmap$cluster]



B12_heatmap_2 <- B12_heatmap[c(47,148, 149,55:59,61,62,64,65,67:91)]

names(B12_heatmap_2) <- c("taxa", "cluster","annotation", "ice-edge","T0_A", "T0_B", "T0_C", "T1_noFe_0","T1_Fe_0","T1_noFe_3","T1_Fe_3","T1_noFe_6","T1_Fe_6", "A_B12_noFe_0.5C", "B_B12_noFe_0.5C", "A_B12_Fe_0.5C", "B_B12_Fe_0.5C", "C_B12_Fe_0.5C", "C_B12_noFe_0.5C", "A_NoFe_0.5", "B_NoFe_0.5", "C_NoFe_0.5", "A_Fe_0.5", "B_Fe_0.5", "C_Fe_0.5", "A_NoFe_3", "B_NoFe_3", "C_NoFe_3", "A_Fe_3", "B_Fe_3", "C_Fe_3","A_NoFe_6", "B_NoFe_6", "C_NoFe_6", "A_Fe_6", "B_Fe_6", "C_Fe_6" )

#this sorts out the dataframe so when the heatmap is made, the rows are organized by taxa, and then by cluster (easier to split late in powerpoint)
B12_heatmap_2<- B12_heatmap_2[
  order(B12_heatmap_2[,1], B12_heatmap_2[,3]),]

#this is because we can't have redundancy in our heatmap row names
B12_heatmap_2$annotation <- paste(B12_heatmap_2$annotation,B12_heatmap_2$taxa)

B12_heatmap_2 <- B12_heatmap_2[-c(1,2)]

B12_heatmap_3 <- data.frame(B12_heatmap_2[,-1], row.names = B12_heatmap_2[,1])

heatmap.2 (as.matrix(B12_heatmap_3), 
             trace = "none",
             density = "none", 
             col = bluered(100),
             breaks = seq(-3,3, length.out = 101),
             cexRow = 0.5, 
             cexCol =1,
             margins = c(8,20),
             scale = "row" ,
             dendrogram = 'none',  
             Colv = FALSE, 
             Rowv = FALSE,
             keysize = 1,
             colsep=c(1,4,6,8,10,13,16,19,22,25, 28, 31),
             sepcolor = "black",
             sepwidth = c(0.02, 0.02))
           
#this here is just to explore the different cobs and see what's what
B12_stuff <- dplyr::filter(all_ORF_sub, grepl('clust_4490 |clust_2147 |clust_3511 |clust_8360 |clust_10329 |clust_10011 |cobC|CobC|cobN|CobN|cobU|CobU|cobS|CobS', name))

B12_stuff <- dplyr::filter(all_ORF_sub, grepl('clust_3511 |CobO|cobO|cobP|CobP', name))

B12_stuff <- dplyr::filter(all_ORF_sub, grepl('alpha-ribazole|CobC', name))
B12_stuff <- dplyr::filter(all_ORF_sub, grepl('adenosylcobinamide-GDP|cobS|CobS|ibazoletransferase', name))

B12_stuff <- dplyr::filter(all_ORF_sub, grepl('CobA|cobA|cobO|CobO|clust_3511 |clust_1492 ', name))


B12_stuff <- dplyr::filter (all_MCL_sub, grepl("clust_3290 |clust_120 |clust_3167 |clust_4490 |clust_2147 |clust_8360 |clust_10329 |clust_19467 ", name))

volcanoplot_interactive(B12_stuff, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "B12_iron addition")

volcanoplot_interactive(B12_stuff, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "B12_temperature increase")

volcanoplot_interactive(B12_stuff, "int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", "pseudo_frag_interaction")

```



Look at overall general trends in the ORF data. 
Important note: Let's say we're looking at Fe_vs_noFe, if the fold change is positive, it means that there was more of that transcript under Fe than noFe. If the fold change is negative, it means there was more of the ORF in the noFe. From Scott: "For that scenario, it means that there was upregulation of ORF x under high iron compared to low iron.
In general, I've kept the notation as X_vx_Y meaning X - Y (ORFs). If X has higher expression than Y, DE = 1 and FC is positive."




norm factors (cell dinsity). 
I want to do a pie-chart like above, but use norm. factors instead of %
```{r}
setwd("C:/Users/Loay Jabre/Desktop/PhD/Data/FragTranscriptome/Transcriptome")
normfactor <- read.csv("norm_factors_mmetsp_fc_pn_reclassified.csv", header = TRUE)


normfactor <- normfactor [c(1, 3,4,5,21:38)]

names(normfactor) <- c("group", "initial_initial_A", "initial_initial_B", "initial_initial_C", "noFe_0 _A","noFe_0_B", "noFe_0_C","Fe_0_A","Fe_0_B", "Fe_0_C", "noFe_3_A","noFe_3_B", "noFe_3_C","Fe_3_A","Fe_3_B", "Fe_3_C", "noFe_6_A","noFe_6_B", "noFe_6_C","Fe_6_A","Fe_6_B", "Fe_6_C")

x <- normfactor [c(2:22)]

normfactor$total <- rowSums(x)

normfactor <- arrange(normfactor, total)

normfactor_sub <- melt(normfactor, id.vars = c("group")) %>% separate(col = variable, into = c("Fe", "temperature", "rep"))

 
normfactor_sub <- na.omit(normfactor_sub)


normfactor_pseudo_frag <- dplyr::filter(normfactor_sub, grepl('Fragilariopsis|Pseudo-nitzschia', group))


#png(filename="nucleartranscriptabundance_all.png",width=60*0.75, height=30*0.75, units="cm", res = 900)

ggplot(normfactor_sub, aes(x=temperature,
                      y=value, 
                      shape = Fe))+
  geom_point(size=5)+
  scale_x_discrete(limits=c("initial", "0","3","6"))+
  scale_y_log10()+
  theme(plot.title = element_text(hjust = 0.5))+
  theme(axis.text.x = element_text( vjust = 1.0, hjust = 1.0))+
  ylab (expression (Nuclear~transcript~abundance))+
  xlab (expression (Temperature ~(degree*C)))+
  theme_bw()+
  scale_shape_manual(name="",
                     labels=c("Fe","noFe","initial"),
                     values = c(19,1,1))+
  theme(legend.text = element_text(size=18))+
  theme(axis.text.x=element_text(face = "bold", size = 20, color = "black"),
         axis.title.x=element_text(size=28,face="bold", color = "black"), 
         axis.text.y=element_text(face = "bold", size = 20, color = "black"),
         axis.title.y=element_text(size=28,face="bold", color = "black"), 
        strip.text = element_text(size=18, face = "italic"),
        strip.background =element_rect(fill="white"))+
facet_wrap(~factor(normfactor_sub$group, levels = unique(normfactor_sub$group)))

#dev.off()






normfactor <- dplyr::filter (normfactor,!grepl('Fragilariopsis|Pseudo-nitzschia|Dinophyta|Ciliophora', group))


normfactor$sum<- rowSums(normfactor[,c(20:22)])


x <- dplyr::filter (normfactor,!grepl('Fragilariopsis|Pseudo-nitzschia|Dinophyta|Ciliophora', group))

#extract the value of the sum for Pnitz

a<- normfactor$sum[normfactor$group == "Fragilariopsis"]
b<- normfactor$sum[normfactor$group == "Pseudo-nitzschia"]
c<- normfactor$sum[normfactor$group == "Dinophyta"]
d<- normfactor$sum[normfactor$group == "Ciliophora"]

e<- sum(x$sum)





slices <- c(a,b,c,d,e)
lbls <- c("Fragilariopsis", "Pseudo-nitzschia", "Dinophyta", "Ciliophora", "Other")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 


pie(slices,labels = lbls, col=gray.colors(length(lbls)),
 main="")


#Stats 
#to see if effects of temperature and Fe on nuclear thing are significant or not. only significant effect is temperature on Pseudo-nitz

normfactor_stat <- dplyr::filter(normfactor_sub, !grepl('initial', Fe))

normfactor_frag <- dplyr::filter(normfactor_stat, grepl('Frag', group))
normfactor_pseudo <- dplyr::filter(normfactor_stat, grepl('Pseudo-nitzschia', group))
#test anova assumptions: 

#using the ggpubr package, we check for normality:
ggdensity(normfactor_frag$value)
ggdensity(normfactor_pseudo$value)
#this shows whether the data are normal/bellshaped. in this case they are for frag, not pseudo

ggqqplot(normfactor_frag$value) 
#this shows whether data follow reference line.
ggqqplot(normfactor_pseudo$value)

shapiro.test(normfactor_pseudo$value)#not normal distributed
shapiro.test(normfactor_frag$value) #normal distributed
#the output for the shapiro test shows that the chl-a data are not normally distributed (p value is <0.05, which means data violate normality)


#an alternative to anova , a non-parametric test would be the kruskal test
kruskal.test(value~temperature, normfactor_pseudo)
kruskal.test(value~Fe, normfactor_pseudo)

frag.aov<- aov(value~temperature+Fe, normfactor_frag)
summary(frag.aov)
```


Nuclear transcripts vs cellcounts
```{r}
#need to have cellcount data loaded up
cellcounts <- read.csv("cellcounts.csv")
cellcounts_sub <- cellcounts [, cellcounts [1, ] == "1000"]
cellcounts_sub$taxa <- cellcounts$X
cellcounts_sub <- cellcounts_sub [-c(1),]
cellcounts_sub<- cellcounts_sub [c(37, 1:6, 10:15, 19:24, 28:30)]
cellcounts_sub <- dplyr::filter(cellcounts_sub, !grepl('Actinula|Amphiprora|Dino cyst|Gastro|colony|single|unkn.|other|subcurv|colonial|Rotifer|Asterom|Corethron|Cylindroth|Eucamp', taxa))
cellcounts_sub$taxa <- gsub("Pseudo-nitzschia total", "Pseudo-nitzschia", cellcounts_sub$taxa)
cellcounts_sub$taxa <- gsub("Ciliates", "Ciliophora", cellcounts_sub$taxa)
cellcounts_sub$taxa <- gsub("Dinoflagellate", "Dinophyta", cellcounts_sub$taxa)

cellcounts_sub$taxa <- gsub("Fragilariopsis ", "Fragilariopsis", cellcounts_sub$taxa)


names(cellcounts_sub) <- c("group","A_initial_initial", "B_initial_initial", "C_initial_initial", "A_noFe_0", "B_noFe_0", "C_noFe_0", "A_Fe_0", "B_Fe_0", "C_Fe_0", "A_noFe_3", "B_noFe_3", "C_noFe_3", "A_Fe_3", "B_Fe_3", "C_Fe_3","A_noFe_6", "B_noFe_6", "C_noFe_6", "A_Fe_6", "B_Fe_6", "C_Fe_6")

cellcount_pseudo_frag <- dplyr::filter(cellcounts_sub,grepl('Fragilariopsis|Pseudo-nitzschia|Dino|Cili', group)) %>%melt (id.vars  = "group")%>% separate(col = variable, into = c("rep", "Fe", "temperature"))



nuclear_vs_cellcounts <- merge (cellcount_pseudo_frag, normfactor_sub, by = c("group", "Fe", "temperature", "rep"))



ggplot(nuclear_vs_cellcounts, aes(x=value.x,
                      y=value.y, 
                      shape = Fe, 
                      color = temperature))+
  geom_point(size =3)+
  facet_wrap(~group)+
  theme(plot.title = element_text(hjust = 0.5))+
  theme(axis.text.x = element_text( vjust = 1.0, hjust = 1.0))+
  ylab (expression (Nuclear~transcript~abundance))+
  xlab (expression (Cell~count))+
  theme_bw()+
  scale_shape_manual(name="", labels=c("Fe","noFe","initial"), values = c(19,1,1))+
  theme(axis.text.x=element_text(face = "bold", size = 20, color = "black"),
        axis.title.x=element_text(size=22,face="bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 20, color = "black"),
        axis.title.y=element_text(size=28,face="bold", color = "black"), strip.text = element_text(size=12, face = "italic"), strip.background =element_rect(fill="white"))+
  theme(panel.grid = element_blank(), legend.text = element_text(size=15))

```


Ven diagram to look at numbers of DE transcripts in diatoms
```{r}
#functions from the venndiagram package to make diagrams
## note: this only looks at the clusters that are found across all three groups. i.e. if cluster x is only found in frag and pseudonitzchia, it is not plotted here.

#Iron effect
ven_alldiatoms <-all_MCL [,c(47,48, 141, 142)]
ven_alldiatoms <- dplyr::filter (ven_alldiatoms, grepl('Pseudo-nit|Fragilariopsis',Taxa.group))

ven_alldiatoms$DE <- 0
ven_alldiatoms [which(ven_alldiatoms['fe_neg_de'] == -1), "DE"]<- -1
ven_alldiatoms [which(ven_alldiatoms['fe_pos_de'] == 1), "DE"]<- 1
ven_alldiatoms <- ven_alldiatoms [,c(1,2,5)]
ven_alldiatoms_sub <- dcast(ven_alldiatoms, cluster ~ Taxa.group, value.var = "DE")
ven_alldiatoms_sub <- data.frame(ven_alldiatoms_sub[,-1], row.names = ven_alldiatoms_sub [,1])
ven_alldiatoms_sub [is.na(ven_alldiatoms_sub)] <- 0


png(filename="Fe_Ven_v2.png",width=38*0.75, height=30*0.75, units="cm", res = 900)
vennDiagram(ven_alldiatoms_sub, 
            include = c ("up", "down"),
            names = c( "Fragilariopsis", "Pseudo-nitzschia"),
            cex = 2.5, 
            circle.col=c("black", "red"),
            counts.col=c("black", "blue", "black"), 
           main = "Iron")
dev.off()
#......................
#Temperature effect
ven_alldiatoms <-all_MCL [,c(47,48, 143, 144)]
ven_alldiatoms <- dplyr::filter (ven_alldiatoms, grepl('Pseudo-nit|Fragilariopsis',Taxa.group))

ven_alldiatoms$DE <- 0

ven_alldiatoms [which(ven_alldiatoms['temp_neg_de'] == -1), "DE"]<- -1
ven_alldiatoms [which(ven_alldiatoms['temp_pos_de'] == 1), "DE"]<- 1
ven_alldiatoms <- ven_alldiatoms [,c(1,2,5)]
ven_alldiatoms_sub <- dcast(ven_alldiatoms, cluster ~ Taxa.group, value.var = "DE")
ven_alldiatoms_sub <- data.frame(ven_alldiatoms_sub[,-1], row.names = ven_alldiatoms_sub [,1])
ven_alldiatoms_sub [is.na(ven_alldiatoms_sub)] <- 0


#countFe <- vennCounts(ven_alldiatoms_sub)

#png(filename="Temp_Ven.png",width=38*0.75, height=30*0.75, units="cm", res = 900)
vennDiagram(ven_alldiatoms_sub, 
            include = c ("up", "down"),
            names = c( "Fragilariopsis", "Pseudo-nitzschia"),
            cex = 2.3, 
            counts.col=c("black", "darkgray", "black"),
            circle.col=c("black", "red"),
            main = "Temp")
#dev.off()
           #circle.col=c("black", "red", "blue"))
.................................
#IronxTemperature effect
ven_alldiatoms <-all_MCL [,c(47,48, 145, 146)]
ven_alldiatoms <- dplyr::filter (ven_alldiatoms, grepl('Pseudo-nit|Fragilariopsis',Taxa.group))

ven_alldiatoms$DE <- 0

ven_alldiatoms [which(ven_alldiatoms['tempfe_neg_de'] == -1), "DE"]<- -1
ven_alldiatoms [which(ven_alldiatoms['tempfe_pos_de'] == 1), "DE"]<- 1
ven_alldiatoms <- ven_alldiatoms [,c(1,2,5)]
ven_alldiatoms_sub <- dcast(ven_alldiatoms, cluster ~ Taxa.group, value.var = "DE")
ven_alldiatoms_sub <- data.frame(ven_alldiatoms_sub[,-1], row.names = ven_alldiatoms_sub [,1])
ven_alldiatoms_sub [is.na(ven_alldiatoms_sub)] <- 0

#countFe <- vennCounts(ven_alldiatoms_sub)

#png(filename="TempxFe_Ven.png",width=38*0.75, height=30*0.75, units="cm", res = 900)
vennDiagram(ven_alldiatoms_sub, 
            include = c ("up", "down"),
            names = c( "Fragilariopsis", "Pseudo-nitzschia"),
            cex = 2.3, 
            counts.col=c("black", "darkgray", "black"),
            circle.col=c("black", "red"),
           main = "Iron temp")
#dev.off()
           #circle.col=c("black", "red", "blue"))

```



Sliding plot showing number of D.E ORFS in each of the various taxa (diatoms and more) under Fe, Temp and FexTemp 
```{r}
# DE plot with positive and negative differences shown.
# blank plot just used for the axis.
all_ORF$Taxa.group <- gsub("Diatom", "Other Diatoms", all_ORF$Taxa.group)

blank_p <- all_ORF %>% 
  group_by(Taxa.group) %>% 
  summarise(fe_grp = sum(abs_fe)) %>%  
  ggplot(aes(x = fe_grp, 
             y = fct_reorder(Taxa.group, fe_grp))) + 
  # geom_point() +
  theme_bw() +
  xlab('') + 
  theme(axis.text.x = element_text(colour = 'white'), 
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        axis.text.y = element_text(face = "bold", size = 10, color = "black"),
        axis.ticks = element_blank()) +
  ylab(''); blank_p




pn_fe <- all_ORF %>% 
  group_by(Taxa.group) %>% 
  summarise(fe_grp = sum(abs_fe),
            fe_grp_neg = sum(fe_neg_de),
            fe_grp_pos = sum(fe_pos_de)) %>%  
  ggplot(aes(x = fe_grp_neg, y = fct_reorder(Taxa.group, fe_grp))) + 
  geom_point(size = 3, alpha = 0.5, colour = 'blue') +
  geom_point(aes(x = fe_grp_pos, y = fct_reorder(Taxa.group, fe_grp)),
             size = 3, alpha = 0.5, colour = 'red') +
  geom_segment(aes(x = fe_grp_neg, xend = 0, 
                   y = fct_reorder(Taxa.group, fe_grp), 
                   yend = fct_reorder(Taxa.group, fe_grp)), colour = 'blue', alpha = 0.2, lwd = 2) +
  geom_segment(aes(x = fe_grp_pos, xend = 0, 
                   y = fct_reorder(Taxa.group, fe_grp), 
                   yend = fct_reorder(Taxa.group, fe_grp)), colour = 'red', alpha = 0.2, lwd = 2) +
  theme_bw() +
  xlab('Iron-Related DE') + 
  ylab('') +
  theme(axis.text.y = element_blank(),  axis.text.x = element_text(face = "bold", size = 10, color = "black"), 
        axis.title.x=element_text(size=13,face="bold", color = "black"));pn_fe


pn_temp <- all_ORF %>% 
  group_by(Taxa.group) %>% 
  summarise(fe_grp = sum(abs_fe),
            temp_grp_neg = sum(temp_neg_de),
            temp_grp_pos = sum(temp_pos_de)) %>%  
  ggplot(aes(x = temp_grp_neg, y = fct_reorder(Taxa.group, fe_grp))) + 
  geom_point(size = 3, alpha = 0.5, colour = 'blue') +
  geom_point(aes(x = temp_grp_pos, y = fct_reorder(Taxa.group, fe_grp)),
             size = 3, alpha = 0.5, colour = 'red') +
  geom_segment(aes(x = temp_grp_neg, xend = 0, 
                   y = fct_reorder(Taxa.group, fe_grp), 
                   yend = fct_reorder(Taxa.group, fe_grp)), colour = 'blue', alpha = 0.2, lwd = 2) +
  geom_segment(aes(x = temp_grp_pos, xend = 0, 
                   y = fct_reorder(Taxa.group, fe_grp), 
                   yend = fct_reorder(Taxa.group, fe_grp)), colour = 'red', alpha = 0.2, lwd = 2) +
  theme_bw() +
  xlab('Temperature-Related DE') + 
  ylab('') +
  theme(axis.text.y = element_blank(),  axis.text.x = element_text(face = "bold", size = 10, color = "black"), 
        axis.title.x=element_text(size=13,face="bold", color = "black"));pn_temp

pn_tempfe <- all_ORF %>% 
  group_by(Taxa.group) %>% 
  summarise(fe_grp = sum(abs_fe),
            fetemp_grp_neg = sum(tempfe_neg_de),
            fetemp_grp_pos = sum(tempfe_pos_de)) %>%  
  ggplot(aes(x = fetemp_grp_neg, y = fct_reorder(Taxa.group, fe_grp))) + 
  geom_point(size = 3, alpha = 0.5, colour = 'blue') +
  geom_point(aes(x = fetemp_grp_pos, y = fct_reorder(Taxa.group, fe_grp)),
             size = 3, alpha = 0.5, colour = 'red') +
  geom_segment(aes(x = fetemp_grp_neg, xend = 0, 
                   y = fct_reorder(Taxa.group, fe_grp), 
                   yend = fct_reorder(Taxa.group, fe_grp)), colour = 'blue', alpha = 0.2, lwd = 2) +
  geom_segment(aes(x = fetemp_grp_pos, xend = 0, 
                   y = fct_reorder(Taxa.group, fe_grp), 
                   yend = fct_reorder(Taxa.group, fe_grp)), colour = 'red', alpha = 0.2, lwd = 2) +
  theme_bw() +
  xlab('Iron-Temperature-Related DE') + 
  ylab('') +
  theme(axis.text.y = element_blank(),  axis.text.x = element_text(face = "bold", size = 10, color = "black"), 
        axis.title.x=element_text(size=13,face="bold", color = "black"));pn_tempfe



g <-
  grid.arrange(blank_p, pn_fe, pn_temp, pn_tempfe, nrow = 1)
 ggsave(file="slidingplot.png",width=37*0.75, height=23*0.75, units="cm", g)

```




Group plots/fig2
```{r}
clustername_vs_foldchange_group <- function(dataset, foldchange, FDR, graphtitle){
  volcano <- dataset[c("name",foldchange, FDR,"Taxa.group", "clustercompartment" )]
  x <- foldchange
  
  names(volcano) <- c( "gene", "Fold", "FDR", "taxa", "clustercompartment")
  volcano ["group"] <- "no-fold-change"
  
  if(grepl("Fe_vs_noFe_foldchange", x)) {
    volcano['size'] <- rowMeans( dataset[, c(4:21)])
  } else if (grepl("temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("temp_6C_vs_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:21)])
  } else if (grepl("int_fe_temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("int_fe_temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("noFe_3C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6,10:12 )])
  } else if (grepl("noFe_6C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6, 16:18)])
  } else if (grepl("Fe_3C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(7:9, 13:15)])
  } else if (grepl("Fe_6C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans(dataset[, c(7:9, 19:21)])
  } else if (grepl("noFe_6C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:18,10:12)])
  } else if (grepl("Fe_6C_vs_Fe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(13:15, 19:21)])
  } else if (grepl("Fe_0C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9)])
  } else if (grepl("Fe_3C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:15)])
  } else if (grepl("Fe_6C_vs_noFe_6C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:21)])
  }
volcano$shapez <-  ifelse (volcano$FDR < 0.05, "1", "16") 

  ggplot (data = volcano, aes(x= Fold, y= clustercompartment, color = taxa, label=gene, shape = shapez)) +
    geom_point (alpha=0.7, size =6)+
       scale_shape_manual (values = c(19, 1), guide = FALSE)+
       ylab("")+ 
    xlab(expression(Log[2]~~Fold-change))+
    scale_color_manual(values = c("black", "red"))+
    scale_y_discrete(limits = compartments)+
    ggtitle(graphtitle)+
    theme_bw()+
    theme(plot.title = element_text(hjust = 0.5))+
    theme(axis.text.x=element_text(face = "bold", size = 20, color = "black"),
        axis.title.x=element_text(size=22,face="bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 21, color = "black"),
        axis.title.y=element_text(size=20,face="bold", color = "black"))+
 theme(legend.text = element_text(size=12))+
     
    # theme(axis.text.x = element_text(
    #         hjust=1,
    #   size = 14, 
    #   color = "black",
    #   face = "bold")) +
    # theme(axis.text.y = element_text(
    #         hjust=1,
    #   size = 11, 
    #   face = "bold")) +
    geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )
}



cluster_dataset <- dplyr::filter(pseudo_frag_MCL, grepl("clust_332 |clust_1236 |clust_80 |clust_402 |clust_1044 |clust_3282 |clust_1194 |clust_3219 |clust_1084 |clust_3551 |clust_646  |clust_1938 |clust_2549 |clust_3551 |clust_1787 |clust_712 |clust_2256 |clust_3363 |clust_1414 |clust_498 |clust_211 |clust_42 |clust_24191 |clust_55 |clust_2763 |clust_1906 |clust_15862 |clust_5856 |clust_27282 |clust_5235 |clust_14895 |clust_86124 |clust_12649 |clust_26101 |clust_231 |clust_1548 |clust_1820 |clust_2212 |clust_1822 |clust_1668 |clust_1196 |clust_1452 |clust_534 |clust_4660 |clust_808 |clust_349 |clust_78 |clust_306 |clust_1846 |clust_129 |clust_816 |clust_3167 |clust_120 |clust_3290 |clust_9740 |clust_343 |clust_1814 |clust_1154 |clust_991 |clust_411 |clust_1816 |clust_2277 |clust_589 |clust_284 |clust_891 |clust_15657 |clust_4956 |clust_6089 |clust_7529 |clust_2658 |clust_557 |clust_3360 |clust_35 |clust_3360 |clust_15652 |clust_10606 |clust_1 |clust_550 |clust_1 |clust_4298 |clust_4298 |clust_823 |clust_6713 |clust_444 |clust_22 |clust_219 |clust_4251 |clust_5230 |clust_7451 |clust_417 ", name))


cluster_dataset$cluster <- as.character(cluster_dataset$cluster)
p680 <- "P680"
mnclus <- "Manganese Cluster"
cytb6f <- "Cytochrome b6f Complex"
atpsyn <- "ATP Synthase"
rib <- "Ribosome"


compartments <- c('Ribosome', '' ,'CBA1', 'MetH','MetE', '','Ferritin', 'ISIPs', '', 'GOGAT', 'Nitrite Reductase','Nitrite Transporter', 'Nitrate Reductase', 'Nitrate Transporter','Ammonium Transporter', '','Rhodopsin_1','Rhodopsin_2', 'RuBisCO', 'ATP Synthase','FNR', 'Flavodoxin','Ferredoxin', 'P700', 'Plastocyanin','Cytochrome c6',  'Cytochrome b6f Complex', "PSII",'' , 'LHCs')




categorymap_cluster <- c("clust_417" = "Ammonium Transporter", "clust_332" = "LHCs", "clust_1236" = "LHCs", "clust_80" = "LHCs", "clust_402" = "LHCs", "clust_1044" = "LHCs", "clust_3282" = "LHCs", "clust_1194" = "LHCs", "clust_3219" = "LHCs", "clust_1084"="LHCs", "clust_3551"="LHCs", "clust_646"="LHCs",  "clust_1938"="LHCs", "clust_2549" = "LHCs", "clust_3551" = "LHCs", "clust_1787" ="LHCs", "clust_712"="LHCs",  "clust_2256"="LHCs", "clust_3363"="LHCs", "clust_1414" = "Cytochrome c6", "clust_498" = "PSII", "clust_211" = "PSII", "clust_42" = "PSII", "clust_24191" = "PSII", "clust_55" = "P700", "clust_2763" = "PSII",  "clust_1906"="PSII",  "clust_15862" = "PSII",  "clust_5856"="PSII",   "clust_27282" ="PSII",  "clust_5235"="PSII",   "clust_14895"="PSII", "clust_86124" = "PSII",  "clust_12649" ="PSII",  "clust_26101" = "PSII", "clust_231" = "Rhodopsin_1", "clust_1548" = "Rhodopsin_2", "clust_1820" = "Plastocyanin", "clust_2212" = cytb6f, "clust_1822" = cytb6f, "clust_1668" = cytb6f, "clust_1196" =cytb6f, "clust_1452" = "Ferredoxin", "clust_534" = "Flavodoxin", "clust_4660" = "Flavodoxin", "clust_808" = "FNR", "clust_78" = atpsyn, "clust_349" = atpsyn, "clust_306" = atpsyn, "clust_1846" = atpsyn, "clust_129" = "RuBisCO", "clust_816" = "RuBisCO", "clust_3167" = "MetE", "clust_120" = "MetH", "clust_3290" = "CBA1", "clust_9740" = "ISIPs", "clust_343" = "ISIPs", "clust_1814" = "ISIPs", "clust_1154" = "ISIPs", "clust_991" = "Nitrate Transporter", "clust_411" = "Nitrate Reductase", "clust_1816" = "Nitrite Reductase", "clust_2277" = "Nitrite Transporter", "clust_589" = "GOGAT", "clust_284" = "GOGAT", "clust_891" = "GOGAT", "clust_15657" = "Ferritin","clust_4956" = "Nitric Oxide Dioxygenase", 
#ribosomes:
"clust_6089" = rib, "clust_7529" = rib,  "clust_2658" = rib, "clust_557" = rib, "clust_3360" = rib, "clust_35"=rib, "clust_3360" = rib, "clust_15652"=rib,  "clust_10606" = rib, "clust_1"=rib, "clust_550" =rib, "clust_1"=rib, "clust_4298" = rib,  "clust_4298" = rib,  "clust_823" = rib, "clust_6713" = rib,  "clust_444"=rib,  "clust_22"=rib,  "clust_219" = rib, "clust_4251"=rib ,"clust_5230" = rib,  "clust_7451" = rib )


cluster_dataset$clustercompartment <- categorymap_cluster[cluster_dataset$cluster]


#png("clusternamevsfolchage_Fe.png", width=45*0.75, height=33*0.75, units="cm", res=900)
#clustername_vs_foldchange_group(cluster_dataset,"Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "Fe_vs_noFe")
#dev.off()

#png("clusternamevsfolchage_6Cvs0C.png", width=45*0.75, height=33*0.75, units="cm", res=900)
#clustername_vs_foldchange_group(cluster_dataset, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "hightemp_vs_lowtemp")
#dev.off()

clustername_vs_foldchange_group(cluster_dataset, "noFe_6C_vs_noFe_0C_foldchange", "noFe_6C_vs_noFe_0C_pvalue", "hightemp_vs_lowtemp")
ggplotly

#png("clusternamevsfolchage_6Cvs0C_interaction.png", width=45*0.75, height=33*0.75, units="cm", res=900)
clustername_vs_foldchange_group(cluster_dataset, "int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", "6Cvs0C_interaction")
#dev.off()
```


Interaction
```{r}
interaction_clusters <- pseudo_frag_MCL[c(1:21, 30,31)]

interaction_clusters_sub <- subset(interaction_clusters,(interaction_clusters$int_fe_temp_6C_vs_0C_pvalue<0.05 ))



commonclusters <- dplyr::filter(interaction_clusters_sub, grepl('clust_1236 |clust_220 |clust_2212 |clust_3251 |clust_332 |clust_3596 |clust_5679 ', name))



onlypseudo <- interaction_clusters_sub [ which (interaction_clusters_sub$Taxa.group == "Pseudo-nitzschia"),]
  
downregulated_pseudo <- dplyr::filter(interaction_clusters_sub, grepl('clust_2398|clust_137411|clust_17298|clust_220275|clust_78042', name))


volcanoplot_interactive(interaction_clusters_sub, "int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", "pseudo_frag_interaction")


whatisthatcluster<- dplyr::filter(pseudo_frag_ORF, grepl('clust_9740 |clust_343 ', name))

#1236 =LHCf
#220 = H2B , histone 
#2212 = Rieski-like protein, 
#3251= PSII RC, PsbM
#332 = LHCf2, chl-a-b binding
#3596 = hypothetical
#5679 = Hypothetical

#downreg in pseudo
#137411 - 1 ORF... nothing
#17298 = leucine rich repeart, gap
#220275 = TMAO-torS, TMAO-reductase    - 1 ORF, 

```



cluster name vs fold change
```{r}
photosynthesis <- dplyr::filter (pseudo_frag_MCL, grepl('rhodopsin|Rhodopsin|P680|OEC|oec|evolving|OEE|PsbO|PsbP|PsbQ|p700|P700|flavodoxin|Flavodoxin|b6f|Rieske|petF_2|petF_1|atpA|atpB|RUBISCO|Rubisco|rubisco|plastocyanin|Plastocyanin|CP47|LHC|Lhc|clust_808 ', name))

volcanoplot(photosynthesis,"temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_pseudo-frag_ORFs")

volcanoplot(photosynthesis,"Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_pseudo-frag_ORFs")

volcanoplot(photosynthesis,"int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", " tempfe_interaction_pseudo-frag_ORFs")
```


sliding plot for the different cellular functions
```{r}
frag_slide<- merge(frag_ORF, all_ORF, by = 'orf_id')
frag_slide<- frag_slide [c(222:229)]
colnames (frag_slide) <- c("fe_neg", "fe_pos", "temp_neg", "temp_pos", "tempfe_neg", "tempfe_pos", "abs_fe", "name")

pseudo_slide <- merge(pseudo_ORF, all_ORF, by = 'orf_id')
pseudo_slide<- pseudo_slide [c(222:229)]
colnames (pseudo_slide) <- c("fe_neg", "fe_pos", "temp_neg", "temp_pos", "tempfe_neg", "tempfe_pos", "abs_fe", "name")

#pick the cell compartments to plot, and give that whole compartment a name
cellcompartment <- function (dataset){
  
Rhodopsin <- dplyr::filter(dataset, grepl("rhodopsin|Rhodopsin", name))
Rhodopsin$cellcompartment <- "Rhodopsin"

PSIIreactioncenter <- dplyr::filter(dataset, grepl("P680", name))
PSIIreactioncenter$cellcompartment <- "P680"

manganesecluster <- dplyr::filter(dataset, grepl("OEC|oec|evolving|OEE|PsbO|PsbP|PsbQ", name))
manganesecluster$cellcompartment <- "Mn-cluster"

PSIreactioncenter <- dplyr::filter(dataset, grepl("p700|P700", name))
PSIreactioncenter$cellcompartment <- "P700"

Flavodoxin <- dplyr::filter(dataset, grepl("flavodoxin|Flavodoxin", name))
Flavodoxin$cellcompartment <- "Flavodoxin"

Ferredoxin <- dplyr::filter(dataset, grepl("petF_1", name))
Ferredoxin$cellcompartment <- "Ferredoxin"

Cyt_B6f <- dplyr::filter(dataset, grepl("b6f|Rieske|petF_2", name))
Cyt_B6f$cellcompartment <- "Cyt_B6f"

ATPsynthase <- dplyr::filter(dataset, grepl("atpA|atpB", name))
ATPsynthase$cellcompartment <- "ATP_synthase"

Rubisco <- dplyr::filter(dataset, grepl("RUBISCO|Rubisco|rubisco", name))
Rubisco$cellcompartment <- "Rubisco"

Plastocyanin<- dplyr::filter(dataset, grepl("plastocyanin|Plastocyanin", name))
Plastocyanin$cellcompartment <- "Plastocyanin"

Ribosome <- dplyr::filter(dataset, grepl("ribosome|ribosom|Ribosom", name))
Ribosome$cellcompartment <- "Ribosome"

All_SODs <- dplyr::filter(dataset, grepl("SOD|superoxide", name))
All_SODs$cellcompartment <- "All_SODs"

Nitrate_reductase <- dplyr::filter(dataset, grepl("clust_411 ", name))
Nitrate_reductase$cellcompartment <- "Nitrate_reductase"

Nitrate_transporter <- dplyr::filter(dataset, grepl("nitrate transporter |clust", name))
Nitrate_transporter$cellcompartment <- "Nitrate_transporter"

Nitrite_reductase <- dplyr::filter(dataset, grepl("clust_1816 |clust_3042 |clust_3563 ", name))
Nitrite_reductase$cellcompartment <- "Nitrite_reductase"

Nitrite_transporter <- dplyr::filter(dataset, grepl("nitrite transporter", name))
Nitrite_transporter$cellcompartment <- "Nitrite_transporter"

CP47<- dplyr::filter(dataset, grepl("CP47", name))
CP47$cellcompartment <- "CP47"

ISIP1 <- dplyr::filter(dataset, grepl("ISIP1", name))
ISIP1$cellcompartment <- "ISIP1"

ISIP2A <- dplyr::filter(dataset, grepl("ISIP2A", name))
ISIP2A$cellcompartment <- "ISIP2A"

ISIP3 <- dplyr::filter(dataset, grepl("ISIP3", name))
ISIP3$cellcompartment <- "ISIP3"

Glutamine_synthetase <-  dplyr::filter(dataset, grepl("glutamine synthetase ", name))
Glutamine_synthetase$cellcompartment <- "Glutamine_synthetase"

Glutamate_synthase <-  dplyr::filter(dataset, grepl("glutamate synthase ", name))
Glutamate_synthase$cellcompartment <- "Glutamate_synthase"

Hydroperoxide_reductase <- dplyr::filter(dataset, grepl("hydroperoxide", name))
Hydroperoxide_reductase$cellcompartment <- "Hydroperoxide_reductase"

Hsp70 <-  dplyr::filter(dataset, grepl("Hsp70", name))
Hsp70 $cellcompartment <- "Hsp70"

Hsp90 <-  dplyr::filter(dataset, grepl("Hsp90|HSP90", name))
Hsp90 $cellcompartment <- "Hsp90"

Hsp20 <-  dplyr::filter(dataset, grepl("Hsp20", name))
Hsp20 $cellcompartment <- "Hsp20"

#Hsp90like_ATPase <-  dplyr::filter(dataset, grepl("HSP90-like", name))
#Hsp90like_ATPase$cellcompartment <- "Hsp90like_ATPase"

coldshock_protein<-  dplyr::filter(dataset, grepl("cold", name))
coldshock_protein$cellcompartment <- "coldshock_protein"

Lhcx<-  dplyr::filter(dataset, grepl("Lhcx", name))
Lhcx$cellcompartment <- "Lhcx"

Lhcf<-  dplyr::filter(dataset, grepl("Lhcf", name))
Lhcf$cellcompartment <- "Lhcf"

Lhcr<-  dplyr::filter(dataset, grepl("Lhcr", name))
Lhcr$cellcompartment <- "Lhcr"

Lhca<-  dplyr::filter(dataset, grepl("Lhca", name))
Lhca$cellcompartment <- "Lhca"

FNR <- dplyr::filter(dataset, grepl("clust_808 ", name))
FNR$cellcompartment <- "FNR"


Urease <- dplyr::filter(dataset, grepl("urease|Urease", name))
Urease$cellcompartment <- "Urease"


MetE <- dplyr::filter(dataset, grepl("Cobalamin-independent|MetE", name)) 
MetE $cellcompartment <- "MetE"

CBA1 <- dplyr::filter(dataset, grepl("CBA1", name)) 
CBA1 $cellcompartment <- "CBA1"


MetH <- dplyr::filter(dataset, grepl("METH", name)) 
MetH $cellcompartment <- "MetH"

Ferritin <-  dplyr::filter(dataset, grepl("Ferritin|ferritin", name))
Ferritin $cellcompartment <- "Ferritin"

NO_dioxygenase <-  dplyr::filter(dataset, grepl("clust_4956  ", name))
NO_dioxygenase $cellcompartment <- "NO_dioxygenase"

Ferric_reductase <-  dplyr::filter(dataset, grepl("ferric reductase  |Ferric", name))
Ferric_reductase $cellcompartment <- "Ferric_reductase"

Cyt_c6 <-  dplyr::filter(dataset, grepl("clust_1414 ", name))
Cyt_c6 $cellcompartment <- "Cyt_c6"

#NO_synthase <-  dplyr::filter(dataset, grepl("clust_16690 |nitric oxide synthase  ", name))
#NO_synthase $cellcompartment <- "NO_synthase" 

#ATPase <- dplyr::filter(dataset, grepl("ATPase", name)) 
#ATPase $cellcompartment <- "ATPase"

cell_compartment <- rbind (Cyt_c6, Ferric_reductase, Ferritin, CBA1, Ferredoxin, MetE, ISIP1, MetH, Urease, FNR, Lhca, Lhcf, Lhcx, Lhcr, coldshock_protein,  Hsp70, Hsp90, Hsp20, Hydroperoxide_reductase, ISIP2A,ISIP3, Nitrate_transporter, Nitrite_transporter, Glutamine_synthetase, Glutamate_synthase, CP47, Nitrite_reductase, Nitrate_reductase, Ribosome, All_SODs,NO_dioxygenase, Rhodopsin,PSIIreactioncenter,Cyt_B6f, manganesecluster, PSIreactioncenter,Flavodoxin, ATPsynthase, Rubisco, Plastocyanin)

#this returns dataframe and puts it into workspace. 
cell_compartment <<- cell_compartment

}



#compartments <- c('Ribosome', '', 'coldshock_protein', 'Hsp90', 'Hsp70','Hsp20', 'All_SODs','NO_dioxygenase', 'Hydroperoxide_reductase', '' ,'CBA1', 'MetH','MetE', '','Ferritin', 'Ferric_reductase', 'ISIP3', 'ISIP2A', 'ISIP1', '', 'Glutamate_synthase','Glutamine_synthetase', 'Nitrite_reductase','Nitrite_transporter', 'Nitrate_reductase', 'Nitrate_transporter', 'Urease','','Rhodopsin', 'Rubisco', 'ATP_synthase','FNR', 'Flavodoxin','Ferredoxin', 'P700', 'Lhca', 'Plastocyanin','Cyt_c6',  'Cyt_B6f', 'Mn-cluster', 'P680','CP47', 'Lhcx','Lhcr', 'Lhcf')

compartments <- c('Ribosome', '',  '' ,'CBA1', 'MetH','MetE', '','Ferritin', 'Ferric_reductase', 'ISIP3', 'ISIP2A', 'ISIP1', '','NO_dioxygenase', 'Glutamate_synthase','Glutamine_synthetase', 'Nitrite_reductase','Nitrite_transporter', 'Nitrate_reductase', 'Nitrate_transporter', 'Urease','','Rhodopsin', 'Rubisco', 'ATP_synthase','FNR', 'Flavodoxin','Ferredoxin', 'P700', 'Plastocyanin','Cyt_c6',  'Cyt_B6f', 'Mn-cluster', 'P680','CP47','' , 'Lhcr', 'Lhcf', 'Lhca')


newRow <- data.frame(fe_neg = NA, fe_pos = NA, temp_neg = NA, temp_pos = NA, tempfe_neg = NA, tempfe_pos= NA, abs_fe = NA, name = "ISIP1")


newRow2 <- data.frame(fe_neg = NA, fe_pos = NA, temp_neg = NA, temp_pos = NA, tempfe_neg = NA, tempfe_pos= NA, abs_fe = NA, name = "MetE")

newRow3 <- data.frame(fe_neg = NA, fe_pos = NA, temp_neg = NA, temp_pos = NA, tempfe_neg = NA, tempfe_pos= NA, abs_fe = NA, name = "petF_1")

pseudo_slide_sub <- rbind(pseudo_slide, newRow)
pseudo_slide_sub2 <- rbind(pseudo_slide_sub, newRow2)
pseudo_slide_sub3 <- rbind(pseudo_slide_sub2, newRow3)


cellcompartment(pseudo_slide_sub3)
pseudo_cell <- cell_compartment


cellcompartment(frag_slide)
frag_cell <- cell_compartment


nitratetranspot <- dplyr::filter(frag_cell, grepl("clust_991 ", name))

blank_p <- frag_cell %>%
    group_by(cellcompartment)%>%
    summarise(fe_grp = sum(abs_fe))%>%
    ggplot(aes(x = fe_grp,
               y = cellcompartment)) +
    theme_bw() +
    xlab('') +
    scale_y_discrete(limits = compartments)+
    theme(axis.text.x = element_text(colour = 'white'),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          axis.ticks = element_blank()) +
    ylab(''); blank_p
 


#volcanoplot_interactive(frag_ORF, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms")

#volcanoplot_interactive(rhodopsin_MCL, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_ORFs_alldiatoms")


#cobalamin dependent
#play <- dplyr::filter(pseudo_ORF, grepl(clust_1414 ", name)) 

#cobalamin indepenedent 
#play <- dplyr::filter(frag_slide, grepl("Cobalamin-independent|MetE", name)) 
#play <- dplyr::filter(frag_slide, grepl("CBA1|Cobalamine-acquisition", name)) 

#Ferredoxin <- dplyr::filter(dataset, grepl("petF_1", name))
#Ferredoxin$cellcompartment <- "Ferredoxin"


# frag_pn_fe <- frag_cell %>% 
#   group_by(cellcompartment) %>%
#   summarise(fe_grp = sum(abs_fe),
#             fe_grp_neg = sum(fe_neg),
#             fe_grp_pos = sum(fe_pos)) %>%
#   ggplot(aes(x = fe_grp_neg, y = cellcompartment)) +
#   geom_point(size = 3, alpha = 0.5, colour = 'blue') +
#   geom_point(aes(x = fe_grp_pos, y = cellcompartment),
#              size = 3, alpha = 0.5, colour = 'red') +
#   geom_segment(aes(x = fe_grp_neg, xend = 0,
#                    y = cellcompartment,
#                    yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
#   geom_segment(aes(x = fe_grp_pos, xend = 0,
#                    y = cellcompartment,
#                    yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
#   theme_bw() +
#   xlab('Iron-Related DE') +
#   ylab('') +
#   scale_y_discrete(limits =compartments)+
#   theme(axis.text.y = element_blank());frag_pn_fe
# 
# 
# 
# 
# pseudo_pn_fe <- pseudo_cell %>% 
#   group_by(cellcompartment) %>%
#   summarise(fe_grp = sum(abs_fe),
#             fe_grp_neg = sum(fe_neg),
#             fe_grp_pos = sum(fe_pos)) %>%
#   ggplot(aes(x = fe_grp_neg, y = cellcompartment)) +
#   geom_point(size = 3, alpha = 0.5, colour = 'blue') +
#   geom_point(aes(x = fe_grp_pos, y = cellcompartment),
#              size = 3, alpha = 0.5, colour = 'red') +
#   geom_segment(aes(x = fe_grp_neg, xend = 0,
#                    y = cellcompartment,
#                    yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
#   geom_segment(aes(x = fe_grp_pos, xend = 0,
#                    y = cellcompartment,
#                    yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
#   theme_bw() +
#   xlab('Iron-Related DE') +
#   ylab('') +
# scale_y_discrete(limits = compartments)+
#   theme(axis.text.y = element_blank());pseudo_pn_fe
# 
# 
# grid.arrange(blank_p,frag_pn_fe,pseudo_pn_fe, nrow = 1, widths = c(1.5,4,4))
# 


frag_pn_temp <- frag_cell %>%
  group_by(cellcompartment) %>%
  summarise(fe_grp = sum(abs_fe),
            fe_grp_neg = sum(temp_neg),
            fe_grp_pos = sum(temp_pos)) %>%
  ggplot(aes(x = fe_grp_neg, y = cellcompartment)) +
  geom_point(size = 3, alpha = 0.5, colour = 'blue') +
  geom_point(aes(x = fe_grp_pos, y = cellcompartment),
             size = 3, alpha = 0.5, colour = 'red') +
  geom_segment(aes(x = fe_grp_neg, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
  geom_segment(aes(x = fe_grp_pos, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
  theme_bw() +
  xlab('Temp-Related DE') +
  ylab('') +
scale_y_discrete(limits = compartments)+
  theme(axis.text.y = element_blank());frag_pn_temp
# 
# 
pseudo_pn_temp <- pseudo_cell %>%
  group_by(cellcompartment) %>%
  summarise(fe_grp = sum(abs_fe),
            fe_grp_neg = sum(temp_neg),
            fe_grp_pos = sum(temp_pos)) %>%
  ggplot(aes(x = fe_grp_neg, y = cellcompartment)) +
  geom_point(size = 3, alpha = 0.5, colour = 'blue') +
  geom_point(aes(x = fe_grp_pos, y = cellcompartment),
             size = 3, alpha = 0.5, colour = 'red') +
  geom_segment(aes(x = fe_grp_neg, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
  geom_segment(aes(x = fe_grp_pos, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
  theme_bw() +
  xlab('Temp-Related DE') +
  ylab('') +
scale_y_discrete(limits = compartments)+
  theme(axis.text.y = element_blank()); pseudo_pn_temp
# 
# 
# frag_pn_tempfe <- frag_cell %>% 
#   group_by(cellcompartment) %>%
#   summarise(fe_grp = sum(abs_fe),
#             fe_grp_neg = sum(tempfe_neg),
#             fe_grp_pos = sum(tempfe_pos)) %>%
#   ggplot(aes(x = fe_grp_neg, y = cellcompartment)) +
#   geom_point(size = 3, alpha = 0.5, colour = 'blue') +
#   geom_point(aes(x = fe_grp_pos, y = cellcompartment),
#              size = 3, alpha = 0.5, colour = 'red') +
#   geom_segment(aes(x = fe_grp_neg, xend = 0,
#                    y = cellcompartment,
#                    yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
#   geom_segment(aes(x = fe_grp_pos, xend = 0,
#                    y = cellcompartment,
#                    yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
#   theme_bw() +
#   xlab('Temp-Fe-Related DE') +
#   ylab('') +
# scale_y_discrete(limits = compartments)+
#   theme(axis.text.y = element_blank());frag_pn_tempfe
# 
# 
# pseudo_pn_tempfe <- pseudo_cell %>% 
#   group_by(cellcompartment) %>%
#   summarise(fe_grp = sum(abs_fe),
#             fe_grp_neg = sum(tempfe_neg),
#             fe_grp_pos = sum(tempfe_pos)) %>%
#   ggplot(aes(x = fe_grp_neg, y = cellcompartment)) +
#   geom_point(size = 3, alpha = 0.5, colour = 'blue') +
#   geom_point(aes(x = fe_grp_pos, y = cellcompartment),
#              size = 3, alpha = 0.5, colour = 'red') +
#   geom_segment(aes(x = fe_grp_neg, xend = 0,
#                    y = cellcompartment,
#                    yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
#   geom_segment(aes(x = fe_grp_pos, xend = 0,
#                    y = cellcompartment,
#                    yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
#   theme_bw() +
#   xlab('Temp-Fe-Related DE') +
#   ylab('') +
# scale_y_discrete(limits = compartments)+
#   theme(axis.text.y = element_blank());pseudo_pn_tempfe
# 
# 
# 
# 
# grid.arrange(blank_p,frag_pn_fe,pseudo_pn_fe, nrow = 1, widths = c(1.5,4,4))
# 
grid.arrange(blank_p,frag_pn_temp,pseudo_pn_temp, nrow = 1, widths = c(1.4,4,4))
# 
# grid.arrange(blank_p,frag_pn_tempfe,pseudo_pn_tempfe, nrow = 1, widths = c(1.4,4,4))



# ISIP <- dplyr::filter(dataset, grepl("ISIP|isip", name))
# ISIP$cellcompartment <- "All_ISIPs"

#TCA <- dplyr::filter(dataset, grepl("TCA", name))
#TCA$cellcompartment <- "TCA"


#lightharvesting <- dplyr::filter(dataset, grepl("Lhc|LHC|harvesting|clust_1236 |clust_1194 ", name))
#lightharvesting$cellcompartment <- "LHC"

```


sliding plot for the different cellular functions
Percent DE per group
```{r}
total_number_frag <- data.frame (table(unlist(strsplit((frag_cell$cellcompartment), " "))))

total_number_pseudo <- data.frame (table(unlist(strsplit((pseudo_cell$cellcompartment), " "))))



frag_de <- frag_cell %>% 
  group_by(cellcompartment) %>%
  summarise(fe_grp = sum(abs_fe),
            fe_grp_neg = sum(fe_neg),
            fe_grp_pos = sum(fe_pos),
            temp_grp_neg = sum(temp_neg),
            temp_grp_pos = sum(temp_pos),
            tempfe_grp_neg = sum(tempfe_neg),
            tempfe_grp_pos = sum(tempfe_pos)
            )

  frag_de$total_number= total_number_frag$Freq
  
  

  #frag_de$colname <- "colname"
  colname <- "percent"
  i=0
  while (i < 7){
    colname <- paste(colname, i)
    col1 = 2+(i*1)
   col2 = 9 
    frag_de[[colname]] <- (frag_de[,col1]/frag_de[,col2])*100
    i <- i+1
  }
  
  frag_de_sub <- do.call(data.frame, frag_de)
  
  colnames (frag_de_sub) <- c("cellcompartment", "fe_grp", "fe_neg", "fe_pos", "temp_neg", "temp_pos", "tempfe_neg", "tempfe_pos", "total", "fe_grpa", "fe_nega", "fe_posa", "temp_nega", "temp_posa", "tempfe_nega", "tempfe_posa" )
  


  
pseudo_de <- pseudo_cell %>% 
  group_by(cellcompartment) %>%
  summarise(fe_grp = sum(abs_fe),
            fe_grp_neg = sum(fe_neg),
            fe_grp_pos = sum(fe_pos),
            temp_grp_neg = sum(temp_neg),
            temp_grp_pos = sum(temp_pos),
            tempfe_grp_neg = sum(tempfe_neg),
            tempfe_grp_pos = sum(tempfe_pos)
            )

  pseudo_de$total_number= total_number_pseudo$Freq
  
  
  colname <- "percent"
  i=0
  while (i < 7){
    colname <- paste(colname, i)
    col1 = 2+(i*1)
   col2 = 9 
    pseudo_de[[colname]] <- (pseudo_de[,col1]/pseudo_de[,col2])*100
    i <- i+1
  }
  
  pseudo_de_sub <- do.call(data.frame, pseudo_de)
  
  colnames (pseudo_de_sub) <- c("cellcompartment", "fe_grp", "fe_neg", "fe_pos", "temp_neg", "temp_pos", "tempfe_neg", "tempfe_pos", "total", "fe_grpa", "fe_nega", "fe_posa", "temp_nega", "temp_posa", "tempfe_nega", "tempfe_posa" )
  
  
   frag_fe_plot<- ggplot(frag_de_sub, aes(x = fe_nega, y = cellcompartment)) +
  geom_point(size = 3, alpha = 0.5, colour = 'blue') +
  geom_point(aes(x = fe_posa, y = cellcompartment),
             size = 3, alpha = 0.5, colour = 'red') +
  geom_segment(aes(x = fe_nega, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
  geom_segment(aes(x = fe_posa, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
  theme_bw() +
  xlab('Iron-Related DE') +
  ylab('') +
  scale_y_discrete(limits =compartments)+
  theme(axis.text.y = element_blank());frag_fe_plot 

  
   frag_temp_plot<- ggplot(frag_de_sub, aes(x = temp_nega, y = cellcompartment)) +
  geom_point(size = 3, alpha = 0.5, colour = 'blue') +
  geom_point(aes(x = temp_posa, y = cellcompartment),
             size = 3, alpha = 0.5, colour = 'red') +
  geom_segment(aes(x = temp_nega, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
  geom_segment(aes(x = temp_posa, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
  theme_bw() +
  xlab('Temp-Related DE') +
  ylab('') +
  scale_y_discrete(limits =compartments)+
  theme(axis.text.y = element_blank());frag_temp_plot 

   frag_tempfe_plot<- ggplot(frag_de_sub, aes(x = tempfe_nega, y = cellcompartment)) +
  geom_point(size = 3, alpha = 0.5, colour = 'blue') +
  geom_point(aes(x = tempfe_posa, y = cellcompartment),
             size = 3, alpha = 0.5, colour = 'red') +
  geom_segment(aes(x = tempfe_nega, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
  geom_segment(aes(x = tempfe_posa, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
  theme_bw() +
  xlab('TempFe-Related DE') +
  ylab('') +
  scale_y_discrete(limits =compartments)+
  theme(axis.text.y = element_blank());frag_tempfe_plot 

  
  
  
  
  pseudo_fe_plot<- ggplot(pseudo_de_sub, aes(x = fe_nega, y = cellcompartment)) +
  geom_point(size = 3, alpha = 0.5, colour = 'blue') +
  geom_point(aes(x = fe_posa, y = cellcompartment),
             size = 3, alpha = 0.5, colour = 'red') +
  geom_segment(aes(x = fe_nega, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
  geom_segment(aes(x = fe_posa, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
  theme_bw() +
  xlab('Iron-Related DE') +
  ylab('') +
  scale_y_discrete(limits =compartments)+
  theme(axis.text.y = element_blank());pseudo_fe_plot 
  

  
    pseudo_temp_plot<- ggplot(pseudo_de_sub, aes(x = temp_nega, y = cellcompartment)) +
  geom_point(size = 3, alpha = 0.5, colour = 'blue') +
  geom_point(aes(x = temp_posa, y = cellcompartment),
             size = 3, alpha = 0.5, colour = 'red') +
  geom_segment(aes(x = temp_nega, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
  geom_segment(aes(x = temp_posa, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
  theme_bw() +
  xlab('Temp-Related DE') +
  ylab('') +
  scale_y_discrete(limits =compartments)+
  theme(axis.text.y = element_blank());pseudo_temp_plot 

    
 pseudo_tempfe_plot<- ggplot(pseudo_de_sub, aes(x = tempfe_nega, y = cellcompartment)) +
  geom_point(size = 3, alpha = 0.5, colour = 'blue') +
  geom_point(aes(x = tempfe_posa, y = cellcompartment),
             size = 3, alpha = 0.5, colour = 'red') +
  geom_segment(aes(x = tempfe_nega, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'blue', alpha = 0.2, lwd = 2) +
  geom_segment(aes(x = tempfe_posa, xend = 0,
                   y = cellcompartment,
                   yend = cellcompartment), colour = 'red', alpha = 0.2, lwd = 2) +
  theme_bw() +
  xlab('TempFe-Related DE') +
  ylab('') +
  scale_y_discrete(limits =compartments)+
  theme(axis.text.y = element_blank());pseudo_tempfe_plot 

blank_p <- frag_cell %>%
    group_by(cellcompartment) %>%
     summarise(fe_grp = sum(abs_fe)) %>%
     ggplot(aes(x = fe_grp,
               y = cellcompartment)) +
    theme_bw() +
    xlab('') +
  scale_y_discrete(limits = compartments)+
         theme(axis.text.x = element_text(colour = 'white'),
          panel.grid.major.x = element_blank(),
           panel.grid.minor.x = element_blank(),
            axis.ticks = element_blank()) +
     ylab(''); blank_p



grid.arrange(blank_p,frag_fe_plot,pseudo_fe_plot, nrow = 1, widths = c(1.5,4,4))
grid.arrange(blank_p,frag_temp_plot,pseudo_temp_plot, nrow = 1, widths = c(1.5,4,4))

grid.arrange(blank_p,frag_tempfe_plot,pseudo_tempfe_plot, nrow = 1, widths = c(1.5,4,4))

#MetE = B12 (cobalamine) independent methionine synthase
#MetH = B12 (cobalamine) dependent methionine synthase
#CBA1 = B12 (cobalamine) acquisition protein
```


foldchange magnitude of groups
```{r}
cellcompartment(frag_ORF)
frag_cell <- cell_compartment

frag_cell$category <- "photosynthesis"


#volcanoplot_interactive(frag_cell, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_clusters_alldiatoms")


#ISIP <- dplyr::filter(pseudo_frag_MCL, grepl("ISIP|isip|clust_9740 |clust_343 |clust_1814 |clust_1154 ", name))

#ISIP <- dplyr::filter(all_ORF_sub, grepl("ferritin|Ferritin|FTN|clust_15657 |clust_3034 |contig_817078_130_537_-", name))

#clustername_vs_foldchange(ISIP, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_clusters_alldiatoms")



#volcanoplot_interactive(ISIP, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_clusters_alldiatoms")


#ISIP1 <- dplyr::filter(frag_ORF, grepl("ISIP1", name))



#ISIP3 <- dplyr::filter(all_ORF, grepl("contig_680152_23_630_-", name))
#ISIP3$cellcompartment <- "ISIP3"


lhcx<-  dplyr::filter(frag_ORF, grepl("Lhcx", name))
Lhcx$cellcompartment <- "Lhcx"

Lhcf<-  dplyr::filter(dataset, grepl("Lhcf", name))
Lhcf$cellcompartment <- "Lhcf"

Lhcr<-  dplyr::filter(dataset, grepl("Lhcr", name))
Lhcr$cellcompartment <- "Lhcr"

Lhca<-  dplyr::filter(dataset, grepl("Lhca", name))
Lhca$cellcompartment <- "Lhca"


ISIP<- dplyr::filter (frag_MCL, grepl('clust_1814 |clust_9740 |clust_1154 ', name))

volcanoplot(frag_cell, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_ISIP_ORFs_Frag")

volcanoplot_interactive(DA, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_ORFs_alldiatoms")


volcanoplot_interactive( DA, "int_fe_temp_3C_vs_0C_foldchange", "int_fe_temp_3C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_MCL_alldiatoms")



#DAd from genbank
DA<- dplyr::filter(pseudo_ORF, grepl("clust_12542 ", name))


#https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4460010/
```


```{r}
cellcompartment <- function (dataset){
  
Rhodopsin <- dplyr::filter(dataset, grepl("rhodopsin|Rhodopsin", name))
Rhodopsin$cellcompartment <- "Rhodopsin"

PSIIreactioncenter <- dplyr::filter(dataset, grepl("P680", name))
PSIIreactioncenter$cellcompartment <- "P680"

manganesecluster <- dplyr::filter(dataset, grepl("OEC|oec|evolving|OEE|PsbO|PsbP|PsbQ", name))
manganesecluster$cellcompartment <- "Mn-cluster"

PSIreactioncenter <- dplyr::filter(dataset, grepl("p700|P700", name))
PSIreactioncenter$cellcompartment <- "P700"

Flavodoxin <- dplyr::filter(dataset, grepl("flavodoxin|Flavodoxin", name))
Flavodoxin$cellcompartment <- "Flavodoxin"

Ferredoxin <- dplyr::filter(dataset, grepl("petF_1", name))
Ferredoxin$cellcompartment <- "Ferredoxin"

Cyt_B6f <- dplyr::filter(dataset, grepl("b6f|Rieske|petF_2", name))
Cyt_B6f$cellcompartment <- "Cyt_B6f"

ATPsynthase <- dplyr::filter(dataset, grepl("atpA|atpB", name))
ATPsynthase$cellcompartment <- "ATP_synthase"

Rubisco <- dplyr::filter(dataset, grepl("RUBISCO|Rubisco|rubisco", name))
Rubisco$cellcompartment <- "Rubisco"

Plastocyanin<- dplyr::filter(dataset, grepl("plastocyanin|Plastocyanin", name))
Plastocyanin$cellcompartment <- "Plastocyanin"

Ribosome <- dplyr::filter(dataset, grepl("ribosome|ribosom|Ribosom", name))
Ribosome$cellcompartment <- "Ribosome"

All_SODs <- dplyr::filter(dataset, grepl("SOD|superoxide", name))
All_SODs$cellcompartment <- "All_SODs"

Nitrate_reductase <- dplyr::filter(dataset, grepl("clust_411 ", name))
Nitrate_reductase$cellcompartment <- "Nitrate_reductase"

Nitrate_transporter <- dplyr::filter(dataset, grepl("nitrate transporter ", name))
Nitrate_transporter$cellcompartment <- "Nitrate_transporter"

Nitrite_reductase <- dplyr::filter(dataset, grepl("clust_1816 |clust_3042 |clust_3563 ", name))
Nitrite_reductase$cellcompartment <- "Nitrite_reductase"

Nitrite_transporter <- dplyr::filter(dataset, grepl("nitrite transporter", name))
Nitrite_transporter$cellcompartment <- "Nitrite_transporter"

CP47<- dplyr::filter(dataset, grepl("CP47", name))
CP47$cellcompartment <- "CP47"

ISIP1 <- dplyr::filter(dataset, grepl("ISIP1", name))
ISIP1$cellcompartment <- "ISIP1"

ISIP2A <- dplyr::filter(dataset, grepl("ISIP2A", name))
ISIP2A$cellcompartment <- "ISIP2A"

ISIP3 <- dplyr::filter(dataset, grepl("ISIP3", name))
ISIP3$cellcompartment <- "ISIP3"

Glutamine_synthetase <-  dplyr::filter(dataset, grepl("glutamine synthetase ", name))
Glutamine_synthetase$cellcompartment <- "Glutamine_synthetase"

Glutamate_synthase <-  dplyr::filter(dataset, grepl("glutamate synthase ", name))
Glutamate_synthase$cellcompartment <- "Glutamate_synthase"

Hydroperoxide_reductase <- dplyr::filter(dataset, grepl("hydroperoxide", name))
Hydroperoxide_reductase$cellcompartment <- "Hydroperoxide_reductase"

Hsp70 <-  dplyr::filter(dataset, grepl("Hsp70", name))
Hsp70 $cellcompartment <- "Hsp70"

Hsp90 <-  dplyr::filter(dataset, grepl("Hsp90|HSP90", name))
Hsp90 $cellcompartment <- "Hsp90"

Hsp20 <-  dplyr::filter(dataset, grepl("Hsp20", name))
Hsp20 $cellcompartment <- "Hsp20"

coldshock_protein<-  dplyr::filter(dataset, grepl("cold", name))
coldshock_protein$cellcompartment <- "coldshock_protein"

Lhcx<-  dplyr::filter(dataset, grepl("Lhcx", name))
Lhcx$cellcompartment <- "Lhcx"

Lhcf<-  dplyr::filter(dataset, grepl("Lhcf", name))
Lhcf$cellcompartment <- "Lhcf"

Lhcr<-  dplyr::filter(dataset, grepl("Lhcr", name))
Lhcr$cellcompartment <- "Lhcr"

Lhca<-  dplyr::filter(dataset, grepl("Lhca", name))
Lhca$cellcompartment <- "Lhca"

FNR <- dplyr::filter(dataset, grepl("clust_808 ", name))
FNR$cellcompartment <- "FNR"


Urease <- dplyr::filter(dataset, grepl("urease|Urease", name))
Urease$cellcompartment <- "Urease"


MetE <- dplyr::filter(dataset, grepl("Cobalamin-independent|MetE", name)) 
MetE $cellcompartment <- "MetE"

CBA1 <- dplyr::filter(dataset, grepl("CBA1", name)) 
CBA1 $cellcompartment <- "CBA1"


MetH <- dplyr::filter(dataset, grepl("METH", name)) 
MetH $cellcompartment <- "MetH"

Ferritin <-  dplyr::filter(dataset, grepl("Ferritin|ferritin", name))
Ferritin $cellcompartment <- "Ferritin"

NO_dioxygenase <-  dplyr::filter(dataset, grepl("clust_4956  ", name))
NO_dioxygenase $cellcompartment <- "NO_dioxygenase"

Ferric_reductase <-  dplyr::filter(dataset, grepl("ferric reductase  |Ferric", name))
Ferric_reductase $cellcompartment <- "Ferric_reductase"

Cyt_c6 <-  dplyr::filter(dataset, grepl("clust_1414 ", name))
Cyt_c6 $cellcompartment <- "Cyt_c6"

cell_compartment <- rbind (Cyt_c6, Ferric_reductase, Ferritin, CBA1, Ferredoxin, MetE, ISIP1, MetH, Urease, FNR, Lhca, Lhcf, Lhcx, Lhcr, coldshock_protein,  Hsp70, Hsp90, Hsp20, Hydroperoxide_reductase, ISIP2A,ISIP3, Nitrate_transporter, Nitrite_transporter, Glutamine_synthetase, Glutamate_synthase, CP47, Nitrite_reductase, Nitrate_reductase, Ribosome, All_SODs,NO_dioxygenase, Rhodopsin,PSIIreactioncenter,Cyt_B6f, manganesecluster, PSIreactioncenter,Flavodoxin, ATPsynthase, Rubisco, Plastocyanin)

cell_compartment <<- cell_compartment

}

cellcompartment(frag_ORF)
frag_cell <- cell_compartment
frag_cell$category <- "other"


LHC <- "Light Harvesting Complex"
core_photosynthesis <- "Core Photosynthesis"
nitrogen <- "Nitrogen Assimilation"
iron <- "Fe Transport/Storage"
B12 <- "B12-related"
anitox <- "Antioxidant"
ribosome <- "Ribosome"
rhodopsin <- "Rhodopsin"


categorymap <- c("Lhcf"= LHC, "Lhcr" = LHC, "Lhca" = LHC, "Lhcx" = LHC, "CP47" = photosynthesis,  "P680"=photosynthesis, "Mn-cluster" = photosynthesis,"Cyt_B6f" = photosynthesis,"Cyt_c6" = photosynthesis, "Plastocyanin" = photosynthesis, "P700" = photosynthesis,"Ferredoxin" = photosynthesis, "Flavodoxin" = photosynthesis, "FNR" = photosynthesis, "ATP_synthase" = photosynthesis, "Rubisco" = photosynthesis, "Rhodopsin" = rhodopsin, "Urease" = nitrogen, "Nitrate_transporter" = nitrogen, "Nitrate_reductase" = nitrogen, "Nitrite_transporter" = nitrogen, "Nitrite_reductase" = nitrogen, "Glutamine_synthetase" = nitrogen, "Glutamate_synthase" = nitrogen, "ISIP1" = iron, "ISIP2A" = iron, "ISIP3" = iron, "Ferritin" = iron, "Ferric_reductase" = iron, "MetE" = B12, "MetH" = B12, "CBA1" = B12, "Hydroperoxide_reductase" = anitox, "NO_dioxygenase" = anitox, "All_SODs" = anitox, "Hsp20" = anitox, "Hsp70" = anitox, "Hsp90" = anitox, "coldshock_protein" = anitox, "Ribosome" = ribosome  )


frag_cell$category <- categorymap[frag_cell$cellcompartment]


#volcanoplot_interactive_groups(frag_cell, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_Frag")


#volcanoplot_interactive_groups(frag_cell, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "hightemp_vs_lowtemp_frag")


#volcanoplot_interactive_groups(frag_cell,"int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", " tempfe_interaction_Frag")


volcanoplot <- function(dataset, foldchange, FDR, graphtitle){
  volcano <- dataset[c("name",foldchange, FDR,"Taxa.group", "category" )]
  x <- foldchange
  
  names(volcano) <- c( "gene", "Fold", "FDR", "taxa", "category")
  volcano ["group"] <- "no-fold-change"
  
  if(grepl("Fe_vs_noFe_foldchange", x)) {
    volcano['size'] <- rowMeans( dataset[, c(4:21)])
  } else if (grepl("temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("temp_6C_vs_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:21)])
  } else if (grepl("int_fe_temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("int_fe_temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("noFe_3C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6,10:12 )])
  } else if (grepl("noFe_6C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6, 16:18)])
  } else if (grepl("Fe_3C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(7:9, 13:15)])
  } else if (grepl("Fe_6C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans(dataset[, c(7:9, 19:21)])
  } else if (grepl("noFe_6C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:18,10:12)])
  } else if (grepl("Fe_6C_vs_Fe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(13:15, 19:21)])
  } else if (grepl("Fe_0C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9)])
  } else if (grepl("Fe_3C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:15)])
  } else if (grepl("Fe_6C_vs_noFe_6C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:21)])
  }
  

  volcano [which(volcano['FDR'] < 0.05 & abs(volcano['Fold']) < 2), "group"]<- "nofoldchange_significant" 
  volcano [which(volcano ['FDR'] > 0.05 & abs(volcano['Fold'])> 2), "group"] <- "fold-change_notsignificant"
  volcano [which(volcano ['FDR'] < 0.05 & abs(volcano['Fold']) > 2), "group"] <- "fold-change_significant"
  
  ggplot (data = volcano, aes(x= Fold, y=-log10(FDR), color = category, label=gene)) +
    geom_point (alpha=0.5)+
    scale_shape (solid=FALSE)+
    geom_hline (yintercept = 1.301, col = "blue", lty = 2, lwd=0.5 )+
    geom_vline (xintercept = c(-1,1), col = "blue", lty = 2, lwd=0.5 )+
    xlab("Log2 Fold Change")+ ylab("-Log10_P-value")+
   # scale_color_manual(values = c("black", "red", "#00FF00"))+
    ggtitle(graphtitle)+
    theme_bw()+
    theme(plot.title = element_text(hjust = 0.5))
    #geom_label_repel(data = subset(volcano, abs(Fold) >1 & -log10(FDR) >1.3),
     # aes(label=gene), box.padding = 0.2, point.padding = 0.5, color= 'black', size =1)
}

































#volcanoplot_interactive_groups <- function(dataset, Fold, FDR, volcanontitle){
  volcano <- dataset[c("name",Fold, FDR,"Taxa.group", "category")]
  x <- Fold
  
  names(volcano) <- c( "gene", "Fold", "FDR", "taxa", "category")
  volcano ["group"] <- "no-fold-change"
  
  volcano [which(volcano['FDR'] < 0.05 & abs(volcano['Fold']) < 2), "group"]<- "nofoldchange_significant" 
  
  volcano [which(volcano ['FDR'] > 0.05 & abs(volcano['Fold'])> 2), "group"] <- "fold-change_notsignificant"
  
  volcano [which(volcano ['FDR'] < 0.05 & abs(volcano['Fold']) > 2), "group"] <- "fold-change_significant"
  
  vline <- function(x = 0, color = "black") {
    list(
      type = "line", 
      y0 = 0, 
      y1 = 1, 
      yref = "paper",
      x0 = x, 
      x1 = x, 
      line = list(color = color, dash='dot', width=1)
    )
  }
  
  hline <- function(y = 0, color = "black") {
    list(
      type = "line", 
      x0 = 0, 
      x1 = 1, 
      xref = "paper",
      y0 = y, 
      y1 = y, 
      line = list(color = color, dash='dot', width=1)
    )
  }
  
  x <- list (title ="Log2 Fold Change")
  y <- list (title = "-Log10_P-value")
 # ax<- list(showline = TRUE, zeroline=FALSE, linecolor = toRGB("black"),
            #linewidth = 6)
  plot_ly(data = volcano, x = volcano$Fold, y = -log10(volcano$FDR), text = volcano$gene, mode = "markers", symbol = volcano$category)%>% #size = volcano$size )%>% 
    layout(title = volcanontitle)%>%
    layout (xaxis = x, yaxis = y)%>%
    layout(shapes = list(vline (-1), vline(1), hline(1.301)))
    #layout (xaxis = ax, yaxis =ax)
}
```


clusters in each of the groups, so we can make a cluster plot
```{r}

clustername_vs_foldchange_group <- function(dataset, foldchange, FDR, graphtitle){
  volcano <- dataset[c("name",foldchange, FDR,"Taxa.group", "clustercompartment" )]
  x <- foldchange
  
  names(volcano) <- c( "gene", "Fold", "FDR", "taxa", "clustercompartment")
  volcano ["group"] <- "no-fold-change"
  
  if(grepl("Fe_vs_noFe_foldchange", x)) {
    volcano['size'] <- rowMeans( dataset[, c(4:21)])
  } else if (grepl("temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("temp_6C_vs_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:21)])
  } else if (grepl("int_fe_temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("int_fe_temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("noFe_3C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6,10:12 )])
  } else if (grepl("noFe_6C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6, 16:18)])
  } else if (grepl("Fe_3C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(7:9, 13:15)])
  } else if (grepl("Fe_6C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans(dataset[, c(7:9, 19:21)])
  } else if (grepl("noFe_6C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:18,10:12)])
  } else if (grepl("Fe_6C_vs_Fe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(13:15, 19:21)])
  } else if (grepl("Fe_0C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9)])
  } else if (grepl("Fe_3C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:15)])
  } else if (grepl("Fe_6C_vs_noFe_6C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:21)])
  }
  
#volcano$shape <- "1"
#if(subset(volcano, -log10(FDR) >1.3)) {volcano$shape <- "2"}

#shape[volcano$ FDR < 0.05] <- "19"
#shape[volcano$ FDR > 0.05] <- "1"

# volcano$shape<- "4"
# if (volcano$FDR > 0.05) {
#     volcano$shape <- "1"
# } 

  
volcano$shapez <-  ifelse (volcano$FDR < 0.05, "1", "16") 



  ggplot (data = volcano, aes(x= Fold, y= clustercompartment, color = taxa, label=gene, shape = shapez)) +
    geom_point (alpha=0.7, size =2)+
   
    scale_shape_manual (values = c(16, 1), guide = FALSE)+
    ylab("cluster")+ 
    xlab("log2 fold-change")+
    scale_color_manual(values = c("black", "red"))+
    scale_y_discrete(limits = compartments)+
    ggtitle(graphtitle)+
    theme_bw()+
    theme(plot.title = element_text(hjust = 0.5))+
    theme(axis.text.x = element_text(
            hjust=1,
      size = 11, 
      face = "bold")) +
    theme(axis.text.y = element_text(
            hjust=1,
      size = 11, 
      face = "bold")) +
    geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )
    
     # geom_text_repel(data = subset(volcano, -log10(FDR) >1.3),
  #aes(label="*"), box.padding = 0, point.padding = 0.1, color= 'blue', size =3)
  
  #geom_text_repel(data = subset(volcano, -log10(FDR) >1.3),
  #aes(label="*"), box.padding = 0, point.padding = 0.1, color= 'blue', size =3)
}

  
  
#the code below goes through each 'group', so LHCs, ATP-synthase etc.. and picks out MCL clusters from each group, and plots them to show fold change. blue stars represent signigicant fold change. 

#this is to pick out the cluster for each of the groups, inside the grepl function is the terminooglogy I used for my sliding plots.  
clusters<- dplyr::filter(pseudo_frag_ORF, grepl("nitrate transporter " , name))
unique(clusters$cluster)


clustername_vs_foldchange(clusters,"Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "Fe_vs_noFe")

clustername_vs_foldchange(clusters, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "hightemp_vs_lowtemp")


volcanoplot_interactive(clusters,"Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "Fe_vs_noFe")

volcanoplot_interactive(pseudo_frag_MCL, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "hightemp_vs_lowtemp")



#create a dataset with all the clusters for all the groups. 
cluster_dataset <- dplyr::filter(pseudo_frag_MCL, grepl("clust_332 |clust_1236 |clust_80 |clust_402 |clust_1044 |clust_3282 |clust_1194 |clust_3219 |clust_1084 |clust_3551 |clust_646  |clust_1938 |clust_2549 |clust_3551 |clust_1787 |clust_712 |clust_2256 |clust_3363 |clust_1414 |clust_498 |clust_211 |clust_42 |clust_24191 |clust_55 |clust_2763 |clust_1906 |clust_15862 |clust_5856 |clust_27282 |clust_5235 |clust_14895 |clust_86124 |clust_12649 |clust_26101 |clust_231 |clust_1548 |clust_1820 |clust_2212 |clust_1822 |clust_1668 |clust_1196 |clust_1452 |clust_534 |clust_4660 |clust_808 |clust_349 |clust_78 |clust_306 |clust_1846 |clust_129 |clust_816 |clust_3167 |clust_120 |clust_3290 |clust_9740 |clust_343 |clust_1814 |clust_1154 |clust_991 |clust_411 |clust_1816 |clust_2277 |clust_589 |clust_284 |clust_891 |clust_15657 |clust_4956 |clust_6089 |clust_7529 |clust_2658 |clust_557 |clust_3360 |clust_35 |clust_3360 |clust_15652 |clust_10606 |clust_1 |clust_550 |clust_1 |clust_4298 |clust_4298 |clust_823 |clust_6713 |clust_444 |clust_22 |clust_219 |clust_4251 |clust_5230 |clust_7451 |clust_417 ", name))

#make a library of all the names
cluster_dataset$cluster <- as.character(cluster_dataset$cluster)
lhcf <- "LHC-f"
lhcr <- "LHC-r"
lhca <- "LHC-a"
lhcx <- "LHC-x"
p680 <- "P680"
mnclus <- "Manganese Cluster"
cytb6f <- "Cytochrome b6f Complex"
atpsyn <- "ATP Synthase"
rib <- "Ribosome"



#compartments <- c('Ribosome', '', 'coldshock_protein', 'Hsp90', 'Hsp70','Hsp20', 'All_SODs','Nitric Oxide Dioxygenase', 'Hydroperoxide_reductase', '' ,'CBA1', 'MetH','MetE', '','Ferritin', 'Ferric_reductase', 'ISIP3', 'ISIP2A', 'ISIP1', '', 'Glutamate Synthase','Glutamine Synthetase', 'Nitrite Reductase','Nitrite Transporter', 'Nitrate Reductase', 'Nitrate Transporter', 'Urease','','Rhodopsin_1', 'Rhodopsin_2', 'RuBisCO', 'ATP Synthase','FNR', 'Flavodoxin','Ferredoxin', 'P700', 'LHC-a', 'Plastocyanin','Cytochrome c6',  'Cytochrome b6f Complex', 'Manganese Cluster', 'P680','PSII CP-47 reaction center protein', 'LHC-x','LHC-r', 'LHC-f')


compartments <- c('Ribosome', '' ,'CBA1', 'MetH','MetE', '','Ferritin', 'Ferric Reductase', 'ISIP3', 'ISIP2A', 'ISIP1', '', 'Glutamate Synthase','Glutamine Synthetase', 'Nitrite Reductase','Nitrite Transporter', 'Nitrate Reductase', 'Nitrate Transporter', 'Urease','Ammonium Transporter', '','Rhodopsin_1','Rhodopsin_2', 'RuBisCO', 'ATP Synthase','FNR', 'Flavodoxin','Ferredoxin', 'P700', 'Plastocyanin','Cytochrome c6',  'Cytochrome b6f Complex', 'Manganese Cluster', 'P680','PSII CP-47 reaction center protein','' , 'LHC-r', 'LHC-f', 'LHC-a')



#match all the clusters to a group name

categorymap_cluster <- c("clust_417" = "Ammonium Transporter", "clust_332" = lhcf, "clust_1236" = lhcf, "clust_80" = lhcf, "clust_402" = lhcf, "clust_1044" = lhcf, "clust_3282" = lhcf, "clust_1194" = lhcf, "clust_3219" = lhcr, "clust_1084"=lhcr, "clust_3551"=lhcr, "clust_646"=lhcr,  "clust_1938"=lhcr, "clust_2549" = lhca, "clust_3551" = lhca, "clust_1787" =lhcx, "clust_712"=lhcx,  "clust_2256"=lhcx, "clust_3363"=lhcx, "clust_1414" = "Cytochrome c6", "clust_498" = "PSII CP-47 reaction center protein", "clust_211" = p680, "clust_42" = p680, "clust_24191" = p680, "clust_55" = "P700", "clust_2763" = mnclus,  "clust_1906"=mnclus,  "clust_15862" = mnclus,  "clust_5856"=mnclus,   "clust_27282" =mnclus,  "clust_5235"=mnclus,   "clust_14895"=mnclus, "clust_86124" = mnclus,  "clust_12649" =mnclus,  "clust_26101" = mnclus, "clust_231" = "Rhodopsin_1", "clust_1548" = "Rhodopsin_2", "clust_1820" = "Plastocyanin", "clust_2212" = cytb6f, "clust_1822" = cytb6f, "clust_1668" = cytb6f, "clust_1196" =cytb6f, "clust_1452" = "Ferredoxin", "clust_534" = "Flavodoxin", "clust_4660" = "Flavodoxin", "clust_808" = "FNR", "clust_78" = atpsyn, "clust_349" = atpsyn, "clust_306" = atpsyn, "clust_1846" = atpsyn, "clust_129" = "RuBisCO", "clust_816" = "RuBisCO", "clust_3167" = "MetE", "clust_120" = "MetH", "clust_3290" = "CBA1", "clust_9740" = "ISIP2A", "clust_343" = "ISIP2A", "clust_1814" = "ISIP1", "clust_1154" = "ISIP3", "clust_991" = "Nitrate Transporter", "clust_411" = "Nitrate Reductase", "clust_1816" = "Nitrite Reductase", "clust_2277" = "Nitrite Transporter", "clust_589" = "Glutamine Synthetase", "clust_284" = "Glutamine Synthetase", "clust_891" = "Glutamate Synthase", "clust_15657" = "Ferritin","clust_4956" = "Nitric Oxide Dioxygenase", 
#ribosomes:
"clust_6089" = rib, "clust_7529" = rib,  "clust_2658" = rib, "clust_557" = rib, "clust_3360" = rib, "clust_35"=rib, "clust_3360" = rib, "clust_15652"=rib,  "clust_10606" = rib, "clust_1"=rib, "clust_550" =rib, "clust_1"=rib, "clust_4298" = rib,  "clust_4298" = rib,  "clust_823" = rib, "clust_6713" = rib,  "clust_444"=rib,  "clust_22"=rib,  "clust_219" = rib, "clust_4251"=rib ,"clust_5230" = rib,  "clust_7451" = rib )


cluster_dataset$clustercompartment <- categorymap_cluster[cluster_dataset$cluster]



clustername_vs_foldchange_group(cluster_dataset,"Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "Fe_vs_noFe")


clustername_vs_foldchange_group(cluster_dataset, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "hightemp_vs_lowtemp")


clustername_vs_foldchange_group(cluster_dataset,"int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", " tempfe_interaction")




#Lhcf: clust_332   clust_1236  clust_80    clust_402   clust_1044  clust_3282  clust_1194 clust_58632 clust_19392
# Lhcr: clust_3219 clust_1084 clust_3551 clust_646  clust_1938
# Lhcx: clust_1787 clust_712  clust_2256 clust_3363
# Lhca : clust_2549 clust_3551


# Cyt_c6: clust_1414
# Ferric_reductase:clust_9816
# Ferritin: clust_15657 clust_3034 , will use 15657
# CBA1 :clust_3290
# Ferredoxin :clust_1452
# MetE : clust_3167
# ISIP2A : clust_343   clust_9740  clust_25598 clust_23469 , will use 343 and 9740, most abundand and most number of ORFs
# MetH: clust_120
# Urease
# FNR :clust_808
# coldshock_protein
# Hsp70
# Hsp90
# Hsp20
# Hydroperoxide_reductase
# ISIP1:"1814"
# ISIP3: clust_1154
# Nitrate_transporter "clust_991"
# Nitrite_transporter clust_2277
# Glutamine_synthetase clust_284   clust_589   clust_19873, will use 589 and 284
# Glutamate_synthase clust_241   clust_35306 clust_4150  clust_1536  clust_891  will use 891
# CP47 : clust_498
# Nitrite_reductase "clust_1816 |clust_3042 |clust_3563, will only use 1816
# Nitrate_reductase clust_411
# Ribosome
# All_SODs
# NO_dioxygenase: clust_4956  
# Rhodopsin: clust_231   clust_10297 clust_4014  clust_22740 clust_1548  clust_8237  clust_19837. only picked 231 and 1548
# PSIIreactioncenter : clust_211   clust_42    clust_24191
# Cyt_B6f : clust_12658 clust_13257 clust_2212  clust_1668  clust_6832  clust_1196  clust_36838 clust_1822  clust_8685 . only used 2212, 1822, 1668, 1196
# manganesecluster:clust_2763  clust_1906  clust_15862 clust_5856  clust_27282 clust_5235  clust_14895 clust_86124 clust_12649 clust_26101
# PSIreactioncenter : clust_55
# Flavodoxin : clust_534   clust_6705  clust_14718 clust_4660  clust_4907  clust_33174 clust_7510 clust_27848 clust_7157  clust_12773. chose 534 (most abundant), and 4660, high number of ORFs
# ATPsynthase: clust_8057 clust_1846 clust_78   clust_349  clust_43   clust_412  clust_306. will use 78, 349, 1846
# Rubisco : clust_9476  clust_753   clust_7380  clust_3897  clust_10038 clust_68857 clust_816  clust_129 . will use 129 (most abundant) and 816, many ORFS
# Plastocyanin: clust_1820 clust_2908, will only choose 1820


clustername_vs_foldchange_group <- function(dataset, foldchange, FDR, graphtitle){
  volcano <- dataset[c("name",foldchange, FDR,"Taxa.group", "clustercompartment" )]
  x <- foldchange
  
  names(volcano) <- c( "gene", "Fold", "FDR", "taxa", "clustercompartment")
  volcano ["group"] <- "no-fold-change"
  
  if(grepl("Fe_vs_noFe_foldchange", x)) {
    volcano['size'] <- rowMeans( dataset[, c(4:21)])
  } else if (grepl("temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("temp_6C_vs_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:21)])
  } else if (grepl("int_fe_temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("int_fe_temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("noFe_3C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6,10:12 )])
  } else if (grepl("noFe_6C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6, 16:18)])
  } else if (grepl("Fe_3C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(7:9, 13:15)])
  } else if (grepl("Fe_6C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans(dataset[, c(7:9, 19:21)])
  } else if (grepl("noFe_6C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:18,10:12)])
  } else if (grepl("Fe_6C_vs_Fe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(13:15, 19:21)])
  } else if (grepl("Fe_0C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9)])
  } else if (grepl("Fe_3C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:15)])
  } else if (grepl("Fe_6C_vs_noFe_6C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:21)])
  }
  
#volcano$shape <- "1"
#if(subset(volcano, -log10(FDR) >1.3)) {volcano$shape <- "2"}

#shape[volcano$ FDR < 0.05] <- "19"
#shape[volcano$ FDR > 0.05] <- "1"

# volcano$shape<- "4"
# if (volcano$FDR > 0.05) {
#     volcano$shape <- "1"
# } 

  
volcano$shapez <-  ifelse (volcano$FDR < 0.05, "1", "16") 



  ggplot (data = volcano, aes(x= Fold, y= clustercompartment, color = taxa, label=gene, shape = shapez)) +
    geom_point (alpha=0.5)+
   
   # geom_text_repel(data = subset(volcano, -log10(FDR) >1.3),
  #aes(label="*"), box.padding = 0, point.padding = 0.1, color= 'blue', size =3)
    scale_shape_manual (values = c(16, 1), guide = FALSE)+
    ylab("cluster")+ 
    xlab("log2 fold-change")+
    scale_color_manual(values = c("black", "red"))+
    scale_y_discrete(limits = compartments)+
    ggtitle(graphtitle)+
    theme_bw()+
    theme(plot.title = element_text(hjust = 0.5))+
    theme(axis.text.x = element_text(
            hjust=1,
      size = 9, 
      face = "bold")) +
    geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )
    
  #geom_text_repel(data = subset(volcano, -log10(FDR) >1.3),
  #aes(label="*"), box.padding = 0, point.padding = 0.1, color= 'blue', size =3)
}


clustername_vs_foldchange_group(cluster_dataset, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "hightemp_vs_lowtemp")





ribosomes <- dplyr::filter(clusters, grepl("clust_6089 |clust_7529 |clust_2658 |clust_557 |clust_3360 |clust_35 |clust_3360 |clust_15652 |clust_10606 |clust_1 |clust_550 |clust_1 |clust_4298 |clust_4298 |clust_823 |clust_6713 |clust_444 |clust_22 |clust_219 |clust_4251 |clust_5230 |clust_7451 " , name))
``` 



Only using a subset of the groups to show molecular pattern for MLD poster
```{r}

clustername_vs_foldchange_group <- function(dataset, foldchange, FDR, graphtitle){
  volcano <- dataset[c("name",foldchange, FDR,"Taxa.group", "clustercompartment" )]
  x <- foldchange
  
  names(volcano) <- c( "gene", "Fold", "FDR", "taxa", "clustercompartment")
  volcano ["group"] <- "no-fold-change"
  
  if(grepl("Fe_vs_noFe_foldchange", x)) {
    volcano['size'] <- rowMeans( dataset[, c(4:21)])
  } else if (grepl("temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("temp_6C_vs_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:21)])
  } else if (grepl("int_fe_temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("int_fe_temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("noFe_3C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6,10:12 )])
  } else if (grepl("noFe_6C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6, 16:18)])
  } else if (grepl("Fe_3C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(7:9, 13:15)])
  } else if (grepl("Fe_6C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans(dataset[, c(7:9, 19:21)])
  } else if (grepl("noFe_6C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:18,10:12)])
  } else if (grepl("Fe_6C_vs_Fe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(13:15, 19:21)])
  } else if (grepl("Fe_0C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9)])
  } else if (grepl("Fe_3C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:15)])
  } else if (grepl("Fe_6C_vs_noFe_6C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:21)])
  }
volcano$shapez <-  ifelse (volcano$FDR < 0.05, "1", "16") 

  ggplot (data = volcano, aes(x= Fold, y= clustercompartment, color = taxa, label=gene, shape = shapez)) +
    geom_point (alpha=0.8, size =5)+
   
    scale_shape_manual (values = c(16, 1), guide = FALSE)+
    ylab("cluster")+ 
    xlab("log2 fold-change")+
    scale_color_manual(values = c("black", "red"))+
    scale_y_discrete(limits = compartments)+
    ggtitle(graphtitle)+
    theme_bw()+
    theme(plot.title = element_text(hjust = 0.5))+
    theme(axis.text.x = element_text(
            hjust=1,
      size = 11, 
      face = "bold", color = "black")) +
    theme(axis.text.y = element_text(
            hjust=1,
      size = 11, 
      face = "bold", color="black")) +
    geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )
    
}



cluster_dataset <- dplyr::filter(pseudo_frag_MCL, grepl("clust_332 |clust_1236 |clust_80 |clust_402 |clust_1044 |clust_3282 |clust_1194 |clust_3219 |clust_1084 |clust_3551 |clust_646  |clust_1938 |clust_2549 |clust_3551 |clust_1787 |clust_712 |clust_2256 |clust_3363 |clust_1414 |clust_498 |clust_211 |clust_42 |clust_24191 |clust_55 |clust_2763 |clust_1906 |clust_15862 |clust_5856 |clust_27282 |clust_5235 |clust_14895 |clust_86124 |clust_12649 |clust_26101 |clust_231 |clust_1548 |clust_1820 |clust_2212 |clust_1822 |clust_1668 |clust_1196 |clust_1452 |clust_534 |clust_4660 |clust_808 |clust_349 |clust_78 |clust_306 |clust_1846 |clust_129 |clust_816 |clust_3167 |clust_120 |clust_3290 |clust_9740 |clust_343 |clust_1814 |clust_1154 |clust_991 |clust_411 |clust_1816 |clust_2277 |clust_589 |clust_284 |clust_891 |clust_15657 |clust_4956 |clust_6089 |clust_7529 |clust_2658 |clust_557 |clust_3360 |clust_35 |clust_3360 |clust_15652 |clust_10606 |clust_1 |clust_550 |clust_1 |clust_4298 |clust_4298 |clust_823 |clust_6713 |clust_444 |clust_22 |clust_219 |clust_4251 |clust_5230 |clust_7451 |clust_417 |clust_9816 ", name))


cluster_dataset$cluster <- as.character(cluster_dataset$cluster)
p680 <- "P680"
mnclus <- "Manganese Cluster"
cytb6f <- "Cytochrome b6f Complex"
atpsyn <- "ATP Synthase"
rib <- "Ribosome"


compartments <- c('Ribosome','','ISIPs','','GOGAT','Ammonium Transporter', '','Rhodopsin_1','Rhodopsin_2', 'RuBisCO', 'Flavodoxin','Ferredoxin','Plastocyanin','Cytochrome c6',  'Cytochrome b6f Complex', 'LHCs')




categorymap_cluster <- c("clust_9816" = "Ferric Reductase", "clust_417" = "Ammonium Transporter", "clust_332" = "LHCs", "clust_1236" = "LHCs", "clust_80" = "LHCs", "clust_402" = "LHCs", "clust_1044" = "LHCs", "clust_3282" = "LHCs", "clust_1194" = "LHCs", "clust_3219" = "LHCs", "clust_1084"="LHCs", "clust_3551"="LHCs", "clust_646"="LHCs",  "clust_1938"="LHCs", "clust_2549" = "LHCs", "clust_3551" = "LHCs", "clust_1787" ="LHCs", "clust_712"="LHCs",  "clust_2256"="LHCs", "clust_3363"="LHCs", "clust_1414" = "Cytochrome c6", "clust_498" = "PSII", "clust_211" = "PSII", "clust_42" = "PSII", "clust_24191" = "PSII", "clust_55" = "P700", "clust_2763" = "PSII",  "clust_1906"="PSII",  "clust_15862" = "PSII",  "clust_5856"="PSII",   "clust_27282" ="PSII",  "clust_5235"="PSII",   "clust_14895"="PSII", "clust_86124" = "PSII",  "clust_12649" ="PSII",  "clust_26101" = "PSII", "clust_231" = "Rhodopsin_1", "clust_1548" = "Rhodopsin_2", "clust_1820" = "Plastocyanin", "clust_2212" = cytb6f, "clust_1822" = cytb6f, "clust_1668" = cytb6f, "clust_1196" =cytb6f, "clust_1452" = "Ferredoxin", "clust_534" = "Flavodoxin", "clust_4660" = "Flavodoxin", "clust_808" = "FNR", "clust_78" = atpsyn, "clust_349" = atpsyn, "clust_306" = atpsyn, "clust_1846" = atpsyn, "clust_129" = "RuBisCO", "clust_816" = "RuBisCO", "clust_3167" = "MetE", "clust_120" = "MetH", "clust_3290" = "CBA1", "clust_9740" = "ISIPs", "clust_343" = "ISIPs", "clust_1814" = "ISIPs", "clust_1154" = "ISIPs", "clust_991" = "Nitrate Transporter", "clust_411" = "Nitrate Reductase", "clust_1816" = "Nitrite Reductase", "clust_2277" = "Nitrite Transporter", "clust_589" = "GOGAT", "clust_284" = "GOGAT", "clust_891" = "GOGAT", "clust_15657" = "Ferritin","clust_4956" = "Nitric Oxide Dioxygenase", 
#ribosomes:
"clust_6089" = rib, "clust_7529" = rib,  "clust_2658" = rib, "clust_557" = rib, "clust_3360" = rib, "clust_35"=rib, "clust_3360" = rib, "clust_15652"=rib,  "clust_10606" = rib, "clust_1"=rib, "clust_550" =rib, "clust_1"=rib, "clust_4298" = rib,  "clust_4298" = rib,  "clust_823" = rib, "clust_6713" = rib,  "clust_444"=rib,  "clust_22"=rib,  "clust_219" = rib, "clust_4251"=rib ,"clust_5230" = rib,  "clust_7451" = rib )


cluster_dataset$clustercompartment <- categorymap_cluster[cluster_dataset$cluster]



clustername_vs_foldchange_group(cluster_dataset,"Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "Fe_vs_noFe")


clustername_vs_foldchange_group(cluster_dataset, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "hightemp_vs_lowtemp")



```


heatmaps for diagram
```{r}
#png("LHC_heatmap_.png", width=23*0.75, height=23*0.75, units="cm", res=900)

PSII_LHC <- dplyr::filter(pseudo_MCL, grepl("clust_332 |clust_1236 |clust_80 |clust_402 |clust_1044 |clust_3282 |clust_1194 |clust_3219 |clust_1084 |clust_646 |clust_1938 " , name))
heatmap_unclustered(PSII_LHC, "clust")

all_LHCs <- dplyr::filter(frag_MCL, grepl("Lhcx|Lhcf|Lhcr|Lhca|clust_332 |clust_1236 |clust_80 |clust_402 |clust_1044 |clust_3282 |clust_1194 |clust_3219 |clust_1084 |clust_3551 |clust_646 |clust_1938 |clust_2549 |clust_3551 |clust_1787 |clust_712 |clust_2256 |clust_3363 ", name))

#png("B_.png", width=30*0.75, height=23*0.75, units="cm", res=900)
heatmap_unclustered(all_LHCs, "clust")

#dev.off()



p680heatmap <- dplyr::filter(pseudo_MCL, grepl("clust_211 |clust_42 |clust_498 " , name))
heatmap_unclustered(p680heatmap, "clust")



mnclust <- dplyr::filter(pseudo_MCL, grepl("clust_2763 |clust_1906 |clust_12649 ", name))
heatmap_unclustered(mnclust, "clust")


# mnclust <- dplyr::filter(frag_MCL, grepl("clust_498 |clust_211 |clust_42 |clust_24191 |clust_2763 |clust_1906 |clust_15862 |clust_5856 |clust_27282 |clust_5235 |clust_14895 " , name))


#Rieske - clust_2212
#PetF 1822

etc <- dplyr::filter(pseudo_MCL, grepl("clust_2212 |clust_1822 |clust_1668 |clust_1196 " , name))
heatmap_unclustered(etc, "clust")

cytc6 <- dplyr::filter(pseudo_frag_MCL, grepl("clust_1414 " , name))
heatmap_unclustered(cytc6, "clust")


plastocyanin <- dplyr::filter(pseudo_MCL, grepl("clust_1820 |Plastocy" , name))
heatmap_unclustered(plastocyanin, "clust")


PSI<- dplyr::filter(pseudo_frag_ORF, grepl("clust_55 " , name))
heatmap_unclustered(PSI, "clust")


fld <- filter(pseudo_MCL, grepl("clust_534 |clust_4660 " , name))
heatmap_unclustered(fld, "clust")

fd <- filter(pseudo_MCL, grepl("clust_534 |clust_4660|clust_808  " , name))
heatmap_unclustered(fd, "clust")

atpsyn <- filter(frag_MCL, grepl("clust_78 |clust_349 |clust_306 |clust_1846 " , name))
heatmap_unclustered(atpsyn, "clust")

rubisco <- filter(frag_MCL, grepl("clust_129 |clust_816 " , name))
heatmap_unclustered(rubisco, "clust")


rhodopsin <- filter(pseudo_frag_MCL, grepl("clust_231 |clust_1548 " , name))
heatmap_unclustered(rhodopsin, "clust")

# "clust_1452" = "Ferredoxin"
# "clust_534" = "Flavodoxin", 
# "clust_4660" = "Flavodoxin",
# "clust_808" = "FNR", 
# "clust_78" = atpsyn,
# "clust_349" = atpsyn,
# "clust_306" = atpsyn,
# "clust_1846" = atpsyn,
# "clust_129" = "RuBisCO",
# "clust_816" = "RuBisCO",
#"clust_2549 |clust_3551 LHC PSI





# PSIIheatmap <- dplyr::filter(frag_MCL, grepl("clust_498 |clust_211 |clust_42 |clust_24191 |clust_2763 |clust_1906 |clust_15862 |clust_5856 |clust_27282 |clust_5235 |clust_14895 " , name))
# heatmap_unclustered(PSIIheatmap, "clust")


volcanoplot(PSII_LHC ,"Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "Fe_vs_noFe")

volcanoplot_interactive(PSII_LHC, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "hightemp_vs_lowtemp")



PSI_LHC <- dplyr::filter(pseudo_MCL, grepl("clust_2549 |clust_3551 " , name))
heatmap_unclustered(PSI_LHC, "clust")


#all of the ribosomes
 ribosomeheatmap <- dplyr::filter(pseudo_MCL, grepl("clust_6089 | clust_7529 |clust_2658 |clust_557 |clust_3360 |clust_35 |clust_10606 |clust_1 |clust_550 |clust_4298 |clust_823 |clust_444 |clust_22 |clust_219 |clust_4251 |clust_5230 |clust_2010 |ribosome|Ribosome" , name))
 
 heatmap_unclustered(ribosomeheatmap, "clust")
 
 #top7 ribosomes pseudo
 ribosome <- dplyr::filter(pseudo_MCL, grepl("clust_1 |clust_4298 |clust_6089 |clust_10606 |clust_550 |clust_3360 |clust_557 " , name))
heatmap_unclustered(ribosome, "clust")
 
 #top5 ribosomes biogenisis pseudo
 ribosomesynthesis <- dplyr::filter(pseudo_MCL, grepl("clust_14209|clust_5130 |clust_9340 |clust_1676 |clust_11454 " , name))
heatmap_unclustered(ribosomesynthesis, "clust")
 
 ribosomerecycling <- dplyr::filter(pseudo_frag_MCL, grepl("clust_35 " , name))
heatmap_unclustered(ribosomerecycling, "clust")
 
#top 5 ribosomes frag
ribosome <- dplyr::filter(frag_MCL, grepl("clust_1 |clust_4298 |clust_2658 |clust_3360 |clust_7529 |clust_823 |clust_219 |clust_35 " , name))
heatmap_unclustered(ribosome, "clust")

#top 5 ribosome biogensis frag
  ribosomesynthesis <- dplyr::filter(frag_MCL, grepl("clust_8337 |clust_5130 |clust_5745 |clust_5230 |clust_11454 " , name))
 heatmap_unclustered(ribosomesynthesis, "clust")



 



#ISIP1:"1814"
 #ISIP2A: 9740
# ISIP3: clust_1154
#Ferritin: clust 15657
#Ferric reductase: clust_9816


isip1heatmap <-dplyr::filter(pseudo_MCL, grepl("clust_1814 |clust_1154 |clust_9740 |clust_9816 |clust_15657 ", name))
heatmap_unclustered(isip1heatmap, "clust")



# "clust_991" = "Nitrate Transporter", 
# "clust_411" = "Nitrate Reductase",
# "clust_1816" = "Nitrite Reductase", 
# "clust_2277" = "Nitrite Transporter",
# "clust_589" = "GOGAT", 
# "clust_284" = "GOGAT", 
# "clust_891" = "GOGAT"

# Glutamine_synthetase 589 and 284

# Glutamate_synthase clust_891

nitrate <- dplyr::filter(pseudo_MCL, grepl("clust_991 |clust_411 ", name))
  heatmap_unclustered(nitrate, "clust")

nitrite <- dplyr::filter(pseudo_MCL, grepl("clust_1816 |clust_2277 ", name))
  heatmap_unclustered(nitrite, "clust")
 


gogat <- dplyr::filter(pseudo_MCL, grepl("clust_589 |clust_284 |clust_891 ", name))
  heatmap_unclustered(gogat, "clust")

  
ammontransport <- dplyr::filter(pseudo_frag_MCL, grepl("clust_417 ", name))
  heatmap_unclustered(ammontransport , "clust")
  
   volcanoplot_interactive(gogat ,"Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", "Fe_vs_noFe")

volcanoplot_interactive(PSII_LHC, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "hightemp_vs_lowtemp") 
  
  
  

# "clust_3167" = "MetE", 
# "clust_120" = "MetH", 
# "clust_3290" = "CBA1"

B12 <- dplyr::filter(pseudo_MCL, grepl("clust_3167 |clust_120 |clust_3290 ", name))
  heatmap_unclustered(B12, "clust")
 


```




Domoic Acid
```{r}
DABC <- dplyr::filter (all_ORF_sub, grepl('contig_254441_113_1024_+|contig_1174518_67_906_+|contig_732674_1_798_-|contig_633551_118_1041_+|contig_627399_1_787_-|contig_1862_1_732_-|contig_660976_1_1043_+|contig_253357_26_973_+|contig_827413_1_313_-' ,orf_id))


volcanoplot_interactive(DABC,"temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_DABC")
volcanoplot_interactive(DABC,"Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_DABC")

#DabD |clust_12542 |clust_7510 |clust_2795 |P450 |clust_2216 

DABD <- dplyr::filter(all_ORF_sub, grepl('contig_596040_24_938_-|contig_625510_955_1881_-|contig_625510_3536_4051_+|contig_82244_85_1023_+|contig_596040_947_1648_-|contig_82244_1395_1877_+|contig_680152_23_630_-|contig_1109498_1_752_-|contig_1147495_1_1248_+|contig_625510_1971_2528_-|contig_795490_14_403_-|contig_602505_1_924_+|contig_644544_26_931_+|contig_1338988_954_1898_-|contig_631357_332_904_-|contig_895090_1_705_-|contig_584212_1095_1886_+|contig_708803_1_573_-|contig_1174000_279_929_-' ,orf_id))

DABCD<- dplyr::filter(all_ORF_sub,grepl('contig_254441_113_1024_+|contig_1174518_67_906_+|contig_732674_1_798_-|contig_633551_118_1041_+|contig_627399_1_787_-|contig_1862_1_732_-|contig_660976_1_1043_+|contig_253357_26_973_+|contig_827413_1_313_-|contig_596040_24_938_-|contig_625510_955_1881_-|contig_625510_3536_4051_+|contig_82244_85_1023_+|contig_596040_947_1648_-|contig_82244_1395_1877_+|contig_680152_23_630_-|contig_1109498_1_752_-|contig_1147495_1_1248_+|contig_625510_1971_2528_-|contig_795490_14_403_-|contig_602505_1_924_+|contig_644544_26_931_+|contig_1338988_954_1898_-|contig_631357_332_904_-|contig_895090_1_705_-|contig_584212_1095_1886_+|contig_708803_1_573_-|contig_1174000_279_929_-' ,orf_id))

#DABD_clust <- dplyr::filter(pseudo_frag_ORF, grepl('clust_12542 |clust_2795 ' ,name))

volcanoplot_interactive(DABD_clust,"temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_DABD")
volcanoplot_interactive(DABD_clust,"Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_DABD")
volcanoplot_interactive(DABD,"Fe_6C_vs_noFe_6C_foldchange", "Fe_6C_vs_noFe_6C_pvalue", " Fe_vs_NoFe_6C_DABD")




all_diatom_fasta <- read.csv ("all_diatom_fasta.csv")
x <- DABCD$name
y <- gsub("(.*);.*", "\\1", x)
DABCD$shortname <- y

DABCD_FASTA <- dplyr::filter(all_diatom_fasta, grepl(paste(DABCD$orf_id,collapse = "|"), orf_id))

write.fasta (sequences = (as.list(DABCD_FASTA$AA_sequence)), names = DABCD_FASTA$orf_id ,file.out = "DAB-CD.fasta")


```
















Rhodopsin plots
```{r}
#png("venndiagram_alldiatoms_clusters_iron.png", width=23*0.75, height=23*0.75, units="cm", res=900)
#dev.off()
rhodopsin_ORF <- dplyr::filter (pseudo_frag_ORF, grepl('clust_231 |rhodopsin|Rhodopsin', name))

rhodopsin_MCL <- dplyr::filter (pseudo_frag_MCL, grepl('rhodopsin|Rhodopsin|clust_10297 |clust_129508 |clust_130846 |clust_142867 |clust_1508 |clust_1548 |clust_16487 |clust_19837 |clust_22740 |clust_231 |clust_28522 |clust_3959 |clust_4014 |clust_779029 |clust_8237 ', name))



rhodopsin_ORF_clust1548 <- dplyr::filter (pseudo_frag_ORF, grepl('clust_1548 ', name))

rhodopsin_ORF_clust231 <- dplyr::filter (pseudo_frag_ORF, grepl('clust_231 ', name))
volcanoplot_interactive(rhodopsin_ORF, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms")



volcanoplot_interactive(rhodopsin_MCL, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms")

volcanoplot_interactive(rhodopsin_MCL, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_ORFs_alldiatoms")


volcanoplot(rhodopsin_MCL, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms")

volcanoplot(rhodopsin_MCL, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_ MCL_alldiatoms")


clustername_vs_foldchange( rhodopsin_MCL, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_clusters_alldiatoms")

clustername_vs_foldchange( rhodopsin_MCL, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_MCL_alldiatoms")

clustername_vs_foldchange( rhodopsin_MCL, "int_fe_temp_3C_vs_0C_foldchange", "int_fe_temp_3C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_MCL_alldiatoms")





heatmap(rhodopsin_MCL, 'clust')
heatmap_mean(rhodopsin_MCL, 'clust')

heatmap(rhodopsin_MCL, "clust_231|clust_1548|clust_28522 |clust_8237 |clust_4014 |clust_3959 |clust_10297 |clust_19837 ")
heatmap_mean(rhodopsin_MCL, "clust_231|clust_1548|clust_28522 |clust_8237 |clust_4014 |clust_3959 |clust_10297 |clust_19837 " )

heatmap_unclustered_mean(rhodopsin_MCL, "clust_231|clust_1548 |clust_28522")
heatmap_unclustered(rhodopsin_MCL, "clust_231|clust_1548 |clust_28522")

heatmap(rhodopsin_MCL, "clust_231|clust_1548 |clust_28522")
heatmap_unclustered(rhodopsin_MCL, "clust_231|clust_1548 |clust_28522")

heatmap(all_diatoms_MCL, "clust_231 |clust_1548 ")
heatmap_unclustered_mean(rhodopsin_MCL, "clust_231|clust_1548 ")

MCLclusterrawplot (frag_MCL, '^clust_231$', "Rhodopsin - cluster 231 Frag")


heatmap(pseudo_frag_MCL, "clust_1548 |clust_231 ")



MCLclusterrawplot (frag_MCL, '^clust_231$', "Rhodopsin - cluster 1548 Frag")
MCLclusterrawplot (frag_MCL, '^clust_10297$', "Rhodopsin - cluster 10297 Frag")
MCLclusterrawplot (frag_MCL, '^clust_$', "Rhodopsin - cluster 231 Frag")
 

rhodopsin_interest <- dplyr::filter (all_MCL_sub, grepl('Pseudo-nit|Fragilariopsis|Rhodobacterales|Dinophy|Alteromona|Cyanobacter', Taxa.group))
heatmap(rhodopsin_interest, 'clust_231 ')
heatmap_unclustered(rhodopsin_interest, 'clust_231 ')


R1p <- MCLclusterrawplot (Frag_MCL, '^clust_231$', "Rhodopsin - cluster 231 Pseudo")
R2f <- MCLclusterrawplot (FragMCL, '^clust_1548$', "Rhodopsin - cluster 1548 Frag")
R2p <- MCLclusterrawplot (PseudoMCL, '^clust_1548$', "Rhodopsin - cluster 1548 Pseudo")
R3f <- MCLclusterrawplot (FragMCL, '^clust_10297$', "Rhodopsin - cluster 10297 Frag")
R3p <- MCLclusterrawplot (PseudoMCL, '^clust_10297$', "Rhodopsin - cluster 10297 Pseudo")
R4f <- MCLclusterrawplot (FragMCL, '^clust_19837$', "Rhodopsin - cluster 19837 Frag")
R4p <- MCLclusterrawplot (PseudoMCL, '^clust_19837$', "Rhodopsin - cluster 19837 Pseudo")         

Lclusterrawplot (frag_MCL, '^clust_1548$', "Rhodopsin - cluster 231 Frag")

grid.arrange(R1f, R1p, R2f, R2p, R3f, R3p, nrow=3, ncol=2)
```


OEC/Mn cluster and rhodopsin
```{r}

OEC_ORF <- dplyr::filter (pseudo_frag_ORF, grepl('evolving|OEE', name))

OEC_MCL <- dplyr::filter (pseudo_frag_MCL, grepl('clust_1906 ', name))

OEC_MCL <- dplyr::filter (pseudo_frag_MCL, grepl('cluster', name))

heatmap_unclustered_mean(OEC_MCL, 'clust')


volcanoplot_interactive(OEC_ORF, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms")

volcanoplot_interactive(OEC_ORF, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_ORFs_alldiatoms")

volcanoplot_interactive(OEC_ORF, "int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_ORFs_alldiatoms")




clustername_vs_foldchange(OEC_MCL, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_clusters_alldiatoms")


MCLclusterrawplot(OEC_MCL, "clust_5856 ", "jdjd")


heatmap_unclustered(all_MCL_sub, 'clust_1906 ')

heatmap(OEC_MCL, 'clust')

heatmap_mean(rhodopsin_MCL, 'clust')




```


rhodopsin/Mn cluster interaction, plastoquinone
```{r}


PQ <- dplyr::filter (all_diatoms_MCL, grepl('clust_231 |clust_1548 |clust_1906 |evolv', name))

clustername_vs_foldchange(PQ,"Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms" )
ggplotly(x)

clustername_vs_foldchange(PQ, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_ MCL_alldiatoms")


clustername_vs_foldchange(PQ, "int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_MCL_alldiatoms")



volcanoplot_interactive(PQ, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms")

volcanoplot_interactive(PQ, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_ MCL_alldiatoms")

volcanoplot(PQ, "int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_MCL_alldiatoms")



```




 ISIP plots 
```{r}
ISIP_cluster <- dplyr::filter (all_diatoms_MCL, grepl("ISIP|isip|clust_1814 |clust_343 |clust_9740 |clust_25598 |clust_23469 |clust_1154  ", name))

ISIP_ORF <- dplyr::filter (all_diatoms_ORF, grepl('ISIP|isip', name))




volcanoplot_interactive(ISIP_cluster, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms")

clustername_vs_foldchange(ISIP_ORF, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms")

volcanoplot_interactive(ISIP_cluster, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_ORFs_alldiatoms")


```

Nitrogen 
```{r}
NiR <- MCLclusterrawplot(frag_MCL, '^clust_411$', "Nitrate reductase - cluster 411Frag")
NiRP <- MCLclusterrawplot(PseudoMCL, '^clust_411$', "Nitrate reductase - cluster 411Pseudo")
NitTrans <- MCLclusterrawplot(FragMCL, '^clust_991$', "Nitrate transporter portein - cluster 991Frag")
NitTransP <- MCLclusterrawplot(PseudoMCL, '^clust_991$', "Nitrate transporter portein - cluster 991Pseudo")
NR <- MCLclusterrawplot(FragMCL, '^clust_3563$', "Nitrite reductase - cluster 3563Frag")
NRP <- MCLclusterrawplot(PseudoMCL, '^clust_3563$', "Nitrite reductase - cluster 3563Pseudo")

heatmap(group, FragORF, "ISIP")



#Urease <- MCLclusterrawplot(FragMCL, '^clust_927$', "Urease - cluster 927Frag")
#UreaseP <- MCLclusterrawplot(PseudoMCL, '^clust_927$', "Urease - cluster 927Pseudo")
#URED <- MCLclusterrawplot(FragMCL, '^clust_20535$', "Urease Accessory Protein UreD - cluster 20535Frag")
#UREDP <- MCLclusterrawplot(PseudoMCL, '^clust_20535$', "Urease Accessory Protein UreD - cluster 20535Pseudo")
#grid.arrange(NiR, NiRP, NitTrans, NitTransP, NR,NRP, Urease, UreaseP, URED,UREDP,  nrow=6, ncol=2)

grid.arrange(NiR, NiRP, NitTrans, NitTransP, NR,NRP,  nrow=3, ncol=2)

ISIP <- dplyr::filter (FragORF, grepl('nitrite|Nitrite|nitrate|Nitrate', name))
volcanoplotinteractive(ISIP, "X12_vs_14.log2fc", "X12_vs_14.fdr", "Temp Test: 0C_VS_3C")
volcanoplotinteractive(ISIP, "X12_vs_16.log2fc", "X12_vs_16.fdr", "Temp Test: 0C_VS_6C")
volcanoplotinteractive(ISIP, "X14_vs_16.log2fc", "X14_vs_16.fdr", "Temp Test: 3C_VS_6C")
volcanoplotinteractive(ISIP, "X12_vs_13.log2fc", "X12_vs_13.fdr", "Fe Test: 0C_VS_Fe0C")
volcanoplotinteractive(ISIP, "X14_vs_15.log2fc", "X14_vs_15.fdr", "Fe Test: 3C_VS_Fe3C")
volcanoplotinteractive(ISIP, "X16_vs_17.log2fc", "X16_vs_17.fdr", "Fe Test: 6C_VS_Fe6C")
volcanoplotinteractive(ISIP, "X12_vs_15.log2fc", "X12_vs_15.fdr", "Synergy Test: 0C_VS_Fe3C")
volcanoplotinteractive(ISIP, "X12_vs_17.log2fc", "X12_vs_17.fdr", "Synergy Test: 0C_VS_Fe6C")



GS<- MCLclusterrawplot(FragMCL, '^clust_589$', "Glutamine Synthetase - cluster 589Frag")
GSP<- MCLclusterrawplot(PseudoMCL, '^clust_589$', "Glutamine Synthetase - cluster 589Pseudo")
GaS <- MCLclusterrawplot(FragMCL, '^clust_241$', "Glutamate Synthase - cluster 241Frag")

GaSP <- MCLclusterrawplot(PseudoMCL, '^clust_241$', "Glutamate Synthase - cluster 241Pseudo")
grid.arrange(GS,GSP, GaS, GaSP, nrow=2, ncol=2)

MCLclusterrawplot(MCL, '^clust_999$', "Nitrite reductase - cluster 3563")

LHC <- dplyr::filter (ORF, grepl('transaminase', name))

NiRMCL <- dplyr::filter(FragORF, grepl ('ferritin|Ferritin', name))


MCLclusterrawplot (PseudoMCL, '^clust_4956$', "Nitric Oxide Synthase -cluster 4956 Pseudo")

MCLclusterrawplot (frag_MCL, '^clust_4956$', "Nitric Oxide Synthase - cluster 4956 Frag")
MCLclusterrawplot (frag_MCL, '^clust_16690$', "Nitric Oxide Synthase Frag")

MCLclusterrawplot (PseudoMCL, '^clust_15657$', "Nitric Oxide Synthase Frag")


#Nitric oxide and iron in plants:an emerging and converging story. Interaction with ferritin ? 

```

Flavodoxin Frag
```{r}
flavodoxin <- dplyr::filter(all_diatoms_MCL, grepl ('flavodoxin|Flavodoxin', name))

flavodoxin_frag <- dplyr::filter(frag_MCL, grepl ('ferredoxin|Ferredoxin', name))
flavodoxin_pseudo <- dplyr::filter(pseudo_MCL, grepl ('flavodoxin|Flavodoxin', name))







clustername_vs_foldchange(flavodoxin, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_flavodoxin")

clustername_vs_foldchange( flavodoxin, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "6C_vs_0C_flavodoxin")

heatmap_unclustered( flavodoxin,"clust_")

heatmap_mean( flavodoxin_pseudo,"clust_")







MCLclusterrawplot (FragMCL, '^clust_534$', " Frag-Flavodoxin- cluster534")

x <- ORFplots(FragORF, "^clust_534$", "title")
ggplotly(x)

contigplots(FragORF,"contig_1360422_94_705+", "Frag-Flavodoxin-cluster534-contig_1175010_45_704_+")

 contigplots(FragORF,"644824_210_908+", "Frag-Flavodoxin-cluster534-contig_1360422_94_705_+")


volcanoplotinteractive(flavodoxin, "X12_vs_14.log2fc", "X12_vs_14.fdr", "Temp Test: 0C_VS_3C")
volcanoplotinteractive(flavodoxin, "X12_vs_16.log2fc", "X12_vs_16.fdr", "Temp Test: 0C_VS_6C")
volcanoplotinteractive(flavodoxin, "X14_vs_16.log2fc", "X14_vs_16.fdr", "Temp Test: 3C_VS_6C")
volcanoplotinteractive(flavodoxin, "X12_vs_13.log2fc", "X12_vs_13.fdr", "Fe Test: 0C_VS_Fe0C")
volcanoplotinteractive(flavodoxin, "X14_vs_15.log2fc", "X14_vs_15.fdr", "Fe Test: 3C_VS_Fe3C")
volcanoplotinteractive(flavodoxin, "X16_vs_17.log2fc", "X16_vs_17.fdr", "Fe Test: 6C_VS_Fe6C")
volcanoplotinteractive(flavodoxin, "X12_vs_15.log2fc", "X12_vs_15.fdr", "Synergy Test: 0C_VS_Fe3C")
volcanoplotinteractive(flavodoxin, "X12_vs_17.log2fc", "X12_vs_17.fdr", "Synergy Test: 0C_VS_Fe6C")

volcanoplot(flavodoxin, "X12_vs_17.log2fc", "X12_vs_17.fdr", "Synergy Test: 0C_VS_Fe6C")

MCLclusterrawplot (FragMCL, '^clust_534$', "Flavodoxin cluster 534. Frag")
MCLclusterrawplot (FragMCL, '^clust_4660$', "Iron responsive Flavodoxin in F. cylindrus")
MCLclusterrawplot (FragMCL, '^clust_6705$', "Iron responsive Flavodoxin in F. cylindrus")
flavodoxin <- dplyr::filter(FragORF, grepl ('flavodoxin|Flavodoxin', name))

MCLclusterrawplot (FragMCL, '^clust_1452$', "Flavodoxin cluster 534. Frag")
MCLclusterrawplot (FragMCL, '^clust_1816$', "Flavodoxin cluster 534. Frag")
MCLclusterrawplot (FragMCL, '^clust_3563$', "Flavodoxin cluster 534. Frag")

ORFplots(FragORF, "^clust_534$", "title")
ORFplots(FragORF, "^clust_4660$", "title")
ORFplots(FragORF, "^clust_6705", "title")


```


List of significant MCL clusters Extract only Significant genes 
```{r}
volcano <- FragMCL[c("name", "X12_vs_14.log2fc", "X12_vs_14.fdr")]
names(volcano) <- c("gene", "Fold", "FDR")

volcano <- volcano [volcano$FDR <0.1, ]



LHC <- dplyr::filter (ORF, grepl('nitrite', name))
#LHC <- dplyr::filter (MCL, grepl('Photosystem|photosystem', name))


volcanoplotinteractive(LHC, "X12_vs_14.log2fc", "X12_vs_14.fdr", "Temp Test: 0C_VS_3C")    
volcanoplotinteractive(LHC, "X12_vs_16.log2fc", "X12_vs_16.fdr", "Temp Test: 0C_VS_6C")     
volcanoplotinteractive(LHC, "X14_vs_16.log2fc", "X14_vs_16.fdr", "Temp Test: 3C_VS_6C")
volcanoplotinteractive(LHC, "X12_vs_13.log2fc", "X12_vs_13.fdr", "Fe Test: 0C_VS_Fe0C")
volcanoplotinteractive(LHC, "X14_vs_15.log2fc", "X14_vs_15.fdr", "Fe Test: 3C_VS_Fe3C")
volcanoplotinteractive(LHC, "X16_vs_17.log2fc", "X16_vs_17.fdr", "Fe Test: 6C_VS_Fe6C")
volcanoplotinteractive(FragMCL, "X12_vs_15.log2fc", "X12_vs_15.fdr", "Synergy Test: 0C_VS_Fe3C")
volcanoplotinteractive(LHC, "X12_vs_17.log2fc", "X12_vs_17.fdr", "Synergy Test: 0C_VS_Fe6C")

```



Light Harvesting Complexes
```{r}
MCLclusterrawplot(frag_MCL,'^clust_1236$', "LHCf1 - cluster 1236Frag")
F1P <- MCLclusterrawplot(pseudo_MCL,'^clust_1236$', "LHCf1 - cluster 1236Pseudo")

F2 <- MCLclusterrawplot(frag_MCL, '^clust_1194$', "LHCf4 - cluster 1194Frag")
F2P <- MCLclusterrawplot(pseudo_MCL, '^clust_1194$', "LHCf4 - cluster 1194Pseudo")

grid.arrange(F1, F1P, F2, F2P,nrow=2, ncol=2)

X1 <- MCLclusterrawplot('^clust_712$', "LHCx1 - cluster 712")
X2 <- MCLclusterrawplot('^clust_960$', "LHCx5 - cluster 960")
X3 <- MCLclusterrawplot('^clust_1787$', "LHCx6 - cluster 1787")
grid.arrange(X1, X2, X3, nrow=2, ncol=2)

MCLclusterplot('clust_1550 ', "PSII Cytb559 - cluster 1550")

MCLclusterplot('clust_38070 ', "Chl A/B binding - 38070")
MCLclusterplot('clust_646 ', "Chl A/B binding - 646")
MCLclusterplot('clust_195 ', "Chl A/B binding - 195")
MCLclusterplot('clust_534 ', "Flavodoxin - 534")



A1 <- MCLclusterrawplot(FragMCL, '^clust_6705$', "Flavodoxin - cluster 6705 Frag")

A2 <- MCLclusterrawplot(FragMCL, '^clust_534$', "Flavodoxin 1 - cluster 534 Frag")

A3 <- MCLclusterrawplot(FragMCL, '^clust_4907$', "Flavodoxin - cluster 4907 Frag")
A4 <- MCLclusterrawplot(FragMCL, '^clust_4660$', "Flavodoxin - cluster 4660 Frag")
A5 <- MCLclusterrawplot(FragMCL, '^clust_33174$', "Flavodoxin-like fold - cluster 33174 Frag")
A6 <- MCLclusterrawplot(FragMCL, '^clust_12773$', "Flavodoxin - cluster 12773 Frag")



A1 <- MCLclusterrawplot(PseudoMCL, '^clust_6705$', "Flavodoxin - cluster 6705 Pseudo")

A2 <- MCLclusterrawplot(PseudoMCL, '^clust_534$', "Flavodoxin 1 - cluster 534 Pseudo")

A3 <- MCLclusterrawplot(PseudoMCL, '^clust_4907$', "Flavodoxin - cluster 4907 Pseudo")
A4 <- MCLclusterrawplot(PseudoMCL, '^clust_4660$', "Flavodoxin - cluster 4660 Pseudo")
A5 <- MCLclusterrawplot(PseudoMCL, '^clust_33174$', "Flavodoxin-like fold - cluster 33174 Pseudo")
A6 <- MCLclusterrawplot(PseudoMCL, '^clust_12773$', "Flavodoxin - cluster 12773 Pseudo")



grid.arrange(A1, A2, A3, A4, A5, A6, nrow=3, ncol=2)

```






ORF volcano for core antenna
```{r}
LHC <- dplyr::filter (PseudoORF, grepl('rhod|Rhod', name))
#LHC <- dplyr::filter (MCL, grepl('Photosystem|photosystem', name))


volcanoplotinteractive(LHC, "X12_vs_14.log2fc", "X12_vs_14.fdr", "Temp Test: 0C_VS_3C")    
volcanoplotinteractive(LHC, "X12_vs_16.log2fc", "X12_vs_16.fdr", "Temp Test: 0C_VS_6C")     
volcanoplotinteractive(LHC, "X14_vs_16.log2fc", "X14_vs_16.fdr", "Temp Test: 3C_VS_6C")
volcanoplotinteractive(LHC, "X12_vs_13.log2fc", "X12_vs_13.fdr", "Fe Test: 0C_VS_Fe0C")
volcanoplotinteractive(LHC, "X14_vs_15.log2fc", "X14_vs_15.fdr", "Fe Test: 3C_VS_Fe3C")
volcanoplotinteractive(LHC, "X16_vs_17.log2fc", "X16_vs_17.fdr", "Fe Test: 6C_VS_Fe6C")
volcanoplotinteractive(LHC, "X12_vs_15.log2fc", "X12_vs_15.fdr", "Synergy Test: 0C_VS_Fe3C")
volcanoplotinteractive(LHC, "X12_vs_17.log2fc", "X12_vs_17.fdr", "Synergy Test: 0C_VS_Fe6C")
```




All of ISIP2A in the ORF
```{r}
ISIP <- dplyr::filter (ORF, grepl('clust_7576', name))
ISIP1 <- ISIP [-c(1:46,50:64, 83:106)]
ISIP1 <- ISIP1 [c(22, 1:21)]
names(ISIP1)<- c("name", "A_T0", "B_T0", "C_T0", "A_0.5", "B_0.5", "C_0.5", "A_Fe_0.5", "B_Fe_0.5", "C_Fe_0.5", "A_3", "B_3", "C_3", "A_Fe_3", "B_Fe_3", "C_Fe_3","A_6", "B_6", "C_6", "A_Fe_6", "B_Fe_6", "C_Fe_6")

ISIP1  <- dplyr::filter (ISIP1, grepl('_75', name))

#this loops takes the means of the triplicate treatments
colname <- "mean"
i=0
while (i < 7){
colname <- paste(colname, i)
col1 = 2+(i*3)
col2 = 4+(i*3)
ISIP1[[colname]] <- rowMeans( ISIP1[, c(col1:col2)] )
i <- i+1
}

ISIP1<- ISIP1 [-c(2:23)]

names(ISIP1) <- c("name","NoFe_0.5", "Fe_0.5", "NoFe_3", "Fe_3", "NoFe_6", "Fe_6" )

ISIP1 <-  melt (data = ISIP1,id.vars = "name", measure.vars = c(2:7))
ISIP1 <- separate(ISIP1, col = variable, into = c("Fe", "temperature"))
ggplot(ISIP1, aes(x=temperature, y=value, colour = name, shape = Fe ))+
#geom_point(size = 2)+
#stat_summary(fun.y = mean, geom = "line", colour = "black", group =1, size = 1.25)+
#stat_summary(fun.data =  mean_se, geom = "errorbar", colour = "black")+
xlab ("Temperature (C)")+
ylab ("Expression-Value")+
ggtitle("ISIP2A")+
theme_bw()+
theme(plot.title = element_text(hjust = 0.5))+
theme(axis.text.x = element_text(angle = 50, vjust = 1.0, hjust = 1.0))+
guides(colour=FALSE)+
#scale_color_manual(name = "Treatment", labels = c("+Fe", "-Fe"), values = c("triangle", "square")) +
geom_jitter(width = 0.11, size =2)+
labs (shape = "Treatment")
#scale_y_log10()
#geom_line(aes(group=name))
```




ORF volcano plot for Frag ISIPs   
```{r echo=FALSE, message=FALSE, warning=FALSE}
ISIP <- dplyr::filter (ORF, grepl('ISIP|isip', name))

#nitratereductase <- dplyr::filter (ORF, grepl('contig_580799_1_929_-|contig_1148680_114_1091_+|contig_1148680_1151_2203_+|contig_1155130_1_1036_+|contig_667814_1_947_-', orf_id))

#volcanoplotinteractive <- function(Dataset, Fold, FDR, volcanontitle)
volcanoplotinteractive(ISIP, "X12_vs_14.log2fc", "X12_vs_14.fdr", "Temp Test: 0C_VS_3C")    
volcanoplotinteractive(ISIP, "X12_vs_16.log2fc", "X12_vs_16.fdr", "Temp Test: 0C_VS_6C")     
volcanoplotinteractive(ISIP, "X14_vs_16.log2fc", "X14_vs_16.fdr", "Temp Test: 3C_VS_6C")
volcanoplotinteractive(ISIP, "X12_vs_13.log2fc", "X12_vs_13.fdr", "Fe Test: 0C_VS_Fe0C")
volcanoplotinteractive(ISIP, "X14_vs_15.log2fc", "X14_vs_15.fdr", "Fe Test: 3C_VS_Fe3C")
volcanoplotinteractive(ISIP, "X16_vs_17.log2fc", "X16_vs_17.fdr", "Fe Test: 6C_VS_Fe6C")
volcanoplotinteractive(ISIP, "X12_vs_15.log2fc", "X12_vs_15.fdr", "Synergy Test: 0C_VS_Fe3C")
volcanoplotinteractive(ISIP, "X12_vs_17.log2fc", "X12_vs_17.fdr", "Synergy Test: 0C_VS_Fe6C")
```


MCL volcano including all of the clusters (annotated or not) 
```{r}
#volcanoplotinteractive <- function(Dataset, Fold, FDR, volcanontitle)
FragMCL <- dplyr::filter (PseudoMCL, grepl('clust_1814|clust_1154|clust_9740', name))


volcanoplotinteractive(FragMCL, "X12_vs_14.log2fc", "X12_vs_14.fdr", "Temp Test: 0C_VS_3C")
volcanoplotinteractive(FragMCL, "X12_vs_16.log2fc", "X12_vs_16.fdr", "Temp Test: 0C_VS_6C")
volcanoplotinteractive(FragMCL, "X14_vs_16.log2fc", "X14_vs_16.fdr", "Temp Test: 3C_VS_6C")
volcanoplotinteractive(FragMCL, "X12_vs_13.log2fc", "X12_vs_13.fdr", "Fe Test: 0C_VS_Fe0C")
volcanoplotinteractive(FragMCL, "X14_vs_15.log2fc", "X14_vs_15.fdr", "Fe Test: 3C_VS_Fe3C")
volcanoplotinteractive(FragMCL, "X16_vs_17.log2fc", "X16_vs_17.fdr", "Fe Test: 6C_VS_Fe6C")
volcanoplotinteractive(FragMCL, "X12_vs_15.log2fc", "X12_vs_15.fdr", "Synergy Test: 0C_VS_Fe3C")
volcanoplotinteractive(FragMCL, "X12_vs_17.log2fc", "X12_vs_17.fdr", "Synergy Test: 0C_VS_Fe6C")
```
 

ORF volcano for LHC
```{r}
#LHC <- dplyr::filter (ORF, grepl('LHC|lhc|Lhc', name))


MCLclusterrawplot(FragMCL,'^clust_746$', "Nickel SOD - cluster 746 Frag")
MCLclusterrawplot(PseudoMCL,'^clust_746$', "Nickel SOD - cluster 746 Pseudo-Nit")

MCLclusterrawplot(FragMCL,'^clust_1500$', "Iron/Manganese SOD - cluster 1500Frag")
MCLclusterrawplot(PseudoMCL,'^clust_1500$', "Iron/Manganese SOD - cluster 1500Frag")


MCLclusterrawplot(FragMCL,'^clust_2501$', "Iron/Manganese SOD - cluster 2501Frag")
MCLclusterrawplot(FragMCL,'^clust_366$', "Iron/Manganese SOD - cluster 2501Frag")


LHC <- dplyr::filter (FragORF, grepl('clust_1820 ', name))
      
MCLclusterrawplot(LHC,'^contig_1019282_1_425_+', "PCY")


 MCLclusterrawplot(PseudoMCL,'^clust_1820$', "PCY")
       
                  


LHC <- dplyr::filter (PseudoORF, grepl('clust_1414 ', name))




volcanoplotinteractive(LHC, "X12_vs_14.log2fc", "X12_vs_14.fdr", "Temp Test: 0C_VS_3C")    
volcanoplotinteractive(LHC, "X12_vs_16.log2fc", "X12_vs_16.fdr", "Temp Test: 0C_VS_6C")     
volcanoplotinteractive(LHC, "X14_vs_16.log2fc", "X14_vs_16.fdr", "Temp Test: 3C_VS_6C")
volcanoplotinteractive(LHC, "X12_vs_13.log2fc", "X12_vs_13.fdr", "Fe Test: 0C_VS_Fe0C")
volcanoplotinteractive(LHC, "X14_vs_15.log2fc", "X14_vs_15.fdr", "Fe Test: 3C_VS_Fe3C")
volcanoplotinteractive(LHC, "X16_vs_17.log2fc", "X16_vs_17.fdr", "Fe Test: 6C_VS_Fe6C")
volcanoplotinteractive(LHC, "X12_vs_15.log2fc", "X12_vs_15.fdr", "Synergy Test: 0C_VS_Fe3C")
volcanoplotinteractive(LHC, "X12_vs_17.log2fc", "X12_vs_17.fdr", "Synergy Test: 0C_VS_Fe6C_Frag")
```



```{r}
ORFclusterrawplot <- function(whichdata, whichprotein, graphtitle){
  NiRMCL <- dplyr::filter(whichdata, grepl (whichprotein, name))
NiRMCL <- NiRMCL [c(107, 1:106)]
  NiRMCL <- NiRMCL [-c(2:65, 84:107)]
  names(NiRMCL)<- c("name", "A_NoFe_0.5", "B_NoFe_0.5", "C_NoFe_0.5", "A_Fe_0.5", "B_Fe_0.5", "C_Fe_0.5", "A_NoFe_3", "B_NoFe_3", "C_NoFe_3", "A_Fe_3", "B_Fe_3", "C_Fe_3","A_NoFe_6", "B_NoFe_6", "C_NoFe_6", "A_Fe_6", "B_Fe_6", "C_Fe_6")
  NiRMCL <-  melt (data = NiRMCL,id.vars = "name")
  NiRMCL <- separate(NiRMCL, col = variable, into = c("Rep", "Fe", "temperature"))
  
  ggplot(NiRMCL, aes(x=temperature, y=value, colour = Fe, shape = Rep ))+
    geom_point(size = 3)+
    scale_color_manual(values = c("red", "black"))+
    xlab ("Temperature (C)")+
    ylab ("")+
    ggtitle(graphtitle)+
    theme_bw()+
    theme(plot.title = element_text(hjust = 0.5))+
    theme(axis.text.x = element_text(angle = 50, vjust = 1.0, hjust = 1.0))+
    #scale_color_manual (name = "", labels = c("+Fe", "-Fe"), values = c("red", "black"))
       guides (color = FALSE, shape = FALSE) 

}


NiRMCL <- dplyr::filter(FragORF, grepl ('contig_1158537_1_423_+', name))


##Cytochrome c6
ORFclusterrawplot(FragORF,'contig_709325_277_602_-', "CytC6") 

ORFclusterrawplot(FragORF,'contig_767007_35_424_+', "CytC6") 


a = ORFclusterrawplot(FragORF,'contig_1175010_45_704_+', 'contig_1175010_45_704_+') 
b = ORFclusterrawplot(FragORF,'contig_1294699_1206_1370_+', 'contig_1294699_1206_1370_+') 
c = ORFclusterrawplot(FragORF,'contig_1294699_769_1095_+', 'contig_1294699_769_1095_+') 
d = ORFclusterrawplot(FragORF,'contig_1315586_1_203_+', 'contig_1315586_1_203_+') 
e = ORFclusterrawplot(FragORF,'contig_1360422_94_705_+', 'contig_1360422_94_705_+') 
f=ORFclusterrawplot(FragORF,'contig_1383095_1_286_+', 'contig_1383095_1_286_+') 
g=ORFclusterrawplot(FragORF,'contig_269925_1_531_-', 'contig_269925_1_531_-') 
h=ORFclusterrawplot(FragORF,'contig_269925_613_777_+', 'contig_269925_613_777_+') 
i=ORFclusterrawplot(FragORF,'contig_301915_1_119_-', 'contig_301915_1_119_-') 
j=ORFclusterrawplot(FragORF,'contig_416319_1_108_+', 'contig_416319_1_108_+') 
k=ORFclusterrawplot(FragORF,'contig_619845_137_895_-', 'contig_619845_137_895_-') 
l=ORFclusterrawplot(FragORF,'contig_691588_22_636_+', 'contig_691588_22_636_+')
m=ORFclusterrawplot(FragORF,'contig_765226_1_562_-', 'contig_765226_1_562_-') 
n=ORFclusterrawplot(FragORF,'contig_88651_465_544_+', 'contig_88651_465_544_+')
o=ORFclusterrawplot(FragORF,'contig_915654_1_462_+', 'contig_915654_1_462_+')


grid.arrange(a,b,c,d,e,f,g,h,i,j,k,l,m,n,o, nrow=5, ncol=3)

```




Manta plots
```{r}
mantadata <- all_MCL_sub [,c(1,2,4:9)]
mantadata <- dplyr::filter (mantadata, grepl('Pseudo-nitzschia|Diatom|Fragilariopsis', Taxa.group))
mantadata$cluster <- make.unique(as.character(mantadata$cluster), sep ="_")
mantadata <- mantadata [,-c(2)]
  
    colname <- "mean"
  i=0
  while (i < 2){
    colname <- paste(colname, i)
    col1 = 2+(i*3)
    col2 = 4+(i*3)
   mantadata[[colname]] <- rowMeans( mantadata[, c(col1:col2)] )
    i <- i+1
  }
  
  
mantadata <- mantadata [,-c(2:7)]


mantadata <-data.frame(mantadata[,-1], row.names = mantadata [,1])
names(mantadata)<- c('0C_0Fe', '0C_+Fe')




mantadata.annot <- all_MCL [,c(48,52, 50,51,47)]
mantadata.annot <- dplyr::filter (mantadata.annot, grepl('Pseudo-nitzschia|Diatom|Fragilariopsis', Taxa.group))
mantadata.annot$cluster <- make.unique(as.character(mantadata.annot$cluster), sep ="_")

mantadata.annot %>% mutate_if(is.factor, as.character) -> mantadata.annot

#mantadata.annot$ann_desc[mantadata.annot$ann_desc==""] <- "NA"
#mantadata.annot$ann_type[mantadata.annot$ann_type==""] <- "NA"
#mantadata.annot$ann_id[mantadata.annot$ann_id==""] <- "NA"



obj1 <- counts2manta(mantadata, annotation = mantadata.annot, a.merge.clmn = 'cluster', agg.clmn = 'ann_desc', meta.clmns = c('Taxa.group','ann_type'), gene.clmns = c('ann_desc', 'ann_type', 'ann_id'))


#plot(obj1, main='Diatom Gene Expression\n at Ocean Station Papa', meta.lev='Taxa.group', lgd.pos='bottomright')

#plot(obj1, main='+/-FE', meta.lev='Taxa.group', lgd.pos='bottomright')


manta.ra(obj1, uniques=0, meta.level = 'Taxa.group', main ="-Fe0C vs +Fe6C" )


caroline::hyperplot(obj1,annout = mantadata.annot, browse=TRUE, cex = 1, w = 72 * 8, h = 72 * 6 )


#plot(obj1, uniques=1,  meta.level = 'Taxa.group', main ="-Fe0C vs +Fe6C")


volcanoplotMCL(DiatomMCL, "X12_vs_17.log2fc", "X12_vs_17.fdr", "Synergy Test: 0C_VS_Fe3C")

#obj <- counts2manta(cts, annotation=cts.annot,
#+ a.merge.clmn='query_seq', agg.clmn='what_def', meta.clmns=c('family','genus_sp'),
#+ gene.clmns=c('what_def','kid','pathway'))
```
 
 
 
 
 RA plot
```{r}
RAp <- read.csv("RA.csv")
raPlot(mantadata$`0C_0Fe`, mantadata$`6C_+Fe`, uniques = 0)


RA <- raPlot(RAp$noFe, RAp$Fe, uniques = TRUE, jitter = 0.3)
cond.unique <- apply(cbind(RAp$noFe, RAp$Fe), 1, function(d) any(d==0))
points(RA$A,RA$R, col=c('black','orange')[cond.unique+1])

RA <- raPlot(FragMCL$TFG_t5_1B, FragMCL$TFG_t5_7C, pch = '', uniques = FALSE)
cond.unique <- apply(cbind(FragMCL$TFG_t5_1B,FragMCL$TFG_t5_7C), 1, function(d) any(d==0))
points(RA$A,RA$R, col=c('black','orange')[cond.unique+1])

RA <- raPlot(a, b, pch='')
cond.unique <- apply(cbind(a,b), 1, function(d) any(d==0))
points(RA$A,RA$R, col=c('black','orange')[cond.unique+1])

#use EdgeR to identify outliers 
```



```{r}

rhodopsin_clust231_clust <- dplyr::filter (all_diatoms_MCL, grepl('rhodopsin|Rhodopsin|clust_10297 |clust_129508 |clust_130846 |clust_142867 |clust_1508 |clust_1548 |clust_16487 |clust_19837 |clust_22740 |clust_231 |clust_28522 |clust_3959 |clust_4014 |clust_779029 |clust_8237 ', name))


rhodopsin_clust231_clust <- dplyr::filter (all_ORF, grepl('clust_1548 ', cluster))

clustername_vs_foldchange_test <- function(dataset, foldchange, FDR, graphtitle){
  volcano <- dataset[c("name",foldchange, FDR,"Taxa.group", "cluster" )]
  
  names(volcano) <- c( "gene", "Fold", "FDR", "taxa", "cluster")
  
  ggplot (data = volcano, aes(x= Fold, y= cluster, color = taxa, label=gene)) +
    geom_point (alpha=0.5)+
    scale_shape (solid=FALSE)+
    ylab("cluster")+ 
    xlab("log2 fold-change")+
    scale_color_manual(values = c("black", "red", "blue"))+
    ggtitle(graphtitle)+
    theme_bw()+
    theme(plot.title = element_text(hjust = 0.5))+
    theme(axis.text.x = element_text(
            hjust=1,
      size = 9, 
      face = "bold")) 
  #geom_text(data = subset(volcano, abs(Fold) >1 & -log10(FDR) >1.3),
  #aes(label="*"), box.padding = 0, point.padding = 0.1, color= 'black', size =5)
 
}


clustername_vs_foldchange_test( rhodopsin_clust231_clust, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms")

clustername_vs_foldchange_test( rhodopsin_clust231_clust, "Fe_6C_vs_Fe_0C_foldchange", "Fe_6C_vs_Fe_0C_pvalue", "temp_6C_vs_3C_ORFs_alldiatoms")

clustername_vs_foldchange_test( rhodopsin_clust231_clust, "int_fe_temp_6C_vs_0C_foldchange", "int_fe_temp_6C_vs_0C_pvalue", "hfRFs_alldiatoms")


```
















LHC plots for Bev
```{r}
all_LHCs <- dplyr::filter(pseudo_frag_ORF, grepl("Lhcx|Lhcf|Lhcr|Lhca|clust_332 |clust_1236 |clust_80 |clust_402 |clust_1044 |clust_3282 |clust_1194 |clust_3219 |clust_1084 |clust_3551 |clust_646 |clust_1938 |clust_2549 |clust_3551 |clust_1787 |clust_712 |clust_2256 |clust_3363 ", name))

all_LHCs <- dplyr::filter(pseudo_ORF,grepl("clust_1820 ", name))#plastocyanin


volcanoplot_interactive_LHC<- function(dataset, Fold, FDR, volcanontitle){
  volcano <- dataset[c("name",Fold, FDR,"Taxa.group", "cluster")]
  x <- Fold
  
  names(volcano) <- c( "gene", "Fold", "FDR", "taxa", "cluster")
  volcano ["group"] <- "no-fold-change"
  
  if(grepl("Fe_vs_noFe_foldchange", x)) {
    volcano['size'] <- rowMeans( dataset[, c(4:21)])
  } else if (grepl("temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("temp_6C_vs_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:21)])
  } else if (grepl("int_fe_temp_6C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9, 16:21)])
  } else if (grepl("int_fe_temp_3C_vs_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:15)])
  } else if (grepl("noFe_3C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6,10:12 )])
  } else if (grepl("noFe_6C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:6, 16:18)])
  } else if (grepl("Fe_3C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(7:9, 13:15)])
  } else if (grepl("Fe_6C_vs_Fe_0C_foldchange", x)){
    volcano['size'] <- rowMeans(dataset[, c(7:9, 19:21)])
  } else if (grepl("noFe_6C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:18,10:12)])
  } else if (grepl("Fe_6C_vs_Fe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(13:15, 19:21)])
  } else if (grepl("Fe_0C_vs_noFe_0C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(4:9)])
  } else if (grepl("Fe_3C_vs_noFe_3C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(10:15)])
  } else if (grepl("Fe_6C_vs_noFe_6C_foldchange", x)){
    volcano['size'] <- rowMeans( dataset[, c(16:21)])
  }
  
  volcano [which(volcano['FDR'] < 0.05 & abs(volcano['Fold']) < 2), "group"]<- "nofoldchange_significant" 
  
  volcano [which(volcano ['FDR'] > 0.05 & abs(volcano['Fold'])> 2), "group"] <- "fold-change_notsignificant"
  
  volcano [which(volcano ['FDR'] < 0.05 & abs(volcano['Fold']) > 2), "group"] <- "fold-change_significant"
  
  vline <- function(x = 0, color = "black") {
    list(
      type = "line", 
      y0 = 0, 
      y1 = 1, 
      yref = "paper",
      x0 = x, 
      x1 = x, 
      line = list(color = color, dash='dot', width=1)
    )
  }
  
  hline <- function(y = 0, color = "black") {
    list(
      type = "line", 
      x0 = 0, 
      x1 = 1, 
      xref = "paper",
      y0 = y, 
      y1 = y, 
      line = list(color = color, dash='dot', width=1)
    )
  }
  
  x <- list (title ="Log2 Fold Change")
  y <- list (title = "-Log10_P-value")
  plot_ly(data = volcano, x = volcano$Fold, y = -log10(volcano$FDR), text = volcano$gene,type = "scatter", mode = "markers", symbol = ~taxa, color =  ~cluster, colors = "Dark2", size = ~size)%>% 
    layout(title = volcanontitle)%>%
    layout (xaxis = x, yaxis = y)%>%
    layout(shapes = hline(1.301))
}
#size = ~size
#list(vline (-1), vline(1)


volcanoplot_interactive_LHC(all_LHCs, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_LHCs_Pseudo-Frag")

volcanoplot_interactive_LHC(all_LHCs, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "hightemp_vs_lowtemp_LHCs_Pseudo-Frag")

heatmap(all_LHCs, "clust")
```










LHC sequences for Bev
```{r}
all_LHCs <- dplyr::filter(pseudo_frag_ORF, grepl("Lhcx|Lhcf|Lhcr|Lhca|clust_332 |clust_1236 |clust_80 |clust_402 |clust_1044 |clust_3282 |clust_1194 |clust_3219 |clust_1084 |clust_3551 |clust_646 |clust_1938 |clust_2549 |clust_3551 |clust_1787 |clust_712 |clust_2256 |clust_3363 ", name))





x <- all_LHCs$name
y <- gsub("(.*);.*", "\\1", x)
all_LHCs$shortname <- y

volcanoplot_interactive(all_LHCs, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms")



volcanoplot_interactive(all_LHCs, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_ORFs_alldiatoms")



all_diatom_fasta <- read.csv ("all_diatom_fasta.csv")


lhc_fasta <- dplyr::filter(all_diatom_fasta, grepl(paste(all_LHCs$orf_id,collapse = "|"), orf_id))




write.csv(lhc_fasta, "all_Lhcs_fasta.csv")



write.csv(all_LHCs, "all_Lhcs_DE.csv")



LHCFASTA <- read.csv("TFG_LHCs_FASTA.csv")

write.fasta (sequences = (as.list(LHCFASTA$AA_sequence)), names = LHCFASTA$name_id ,file.out = "TFG_LHCsequences.fasta")


write.csv(lhc_fasta, "a.csv")
a <- read.csv("a.csv")
Pseudo_lhc <- dplyr::filter(a, grepl("Pseudo-nit", orf_id))

write.fasta (sequences = (as.list(Pseudo_lhc$AA_sequence)), names = Pseudo_lhc$orf_id ,file.out = "pseudo_lhc.fasta")

```

Plastocyanin sequences for tree
```{r}
plastocyanin <- dplyr::filter(pseudo_frag_ORF, grepl("clust_1820 ", name))

x <- plastocyanin$name
y <- gsub("(.*);.*", "\\1", x)
plastocyanin$shortname <- y



volcanoplot_interactive(plastocyanin, "Fe_vs_noFe_foldchange", "Fe_vs_noFe_pvalue", " Fe_vs_NoFe_rhodopsin_ORFs_alldiatoms")



volcanoplot_interactive(plastocyanin, "temp_6C_vs_0C_foldchange", "temp_6C_vs_0C_pvalue", "temp_6C_vs_0C_rhodopsin_ORFs_alldiatoms")



all_diatom_fasta <- read.csv ("all_diatom_fasta.csv")


plastocyanin_fasta <- dplyr::filter(all_diatom_fasta, grepl(paste(plastocyanin$orf_id,collapse = "|"), orf_id))


write.csv(plastocyanin_fasta, "plastocyanin_fasta.csv")

write.csv(plastocyanin, "plastocyanin_DE.csv")

#plastocyanin_fasta <- read.csv("TFG_LHCs_FASTA.csv")

write.fasta (sequences = (as.list(plastocyanin_fasta$AA_sequence)), names = plastocyanin_fasta$orf_id ,file.out = "plastocyanin.fasta")
```


ISIP1 sequences for tree
```{r}
all_diatom_fasta <- read.csv ("all_diatom_fasta.csv")

isip1<- dplyr::filter(all_ORF_sub, grepl('isip1|ISIP1|clust_1814 ', name))

isip1_fasta <- dplyr::filter(all_diatom_fasta, grepl(paste(isip1$orf_id,collapse = "|"), orf_id))
write.csv(isip1_fasta, "isip1_fasta.csv")

isip1_fasta <- read.csv("isip1_fasta.csv")

write.fasta (sequences = (as.list(isip1_fasta$AA_sequence)), names = isip1_fasta$orf_id ,file.out = "isip1.fasta")

```



How the FASTA file was converted into DF
```{r}
# read.fast(ldlld )
# x<- merge(FASTA, all_diatoms_ORF, by = 'orf_id')
# tree_sub <- merge(tree, clust_231, by = "orf_id" )
# x$Names <- paste(x$Taxa.group,"_",x$orf_id,x$cluster, sep = "")
# 
# x <- x[c("Names", "AA_sequence")]
# names(x)<- c("orf_id", "AA_sequence")
# 
 write.csv(all_diatom_fasta, "all_diatom_fasta.csv")
    
```



```{r}
all_diatom_fasta <- read.csv ("all_diatom_fasta.csv")

rhodopsin <- dplyr::filter (all_diatoms_ORF, grepl("clust_1548 ", name))


rhodopsin_Fe <- subset (rhodopsin,(rhodopsin$Fe_vs_noFe_pvalue <0.05 & rhodopsin$temp_6C_vs_0C_pvalue <0.05))

rhodopsin_temp <- subset (rhodopsin,(rhodopsin$temp_6C_vs_0C_pvalue <0.05))

rhodopsin_inter <- subset (rhodopsin,(rhodopsin$int_fe_temp_6C_vs_0C_pvalue <0.05))




rhodopsin_fasta <- dplyr::filter(all_diatom_fasta, grepl(paste(rhodopsin$orf_id,collapse = "|"), orf_id))

write.fasta (sequences = (as.list(rhodopsin_fasta$AA_sequence)), names = rhodopsin_fasta$orf_id ,file.out = "rhodopsin.fasta")

```





KEGG pathway analysis using TFG data and annotations
```{r}
frag_pathway <- dplyr::filter (all_ORF, grepl('Frag', Taxa.group))  

frag_pathway_2 <- frag_pathway[c(47,2, 4, 10 , 8, 51,52, 53:55, 171:176, 4, 10 )] 

frag_pathway_allKO <-  dplyr::filter (frag_pathway_2, grepl('K', KO)) #filter out the ORFs with no KO annotation
length(unique(frag_pathway_allKO$KO))


#Subsetting for KO_pathways with at least n# of unique KO IDs
#this loops goes through all the pathways, and picks  only the pathways with a >n # of unique KO IDs in a new dataframe
pathways<-unique(frag_pathway_allKO$KO_pathway)
frag_pathway_KOfiltered<-data.frame()
for (i in 1:length(pathways)){
  temp<-base::subset(frag_pathway_allKO,KO_pathway==pathways[i])
  if (length(unique(temp$KO))>10) {frag_pathway_KOfiltered<-rbind(frag_pathway_KOfiltered,temp)}
}

length(unique(frag_pathway_KOfiltered$kegg_desc))

#frag_pathway_KOfiltered$temp_6C_vs_0C_de <-  as.character(frag_pathway_KOfiltered$temp_6C_vs_0C_de)


#temperature 
ggplot(frag_pathway_KOfiltered, aes(x = temp_6C_vs_0C_foldchange, y = KO_pathway, color = str_wrap(temp_6C_vs_0C_de, 20)))+
  geom_jitter(alpha=0.3, size = 2,  height = 0.2)+
  scale_color_manual(values = c("red", "grey", "red"))+
  geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )+
  theme_bw()


#iron 
ggplot(frag_pathway_KOfiltered, aes(x = Fe_vs_noFe_foldchange, y = KO_pathway, color = str_wrap(Fe_vs_noFe_de, 10)))+
  geom_jitter(alpha=0.3, size = 2,  height = 0.2)+
  scale_color_manual(values = c("red", "grey", "red"))+
  geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )+
  theme_bw()




#filter out KO IDs with less than n# of ORFS (ideally we want several ORFs per KO ID ).
# frag_pathway_KOfiltered_ORF <- frag_pathway_KOfiltered[frag_pathway_KOfiltered$KO %in% names(which(table(frag_pathway_KOfiltered$KO) >2)), ]
# length(unique(frag_pathway_KOfiltered_ORF$KO))



#frag_pathway_KOfiltered_ORF$DE_temp <- abs(frag_pathway_KOfiltered_ORF$temp_6C_vs_0C_de)
 
# ggplot(frag_pathway_KOfiltered_ORF_temp, aes(x = temp_6C_vs_0C_foldchange, y = KO_desc))+ #color = str_wrap(KO_pathway, 30),shape = DE
#   geom_jitter(alpha=0.3, size = 2,  height = 0.2)+
#  # scale_color_manual(values = c("black", "red"))+
#   #geom_point(alpha = 0.8, size = 3)+
#  #scale_shape_manual(values = c(1, 16))+
#   geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )+
#       theme_bw()

```
 
 
 
KEGG pathway analysis for KEGG database
```{r}
#matching the downloaded KEGG annotation to my data
kegg<- read.csv("KEGG2.csv", header = TRUE)
kegg<- kegg[c(1:5)]

frag_pathway <- dplyr::filter (all_ORF, grepl('Frag', Taxa.group))  


frag_pathway_2 <- frag_pathway[c(47,2, 4, 10 , 8, 51,52, 53:55, 171:176, 4, 10 )] 

frag_pathway_kegg<- merge(frag_pathway_2, kegg, by.x = "KO", by.y = "KO")
length(unique(frag_pathway_kegg$KO))

#testes <- dplyr::filter(frag_pathway_kegg, grepl("00789", KO))

#Subsetting for KO_pathways with at least n# of unique KO IDs
#this loops goes through all the pathways, and picks puts only the pathways with a >n # of unique KO IDs in a new dataframe
pathways<-unique(frag_pathway_kegg$C)

frag_pathway_kegg_filtered<-data.frame()
for (i in 1:length(pathways)){
  temp<-base::subset(frag_pathway_kegg,C==pathways[i])
  if (length(unique(temp$KO))>10) {frag_pathway_kegg_filtered<-rbind(frag_pathway_kegg_filtered,temp)}
}

length(unique(frag_pathway_kegg_filtered$C))


frag_pathway_kegg_filtered_2 <- dplyr::filter(frag_pathway_kegg_filtered, grepl("BR:", C))
length(unique(frag_pathway_kegg_filtered_2$C))


#y = reorder(A,temp_6C_vs_0C_foldchange)

#temperature 
ggplot(frag_pathway_kegg_filtered_2, aes(x = temp_6C_vs_0C_foldchange, y = reorder(C,-temp_6C_vs_0C_foldchange) , color = str_wrap(temp_6C_vs_0C_de, 20)))+
  geom_jitter(alpha=0.4, size = 2,  height = 0.2)+
  scale_color_manual(values = c("red", "grey", "red"))+
  geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )+
  theme_bw()


#iron 
ggplot(frag_pathway_kegg_filtered_2, aes(x = Fe_vs_noFe_foldchange, y = reorder(C,-Fe_vs_noFe_foldchange), color = str_wrap(Fe_vs_noFe_de, 10)))+
  geom_jitter(alpha=0.4, size = 2,  height = 0.2)+
  scale_color_manual(values = c("red", "grey", "red"))+
  geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )+
  theme_bw()
```


TopGO
```{r}
library(topGO)
library(data.table)
library(stringi)

GO <- fread("all_tfg_GO.table.txt", select = c("orf_id", "GO_terms"), fill = TRUE) #select only orf_id and GO_terms colummns to import

all_ORF_GO <- merge(all_ORF, GO, by = "orf_id")  #this puts GO terms in for each ORF that was GO annotated

frag_GO <- dplyr::filter (all_ORF_GO, grepl('Frag', Taxa.group))
  
frag_GO_tempFe <- frag_GO [c(1, 179, 3, 9)] #53 = column for KO

frag_GO_tempFe_filtered <- dplyr::filter (frag_GO_tempFe, grepl('0', GO_terms))#select only the genes with GO terms associated with them

frag_GO_tempFe_filtered  <- separate(data = frag_GO_tempFe_filtered ,col =  GO_terms, into = c("a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k"), sep = "\\|\\|")

frag_GO_tempFe_filtered  <- melt(frag_GO_tempFe_filtered , id.vars = c(1, 13,14)) #this seperates ORFS with several go terms into several rows

frag_GO_tempFe_filtered <- na.omit(frag_GO_tempFe_filtered) #remove rows with no annotations
frag_GO_tempFe_filtered$value <- paste0("GO:", frag_GO_tempFe_filtered$value)


#create list of all orfs with GO in the TFG (gene universe)
geneID2GO_list_allGO_Fe <- split(frag_GO_tempFe_filtered$value, frag_GO_tempFe_filtered$orf_id, drop = TRUE)

#create list of outlier genes. This is a list of ORFs that are DE (up, down, or both ways with Fe or temp)
#firt do it for ALL the DE egens (up AND down DE)
outliers <- dplyr::filter(frag_GO_tempFe_filtered, grepl("-1", temp_6C_vs_0C_de))

#Get the gene names
geneNames <- names(geneID2GO_list_allGO_Fe)

#Make a vector called geneList which has a 1 if a gene in the TFG (gene universe) in the list of outlierproteins, 0 if the gene in the gene universe is not in the list of outlierproteins.
geneList <- factor(as.integer(geneNames %in% outliers$orf_id))

#Associate the gene names of all genes in the TFG gene universe with the vector we just made
names(geneList) <- geneNames


#Now we're going to make a GOdata file, we specify what ontology we are interested in (MF = molecular function, BP = Biological Process, CC = Cellular Component), we specify our genes as the geneList, we specify that our annotation is a custom gene2GO file (annot=annFUN.gene2GO), we specify the gene2GO file.


GOdataBP <- new("topGOdata", 
                ontology = "BP", 
                allGenes = geneList,
                annot = annFUN.gene2GO,
                gene2GO = geneID2GO_list_allGO_Fe,
                nodeSize= 10)
  

#Now graph the GOdata
graph(GOdataBP)


#Now run enrichment analysis with all algorithms
resultBPFishclassic <- runTest(GOdataBP, algorithm = "classic", statistic = "fisher")
resultBPFishelim <- runTest(GOdataBP, algorithm = "elim", statistic = "fisher")
resultBPFishweight <- runTest(GOdataBP, algorithm = "weight", statistic = "fisher")
resultBPFishweight01 <- runTest(GOdataBP, algorithm = "weight01", statistic = "fisher")
resultBPFishlea <- runTest(GOdataBP, algorithm = "lea", statistic = "fisher")
resultBPFishparentchild <- runTest(GOdataBP, algorithm = "parentchild", statistic = "fisher")
allGOBP = usedGO(object = GOdataBP)
allResBP <-GenTable(GOdataBP, classic = resultBPFishclassic, elim = resultBPFishelim, weight = resultBPFishweight, weight01 = resultBPFishweight01, lea = resultBPFishlea, parentchild = resultBPFishparentchild, orderBy = "weight01", ranksOf = "weight01", topNodes = length(allGOBP))
allResBP$weight01padj <- p.adjust(allResBP$weight01, method = 'fdr')
write.table(allResBP, "GOresultsBP.txt", row.names = F, quote = F, sep = "%")

allResBP$classic <- as.numeric(allResBP$classic)
allResBP_temp_negDE <- allResBP[ which(allResBP$classic < 0.05),]

ggplot(allResBP_temp_negDE, aes(y= reorder(Term, -classic), x = classic))+
  geom_point()


allResBP_temp_negDE_2 <- allResBP_temp_negDE [c(1:6)]
allResBP_temp_negDE_2$DE <- "depleted"

#top15% DE with iron
# n <- 50
# allResBP_temp_posDE_2 <- allResBP_temp_posDE[allResBP_temp_posDE$Significant > quantile(allResBP_temp_posDE$Significant,prob=1-n/100),]

allResBP_temp_posDE_2 <- allResBP_temp_posDE [c(1:6)]
allResBP_temp_posDE_2$DE <- "enriched"


BP_DE<- rbind(allResBP_temp_posDE_2, allResBP_temp_negDE_2)

BP_DE_2<- subset(BP_DE, Significant >10)


ggplot(BP_DE, aes(y = reorder(Term,-classic), x = classic, color = DE))+
     geom_point(alpha = 0.8, size = 3)+
  theme_bw()+
  scale_color_manual(values = c("grey", "black"))+
xlab("p-value")+
  ylab("GO - Biological Process")



#test <- dplyr::filter (frag_GO, grepl('contig_702374_1_435_-', orf_id))


#https://support.bioconductor.org/p/29775/


############


#  myterms = c("GO:0006553")
#  mygenes <- genesInTerm(GOdataBP, myterms)
#  
# for (i in 1:length(myterms))
# x<-  list({
#        myterm <- myterms[i]
#        mygenesforterm <- mygenes[myterm][[1]]
#        mygenesforterm <- paste(mygenesforterm, collapse=',')
#        print(paste("Term",myterm,"genes:",mygenesforterm))
#      })



#http://avrilomics.blogspot.com/2015/07/using-topgo-to-test-for-go-term.html



###########


#plotting the GO
DEGO <- dplyr::filter(frag_GO_tempFe_filtered, grepl("GO:0009654|GO:0005840|GO:0009538|GO:0009523|GO:0009522|GO:0009521|GO:0009579|GO:0032991|GO:0034357|GO:0042651|GO:0043232|GO:0044436", value))



#test <- dplyr::filter(frag_GO, grepl("contig_1195125_202_417_+", orf_id))


unique(DEGO$value)

ggplot(DEGO, aes(x = Fe_vs_noFe_de, y = value))+
  geom_jitter(width= 0.3, height=0.2)

#x = reorder(miRNA, -value)

# #Now repeat GO enrichment analysis for CC ontology
# GOdataCC <- new("topGOdata", 
#                 ontology = "CC", 
#                 allGenes = geneList,
#                 annot = annFUN.gene2GO,
#                 gene2GO = geneID2GO_list_allGO_Fe,
#                 nodeSize= 5)
# #Now graph the GOdata
# graph(GOdataCC)
# #Now run enrichment analysis with all algorithms
# resultCCFishclassic <- runTest(GOdataCC, algorithm = "classic", statistic = "fisher")
# resultCCFishelim <- runTest(GOdataCC, algorithm = "elim", statistic = "fisher")
# resultCCFishweight <- runTest(GOdataCC, algorithm = "weight", statistic = "fisher")
# resultCCFishweight01 <- runTest(GOdataCC, algorithm = "weight01", statistic = "fisher")
# resultCCFishlea <- runTest(GOdataCC, algorithm = "lea", statistic = "fisher")
# resultCCFishparentchild <- runTest(GOdataCC, algorithm = "parentchild", statistic = "fisher")
# allGOCC = usedGO(object = GOdataCC)
# allResCC <-GenTable(GOdataCC, classic = resultCCFishclassic, elim = resultCCFishelim, weight = resultCCFishweight, weight01 = resultCCFishweight01, lea = resultCCFishlea, parentchild = resultCCFishparentchild, orderBy = "weight01", ranksOf = "weight01", topNodes = length(allGOCC))
# allResCC$weight01padj <- p.adjust(allResCC$weight01, method = 'fdr')
# write.table(allResCC, "GOresultsCC.txt", row.names = F, quote = F, sep = "%")



 


#Now we're going to make a GOdata file, we specify what ontology we are interested in (MF = molecular function, BP = Biological Process, CC = Cellular Component), we specify our genes as the geneList, we specify that our annotation is a custong .gene2GO file (annot=annFUN.gene2GO), we specify the gene2GO file.
# GOdataMF <- new("topGOdata", 
#                 ontology = "MF", 
#                 allGenes = geneList,
#                 annot = annFUN.gene2GO,
#                 gene2GO = geneID2GO_list_allGO_Fe,
#                 nodeSize= 5)#LJ - this is to specify that only use pathways with 10 or more GO IDs
# #Now graph the GOdata
# 
# graph(GOdataMF)
# 
# 
# 
# resultMFFishclassic <- runTest(GOdataMF, algorithm = "classic", statistic = "fisher")
# #Repeat for all possible algorithms:
# resultMFFishelim <- runTest(GOdataMF, algorithm = "elim", statistic = "fisher")
# resultMFFishweight <- runTest(GOdataMF, algorithm = "weight", statistic = "fisher")
# resultMFFishweight01 <- runTest(GOdataMF, algorithm = "weight01", statistic = "fisher")
# resultMFFishlea <- runTest(GOdataMF, algorithm = "lea", statistic = "fisher")
# resultMFFishparentchild <- runTest(GOdataMF, algorithm = "parentchild", statistic = "fisher")
# #Get a list of all GO terms considered in analysis
# allGOMF = usedGO(object = GOdataMF)
# #Make a table of all your results
# allResMF <-GenTable(GOdataMF, classic = resultMFFishclassic, elim = resultMFFishelim, weight = resultMFFishweight, weight01 = resultMFFishweight01, lea = resultMFFishlea, parentchild = resultMFFishparentchild, orderBy = "weight01", ranksOf = "weight01", topNodes = length(allGOMF))
# #Make a new column in the table which does a False Discovery adjustment to p-values calculated using the weight01 algorithm (a way to correct for Type I errors due to multiple comparisons)
# #allResMF$weight01padj <- p.adjust(allResMF$weight01, method = 'fdr')
# write.table(allResMF, "GOresultsMF.txt", row.names = F, quote = F, sep = "%")





#................#
#allGO = genesInTerm(GOdataBP)

#SAM_ANOTATION = lapply(allGO,function(x) x[x %in% outliers$orf_id] )

# Your significant genes for GO:0051427
#SAM_ANOTATION[["GO:0043232"]]


#````#


#choose only upregulated GOs

# pathway_allGO_sub4 <- dplyr::filter (pathway_allGO_sub3, grepl('-1', temp_6C_vs_0C_de))
#   
# pathway_allGO_sub4$Evidence <- "IEA"
# pathway_allGO_sub5 <-   pathway_allGO_sub4 [c(3,4,1)]
# 
# colnames(pathway_allGO_sub5) <- c("GO", "Evidence", "GID")
# 
# write.csv(pathway_allGO_sub5, "testGOminus.csv")

```


GOstats using GO
```{r}
#https://bioconductor.org/packages/release/bioc/vignettes/GOstats/inst/doc/GOstatsForUnsupportedOrganisms.pdf
library("GSEABase")
library("GOstats")

GO <- fread("all_tfg_GO.table.txt", select = c("orf_id", "GO_terms"), fill = TRUE) #select only orf_id and GO_terms colummns to import

all_ORF_GO <- merge(all_ORF, GO, by = "orf_id")  #this puts GO terms in for each ORF that was GO annotated

frag_GO <- dplyr::filter (all_ORF_GO, grepl('Frag', Taxa.group))
  
frag_GO_tempFe <- frag_GO [c(1, 179, 3, 9)] #53 = column for KO

frag_GO_tempFe_filtered <- dplyr::filter (frag_GO_tempFe, grepl('0', GO_terms))#select only the genes with GO terms associated with them

frag_GO_tempFe_filtered  <- separate(data = frag_GO_tempFe_filtered ,col =  GO_terms, into = c("a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k"), sep = "\\|\\|")

frag_GO_tempFe_filtered  <- melt(frag_GO_tempFe_filtered , id.vars = c(1, 13,14)) #this seperates ORFS with several go terms into several rows



##creating a goframaData for frag
frag_GO_tempFe_filtered <- na.omit(frag_GO_tempFe_filtered) #remove rows with no annotations
frag_GO_tempFe_filtered$evidence <- "IEA"
frag_GO_tempFe_filtered$value <- paste0("GO:", frag_GO_tempFe_filtered$value)
goframeData_frag <- frag_GO_tempFe_filtered[c(5,6,1)]


# goFrame=GOFrame(goframeData,organism="Homo sapiens")
# goAllFrame=GOAllFrame(goFrame)
goFrame=GOFrame(goframeData_frag,organism="Fragilariopsis")
goAllFrame=GOAllFrame(goFrame)

# we pass this kind of custom information around, we
#have chosen to support geneSetCollection objects from GSEABase package. 
gsc <- GeneSetCollection(goAllFrame, setType = GOCollection())


universe_frag <- names(goAllFrame) #this is a list of all the orfs in the frag list that have GO term associated with them

#genes_frag: this is a list of ORFs with something interesting (outliers)
outliers <- dplyr::filter(frag_GO_tempFe_filtered, grepl("-1", temp_6C_vs_0C_de))
outliers_list <- split(outliers$value, outliers$orf_id, drop = TRUE)
genes_frag <- names(outliers_list) #list of orfs that are outliers (DE, DeUP, DeDown etc)

#then make a GOstat object
 
params <- GSEAGOHyperGParams(name="My Custom GSEA based annot Params",
                              geneSetCollection=gsc,
                              geneIds = genes_frag,#outliers/up and down reg orfs
                              universeGeneIds = universe_frag,#all of the ORFs
                              ontology = "BP",
                              pvalueCutoff = 0.05,
                              conditional = FALSE,
                              testDirection = "over"
                            )

#then perform a hypergeometric test on the GO terms
Over <- hyperGTest(params)
summary(Over)
x<- data.frame (summary(Over))

#I want to see which orfs were used in the analysis/which orfs are associated with enriched GO terms


enrichedORFs <- geneIdsByCategory(Over,c('GO:0015979')) #this gets the genes involved in the 'count'
y<- data.frame((sapply(enrichedORFs, '[', seq(max(sapply(enrichedORFs, length))))))
y$term <- "photosynthesis"
names(y) <- c("orf_id", "term")
test <- merge (y, frag_GO, by = c("orf_id"))



#test <- frag_GO[unlist(lapply(enrichedORFs, function(x) grep(x, frag_GO$orf_id, fixed = TRUE))),]


#this converts the list above into a dataframe
#y<- t(sapply(enrichedORFs, '[', seq(max(sapply(enrichedORFs, length)))))


allorfsused <- geneIdUniverse(Over)

y<- data.frame((sapply(allorfsused, '[', seq(max(sapply(allorfsused, length))))))
y <- y[c(4)]
y$term <- "photosynthesis"
names(y) <- c("orf_id", "term")
test <- merge (y, frag_GO, by = c("orf_id"))


ggplot(test, aes(x = temp_6C_vs_0C_foldchange, y = term, color = str_wrap(temp_6C_vs_0C_de)))+
  geom_jitter(alpha=0.4, size = 2,  height = 0.2)+
  scale_color_manual(values = c("red", "grey", "red"))+
  geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )+
  theme_bw()
       
       
       #this selects ALL of the orfs used in the analysis of each GO term
#allorfsuseda <- t(sapply(allorfsused, '[', seq(max(sapply(allorfsused, length)))))

```


GOstats using KO
```{r}
#https://bioconductor.org/packages/release/bioc/vignettes/GOstats/inst/doc/GOstatsForUnsupportedOrganisms.pdf

library("GSEABase")
library("GOstats")
library("KEGG.db")
#library("KEGGREST")
#BiocManager::install("KEGGREST")

#load the KEGG data
kegg<- read.csv("KEGG2.csv", header = TRUE)
kegg<- kegg[c(1:5)]
kegg$C <- str_sub(kegg$C, end=-14)
length(unique(kegg_2$KO))

#this combines all the "C" pathways into one column. 
kegg_2 <- aggregate(C~KO, kegg, FUN = toString)
length(unique(kegg$C))

#this is to reduce the length of the annotations so they plot better
categorymap <- c(
"Glycolysis / Gluconeogenesis [, Carbon fixation in photosynthetic organisms [, HIF-1 signaling pathway [, Exosome" ="Carbon fixation in photosynthetic organisms, Glycolysis/Gluconeogenesis",  
"Oxidative phosphorylation [" = "Oxidative phosphorylation",

"Glyoxylate and dicarboxylate metabolism [, Nitrogen metabolism [, Alanine, aspartate and glutamate metabolism [, Arginine biosynthesis [, Two-component system [, Exosome" = "Nitrogen metabolism, Alanine, aspartate and glutamate metabolism Glyoxylate and dicarboxylate metabolism", 

"Photosynthesis [, Photosynthesis proteins" = "Photosynthesis", 

"Proteasome [, Peptidases, Proteasome" = "Proteasome, Peptidases", 

"Ribosome [, Ribosome" = "Ribosome", 

"Ribosome [" = "Ribosome", 

"RNA transport [, Transfer RNA biogenesis, Messenger RNA biogenesis, Translation factors, Exosome" = "RNA transport, RNA biogenesis", 

"RNA degradation [, Mitochondrial biogenesis, Mitochondrial biogenesis, Chaperones and folding catalysts, Transporters, Exosome, Exosome, Exosome" = "RNA degradation",

"RNA transport [, Ribosome biogenesis in eukaryotes [, Transfer RNA biogenesis, Messenger RNA biogenesis, Ribosome biogenesis, Ribosome biogenesis, Chromosome and associated proteins, Chromosome and associated proteins, GTP-binding   protein, Exosome" = "RNA transport, Ribosome biogenesis, RNA biogenesis",

"Chromosome and associated proteins, Exosome" = "Chromosome and associated proteins, Exosome", 

"Peroxisome [" = "Peroxisome",

"Nicotinate and nicotinamide metabolism [" = "Nicotinate and nicotinamide metabolism", 

"Oxidative phosphorylation [" = "Oxidative phosphorylation",

"Oxidative phosphorylation [, Two-component system [, Mitochondrial biogenesis" = "Oxidative phosphorylation, Mitochondrial biogenesis",

"Purine metabolism [, Pyrimidine metabolism [, RNA polymerase [, Transcription machinery" = "Purine/Pyrimidine metabolism, RNA polymerase, Transcription machinery",

"Protein export [" = "Protein export",

"Chaperones and folding catalysts" = "Chaperones and folding catalysts",

"Protein kinases, Transcription machinery, Chromosome and associated proteins, Chromosome and associated proteins, Chromosome and associated proteins, Chromosome and associated proteins, Chromosome and associated proteins, Chromosome and associated proteins, Chromosome and associated proteins" = "Protein kinases, Transcription machinery, Chromosome and associated proteins", 
  
"Pyruvate metabolism [, Propanoate metabolism [, Fatty acid biosynthesis [, Aflatoxin biosynthesis [, AMPK signaling pathway [" = "Pyruvate/Propanoate metabolism, fatty acid biosynthesis", 

"Ribosome biogenesis in eukaryotes [, Ribosome biogenesis" = "Ribosome biogenesis", 

"RNA degradation [, Mitochondrial biogenesis, Chaperones and folding catalysts, Exosome, Exosome, Exosome, Exosome, Exosome" = "RNA degradation", 
 
"RNA transport [, Ribosome biogenesis in eukaryotes [, Transfer RNA biogenesis, Messenger RNA biogenesis, Ribosome biogenesis, Ribosome biogenesis, Chromosome and associated proteins, Chromosome and associated proteins, GTP-binding protein, Exosome" = "Ribosome biogenesis, RNA transport", 

"RNA degradation [, Mitochondrial biogenesis, Chaperones and folding catalysts, Exosome, Exosome, Exosome, Exosome, Exosome" = "RNA degradation", 

"Glycolysis / Gluconeogenesis [, Citrate cycle (TCA cycle) [, Pyruvate metabolism [, Glyoxylate and dicarboxylate metabolism [, Propanoate metabolism [, Glycine, serine and threonine metabolism [, Valine, leucine and isoleucine degradation [, Exosome" = "Glycolysis/Gluconeogenesis, Citrate cycle (TCA cycle)", 

"Cysteine and methionine metabolism [" = "Cysteine and methionine metabolism, Amino acid biosynthesis", 

"Pentose phosphate pathway [, Purine metabolism [" = "Pentose phosphate pathway, Purine metabolism, Carbon metabolism", 

"Sulfur metabolism [, Purine metabolism [, Selenocompound metabolism [, Monobactam biosynthesis [" =  "Sulfur metabolism, Purine metabolism, Selenocompound metabolism", 

"Cysteine and methionine metabolism [, Protein phosphatase and associated proteins, Exosome" = "Cysteine and methionine metabolism, Amino acid biosynthesis",

"Citrate cycle (TCA cycle) [, Glyoxylate and dicarboxylate metabolism [" = "Citrate cycle (TCA cycle), Glyoxylate and dicarboxylate metabolism", 

"Proteasome [, Peptidases, Membrane trafficking, Proteasome" = "Proteasome", 

"AMPK signaling pathway [, Translation factors, Exosome" = "AMPK signaling pathway", 

"RNA transport [, Messenger RNA biogenesis, Messenger RNA biogenesis, Translation factors, Exosome, Exosome" = "RNA transport",

"mRNA surveillance pathway [, MAPK signaling pathway - fly [, TGF-beta signaling pathway [, Hippo signaling pathway [, Hippo signaling pathway -fly [, Sphingolipid signaling pathway [, PI3K-Akt signaling pathway [, AMPK signaling pathway [, Cell cycle - yeast [, Meiosis - yeast [, Oocyte meiosis [, Tight junction [, Protein phosphatase and associated proteins, Chromosome and associated proteins, Exosome" = "mRNA surveillance pathway, MAPK signaling pathway", 

"mRNA surveillance pathway [, Hippo signaling pathway [, cAMP signaling pathway [, cGMP - PKG signaling pathway [, Regulation of actin cytoskeleton [, Meiosis - yeast [, Oocyte meiosis [, Focal adhesion [, Protein phosphatase and associated proteins, Transcription machinery, Spliceosome, Messenger RNA biogenesis" = "mRNA surveillance pathway", 

"Membrane trafficking, GTP-binding protein, Exosome, Exosome, Exosome" = "Membrane trafficking, Exosome", 

"Ras signaling pathway [, Endocytosis [, Phagosome [, Membrane trafficking, Membrane trafficking, GTP-binding protein, Exosome, Exosome, Exosome, Exosome" = "Ras signaling pathway, Membrane trafficking, Endocytosis", 

"GTP-binding protein" = "GTP-binding protein", 

"Membrane trafficking" = "Membrane trafficking", 

"Messenger RNA biogenesis, Exosome, Exosome, Exosome" = "RNA biogenesis, Exosome", 

"Sulfur metabolism [, Cysteine and methionine metabolism [" = "Sulfur metabolism, Cysteine and methionine metabolism", 

"Glycolysis / Gluconeogenesis [, Carbon fixation in photosynthetic organisms [, Exosome" = "Carbon fixation in photosynthetic organisms, Glycolysis/Gluconeogenesis", 

"Glutathione metabolism [, Peptidases" = "Glutathione metabolism, Peptidases", 

"Pentose phosphate pathway [, Pentose and glucuronate interconversions [, Carbon fixation in photosynthetic organisms [" = "Carbon fixation in photosynthetic organisms, Pentose phosphate pathway",

"Oxidative phosphorylation [, Photosynthesis [, Photosynthesis proteins" = "Photosynthesis",

"Oxidative phosphorylation [, mTOR signaling pathway [, Phagosome [" = "Oxidative phosphorylation, mTOR signaling", 

"Translation factors" = "Translation factors", 

"Porphyrin and chlorophyll metabolism [" = "Porphyrin and chlorophyll metabolism", 

"Lysosome [, Transporters" = "Lysosome, Transporters, Mineral absorption", 

"Glycolysis / Gluconeogenesis [, Pentose phosphate pathway [, Fructose and mannose metabolism [, Carbon fixation in photosynthetic organisms [, Methane metabolism [" = "Carbon fixation in photosynthetic organisms, Pentose phosphate pathway",

"Sulfur metabolism [, Cysteine and methionine metabolism [" = "Sulfur metabolism, Cysteine and methionine metabolism"
) 
  




#then choose only the pathways with >9 unique KOid associated with them
# pathways<-unique(Frag_tempDE$pathway)
#  Frag_tempDE_filtered<-data.frame()
#  for (i in 1:length(pathways)){
# temp<-base::subset(Frag_tempDE,pathway==pathways[i])
# if (length(unique(temp$KO))>1)
#   {Frag_tempDE_filtered<-rbind(Frag_tempDE_filtered,temp)}
# }







# c<- grid.arrange(a, b, nrow = 1)
 
#I want to see which orfs were used in the analysis/which orfs are associated with enriched GO terms
# y<- data.frame((sapply(enrichedORFs, '[', seq(max(sapply(enrichedORFs, length))))))
# y$term <- "photosynthesis"
# names(y) <- c("orf_id", "term")
# test <- merge (y, frag_GO, by = c("orf_id"))


#test <- frag_GO[unlist(lapply(enrichedORFs, function(x) grep(x, frag_GO$orf_id, fixed = TRUE))),]

#this converts the list above into a dataframe
#y<- t(sapply(enrichedORFs, '[', seq(max(sapply(enrichedORFs, length)))))
# 
# allorfsused <- geneIdUniverse(Over) # this gets list of ORFS used in analysis
# 
# y<- data.frame((sapply(allorfsused, '[', seq(max(sapply(allorfsused, length))))))
# y$placeholder <- "placeholder"
# ymelt <- melt(y, id.vars="placeholder")
# ymelt<- na.exclude(ymelt)
# 
# 
# pathways<-unique(test_1$C)
# 
# test_2<-data.frame()
# for (i in 1:length(pathways)){
#   temp<-base::subset(test_1,C==pathways[i])
#   if (length(unique(temp$KO.y))>10) {test_2<-rbind(test_2,temp)}
# }
# 
# ggplot(test_2, aes(x = temp_6C_vs_0C_foldchange, y = str_wrap(B,40) , color = str_wrap(temp_6C_vs_0C_de, 20)))+
#   geom_jitter(alpha=0.4, size = 2,  height = 0.2)+
#   scale_color_manual(values = c("red", "grey", "red"))+
#   geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )+
#   #scale_y_discrete(labels = abbreviate)+
#   theme_bw()
```

GOstats using KO: Frag
```{r}
frag_KO <- dplyr::filter (all_ORF, grepl('Frag', Taxa.group))

#trim down the dataframe to make it easier to look at. 
frag_KO<- frag_KO[c(47,53, 55, 2, 4, 8, 10 )] 

#select only the genes with KO terms associated with them. This is the 'universe'.
frag_all_KO <- dplyr::filter (frag_KO, grepl('K', KO)) 
length(unique(frag_all_KO$KO_pathway))

##create a KO dataframe. The GoStats package requires two columns for the 
#KO analysis, a gene ID column and KO number column. The KO number cannot begin with a K. 
frag_all_KO_universe <- frag_all_KO [c(2,1)]
frag_all_KO_universe$KO_id <- str_sub(frag_all_KO_universe$KO, start=2)
frag_all_KO_universe <- frag_all_KO_universe [c(3,2)]

#after having the dataframe in the correct format, we convert it into a 'keggframe' which can be used by GOstats
keggFrame=KEGGFrame(frag_all_KO_universe,organism="Fragilariopsis")

#Then the keggframe is passed through the KEGG collection to standardize things."we pass this kind of custom information around, we have chosen to support geneSetCollection objects from GSEABase package." 
#gsc = gene set collection
gsc <- GeneSetCollection(keggFrame, setType = KEGGCollection())

#then we specify our univese. this is a list of all the orfs in the frag list that have KO term associated with them.
universe <- names(keggFrame) 

#Now we select a list of ORFs with something interesting (outliers). So, this can be ORFs with DE with iron/temp. in any direction etc. 
outliers<- frag_all_KO[which(frag_all_KO$Fe_vs_noFe_de <0 ) ,]
  
outliers_list <- split(outliers$KO, outliers$orf_id, drop = TRUE)
outliers_list <- names(outliers_list) #list of orfs that are outliers (DE, DeUP, DeDown etc)

#then make a GOstat object, which will be used for the KEGG GSEA analysis
params <- GSEAKEGGHyperGParams(name="TFG_GSEA",
                              geneSetCollection=gsc,
                              geneIds = outliers_list,#outliers/up and down reg orfs
                              universeGeneIds = universe,#all of the ORFs
                              pvalueCutoff = 0.05,
                              testDirection = "over"  
                              ) 

#then perform a hypergeometric test on the KEGG terms in the GOstat object
Over <- hyperGTest(params)
summary(Over)
hyperG_KO<- data.frame (summary(Over))

#we then select only the pathways with >9 ORFs associated with them
hyperG_KO_filtered <- hyperG_KO[which(hyperG_KO$Size >9),]
colnames(hyperG_KO_filtered)[1] <- "KO"
hyperG_KO_filtered$KO <- paste0("K", hyperG_KO_filtered$KO)

#enrichedORFs <- geneIdsByCategory(Over) #this gets the genes involved in the 'count'/outliers

allORFs <- geneIdUniverse(Over) # this gets list of all ORFS used in analysis
allORFs_df<- data.frame((sapply(allORFs, '[', seq(max(sapply(allORFs, length))))))#convert this list into DF

allORFs_df$placeholder <- "placeholder"

allORFs_df <- melt(allORFs_df, id.vars="placeholder")
allORFs_df<- na.exclude(allORFs_df)#put the df in long format 
names(allORFs_df)<- c("placeholder", "KO", "orf_id")
allORFs_df$KO <- str_sub(allORFs_df$KO, start=2)
allORFs_df$KO <- paste0("K", allORFs_df$KO)#fix the KO number. 
length(unique(allORFs_df$KO))

#now this df includes ALL of the orfs/KOs used in the analysis. But we want a cut off of >9 ORFs per KO
#so, we merge the dataframes to only select for KOs with >9 ORFs

DE_orfs <- merge(allORFs_df, hyperG_KO_filtered, by = c("KO"))
length(unique(DE_orfs$KO))
DE_orfs_2 <- merge(DE_orfs, frag_all_KO, by = c("KO", "orf_id"))
DE_orfs_3 <- merge(DE_orfs_2, kegg_2, by = c ("KO")) #then we megrge with kegg dataframe to get annotation and DE

#then we make dataframes for all the enriched pathways (temp and fe) and export them as .csv so we don't have to do enrichment every time. 
#Frag_temp_posDE <- DE_orfs_3 
#Frag_temp_posDE$DE <- "pos_temp"

Frag_Fe_negDE <- DE_orfs_3 
Frag_Fe_negDE$DE <- "neg_Fe"

Frag_FeDE <- rbind(Frag_Fe_negDE, Frag_Fe_posDE)

#combine the category map we made with the annotations from the CKegg
Frag_FeDE$pathway <- categorymap[Frag_FeDE$C]

Frag_FeDE$absFeDE <- abs(Frag_FeDE$Fe_vs_noFe_de)


write.csv(file = "Frag_FeDE_KOPathway.csv",Frag_tempDE)

a<- ggplot(Frag_FeDE, aes(x = Fe_vs_noFe_foldchange, y = str_wrap(pathway), color = str_wrap(absFeDE)))+
  #geom_point(alpha = 0.6, size =5)+
  geom_jitter(alpha=0.8, size = 5 ,width = 0,  height = 0.1)+
  scale_color_manual(values = c("gray", "black"), 
                     labels = c("non-significant", "significant"))+
  xlab ("Log2 fold-change")+
  ylab ("KO Pathway")+
  geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )+
  theme_bw()+
  theme(axis.text.x = element_text(
            hjust=1, size = 14, face = "bold", color = "black"))+
  theme(axis.title.x = element_text(
          size = 20, face = "bold"))+
  theme(axis.text.y = element_blank())+
  theme(axis.title.y = element_blank())+
  #theme(axis.title.y = element_text(
         # size = 20, face = "bold"))+
  #theme(axis.text.y = element_text(
            #hjust=1,size = 12, face = "bold", color="black"))+
  guides(color=guide_legend(title="Differential Expression"))+
  theme(legend.title = element_text(color = "black", size = 18), 
        legend.text = element_text(color = "black", size = 18)) 
        #legend.position = "top")


b<- ggplot(Frag_FeDE, aes(x = DE, y = str_wrap(pathway)))+
  geom_point(size = 4, shape = "square")+ 
  theme_bw()+
  ylab ("KO Pathway")+
  xlab (" ")+
  scale_x_discrete(labels= c("Depleted", "Enriched"))+
  theme(axis.title.y = element_text(
         size = 20, face = "bold"))+
  theme(axis.text.y = element_text(
            hjust=1,size = 12, face = "bold", color="black"))+
  theme(axis.text.x = element_text(
     size =12, color = "black", face = "bold"))+
    theme(axis.title.x = element_text( size = 20, face = "bold"))
      
   


c<- grid.arrange(b, a, nrow = 1)

photosynthesis <- dplyr::filter(Frag_FeDE, grepl("Photosynthesis", C))

photo <- dplyr::filter(frag_ORF,grepl("contig_1077755_1_247_+|contig_1210275_1_754_+|contig_1077660_1_409_+|contig_1181458_415_1194_-", orf_id))

```

GOstats using KO: Pseudo-nitzschia 
```{r}
frag_KO <- dplyr::filter (all_ORF, grepl('Pseudo', Taxa.group))

#trim down the dataframe to make it easier to look at. 
frag_KO<- frag_KO[c(47,53, 55, 2, 4, 8, 10 )] 

#select only the genes with KO terms associated with them. This is the 'universe'.
frag_all_KO <- dplyr::filter (frag_KO, grepl('K', KO)) 
length(unique(frag_all_KO$KO_pathway))

##create a KO dataframe. The GoStats package requires two columns for the 
#KO analysis, a gene ID column and KO number column. The KO number cannot begin with a K. 
frag_all_KO_universe <- frag_all_KO [c(2,1)]
frag_all_KO_universe$KO_id <- str_sub(frag_all_KO_universe$KO, start=2)
frag_all_KO_universe <- frag_all_KO_universe [c(3,2)]

#after having the dataframe in the correct format, we convert it into a 'keggframe' which can be used by GOstats
keggFrame=KEGGFrame(frag_all_KO_universe,organism="Pseudo-nitzschia")

#Then the keggframe is passed through the KEGG collection to standardize things."we pass this kind of custom information around, we have chosen to support geneSetCollection objects from GSEABase package." 
#gsc = gene set collection
gsc <- GeneSetCollection(keggFrame, setType = KEGGCollection())

#then we specify our univese. this is a list of all the orfs in the frag list that have KO term associated with them.
universe <- names(keggFrame) 

#Now we select a list of ORFs with something interesting (outliers). So, this can be ORFs with DE with iron/temp. in any direction etc. 
outliers<- frag_all_KO[which(frag_all_KO$Fe_vs_noFe_de >0 ) ,]
  
outliers_list <- split(outliers$KO, outliers$orf_id, drop = TRUE)
outliers_list <- names(outliers_list) #list of orfs that are outliers (DE, DeUP, DeDown etc)

#then make a GOstat object, which will be used for the KEGG GSEA analysis
params <- GSEAKEGGHyperGParams(name="TFG_GSEA",
                              geneSetCollection=gsc,
                              geneIds = outliers_list,#outliers/up and down reg orfs
                              universeGeneIds = universe,#all of the ORFs
                              pvalueCutoff = 0.05,
                              testDirection = "over"  
                              ) 

#then perform a hypergeometric test on the KEGG terms in the GOstat object
Over <- hyperGTest(params)
summary(Over)
hyperG_KO<- data.frame (summary(Over))

#we then select only the pathways with >9 ORFs associated with them
hyperG_KO_filtered <- hyperG_KO[which(hyperG_KO$Size >9),]
colnames(hyperG_KO_filtered)[1] <- "KO"
hyperG_KO_filtered$KO <- paste0("K", hyperG_KO_filtered$KO)

#enrichedORFs <- geneIdsByCategory(Over) #this gets the genes involved in the 'count'/outliers

allORFs <- geneIdUniverse(Over) # this gets list of all ORFS used in analysis
allORFs_df<- data.frame((sapply(allORFs, '[', seq(max(sapply(allORFs, length))))))#convert this list into DF

allORFs_df$placeholder <- "placeholder"

allORFs_df <- melt(allORFs_df, id.vars="placeholder")
allORFs_df<- na.exclude(allORFs_df)#put the df in long format 
names(allORFs_df)<- c("placeholder", "KO", "orf_id")
allORFs_df$KO <- str_sub(allORFs_df$KO, start=2)
allORFs_df$KO <- paste0("K", allORFs_df$KO)#fix the KO number. 
length(unique(allORFs_df$KO))

#now this df includes ALL of the orfs/KOs used in the analysis. But we want a cut off of >9 ORFs per KO
#so, we merge the dataframes to only select for KOs with >9 ORFs

DE_orfs <- merge(allORFs_df, hyperG_KO_filtered, by = c("KO"))
length(unique(DE_orfs$KO))
DE_orfs_2 <- merge(DE_orfs, frag_all_KO, by = c("KO", "orf_id"))
DE_orfs_3 <- merge(DE_orfs_2, kegg_2, by = c ("KO")) #then we megrge with kegg dataframe to get annotation and DE

#then we make dataframes for all the enriched pathways (temp and fe) and export them as .csv so we don't have to do enrichment every time. 
#Pn_temp_posDE <- DE_orfs_3 
#Pn_temp_posDE$DE <- "pos_temp"
length(unique(Pn_temp_posDE$KO))

Pn_Fe_posDE <- DE_orfs_3 
Pn_Fe_posDE$DE <- "pos_Fe"


Pn_FeDE <- rbind(Pn_Fe_negDE, Pn_Fe_posDE)

#combine the category map we made with the annotations from the CKegg
Pn_FeDE$pathway <- categorymap[Pn_FeDE$C]

Pn_FeDE$absFeDE <- abs(Pn_FeDE$Fe_vs_noFe_de)


write.csv(file = "Pn_FeDE_KOPathway.csv",Pn_FeDE)

a<- ggplot(Pn_FeDE, aes(x = Fe_vs_noFe_foldchange, y = str_wrap(pathway), color = str_wrap(absFeDE)))+
  #geom_point(alpha = 0.6, size =5)+
  geom_jitter(alpha=0.8, size = 5 ,width = 0,  height = 0.1)+
  scale_color_manual(values = c("gray", "black"), 
                     labels = c("non-significant", "significant"))+
  xlab ("Log2 fold-change")+
  ylab ("KO Pathway")+
  geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )+
  theme_bw()+
  theme(axis.text.x = element_text(
            hjust=1, size = 14, face = "bold", color = "black"))+
  theme(axis.title.x = element_text(
          size = 20, face = "bold"))+
  theme(axis.text.y = element_blank())+
  theme(axis.title.y = element_blank())+
  #theme(axis.title.y = element_text(
         # size = 20, face = "bold"))+
  #theme(axis.text.y = element_text(
            #hjust=1,size = 12, face = "bold", color="black"))+
  guides(color=guide_legend(title="Differential Expression"))+
  theme(legend.title = element_text(color = "black", size = 18), 
        legend.text = element_text(color = "black", size = 18)) 
        #legend.position = "top")


b<- ggplot(Pn_FeDE, aes(x = DE, y = str_wrap(pathway)))+
  geom_point(size = 4, shape = "square")+ 
  theme_bw()+
  ylab ("KO Pathway")+
  xlab (" ")+
  scale_x_discrete(labels= c("Depleted", "Enriched"))+
  theme(axis.title.y = element_text(
         size = 20, face = "bold"))+
  theme(axis.text.y = element_text(
            hjust=1,size = 12, face = "bold", color="black"))+
  theme(axis.text.x = element_text(
     size =12, color = "black", face = "bold"))+
    theme(axis.title.x = element_text( size = 20, face = "bold"))
      

c<- grid.arrange(b, a, nrow = 1)



ribosome <- dplyr::filter(all_ORF, grepl("Ribosom|ribosom", KO_pathway))
ribosome_2 <- dplyr::filter(ribosome, grepl("Frag", Taxa.group))

Pn_tempDE_KOPathway <- read.csv("Pn_tempDE_KOPathway.csv", header=TRUE)
PN <- dplyr::filter(Pn_tempDE_KOPathway, grepl("sulfur|Sulf",pathway))
PN2 <- merge(PN, pseudo_ORF, by = "orf_id")
```




plotting GO enrichment - bar chart with left and right. 
```{r}
frag_fe_pos_enrich <- read.csv("frag_fe_pos_GO.csv", header = TRUE)
frag_fe_pos_enrich <- frag_fe_pos_enrich [c(1:4)]
frag_fe_pos_enrich$direction <- "positive"

#colnames(frag_fe_pos_enrich) <- paste(colnames(frag_fe_pos_enrich), "pos", sep = "_")


frag_fe_neg_enrich <- read.csv("frag_fe_neg_GO.csv", header = TRUE)
frag_fe_neg_enrich <- frag_fe_neg_enrich [c(1:4)]
frag_fe_neg_enrich$direction <- "negative"
frag_fe_neg_enrich$Genes.in.list <- -1* frag_fe_neg_enrich$Genes.in.list
#colnames(frag_fe_neg_enrich) <- paste(colnames(frag_fe_neg_enrich), "neg", sep = "_")


frag_fe_enrich <- rbind(frag_fe_neg_enrich, frag_fe_pos_enrich)

ggplot(frag_fe_enrich, aes(y = Genes.in.list, x = Functional.Category))+
     #geom_point(alpha = 0.8, size = 3)+
  geom_bar(stat="identity", aes(fill = direction))+
  scale_fill_manual(values = c("grey", "black"))+
coord_flip()+
  theme_bw()
        
```


